
###Outlier and allele frequency correlation analyses using Bayenv2

`Bayen2` was used to for identifying candidate outlier loci as well as loci for which population allele frequencies displayed strong correlations with morphological measurments.

SNP dataset generated via *populations*; details:
 - data for all 4 populations, minimum 5 individuals per population
 - one single SNP per RAD tag

The Jupyter notebook `Diplotaxodon_Bayenv2_excl_singletons_2.ipynb` details the following steps:
 - convert SNP data from `*.vcf` as generated by *populations* to suitable `Bayenv2` input format, filtering out singletons SNPs
 - construct of Covariance matrix to be used as neutral model
 - normalize morphological measurements to be used as 'environmental varables' in Bayenv analyses and convert to suitable input format
 - run 20 indpendent `Bayenv2` analyses
 - calculate per locus rank statistics across the 20 runs
 - identify highly correlated/candidate outlier loci and do visualizations of results
 - identify candidate genes in vicinity of candidate loci


