{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def parse_popmap(popmap, individual_dict):\n",
    "    \"\"\"\n",
    "    The function parses a text (i.e. a list of tags)\n",
    "    and puts the content of the first column of the \n",
    "    into a list\"\"\"\n",
    "    \n",
    "    import gzip\n",
    "#    from collections import defaultdict\n",
    "    \n",
    "#    pop_dict = defaultdict(dict)\n",
    "    \n",
    "    if popmap.endswith('.gz'):\n",
    "        FH = gzip.open(popmap,'rb')\n",
    "    else:\n",
    "        FH = open(popmap,'r')\n",
    "        \n",
    "    for line in FH:\n",
    "        line = line.strip()\n",
    "        per_line = line.split('\\t')\n",
    "        individual_dict[per_line[0]]['population'] = per_line[1]\n",
    "    \n",
    "    FH.close()\n",
    "    \n",
    "#    return pop_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['sample_BBNB1', 'sample_BBNB10', 'sample_BBNB2', 'sample_BBNB3', 'sample_BBNB4', 'sample_BBNB5', 'sample_BBNB6', 'sample_BBNB7', 'sample_BBNB8', 'sample_BBNB9', 'sample_LBPNK2', 'sample_LBPNK3', 'sample_LBPNK4', 'sample_LBPNK5', 'sample_LBPNK6', 'sample_LBPNK7', 'sample_MONB11', 'sample_MONB2', 'sample_MONB3', 'sample_MONB4', 'sample_MONB5', 'sample_MONB7', 'sample_MONB8', 'sample_MONB9', 'sample_NJ1', 'sample_NJ10', 'sample_NJ2', 'sample_NJ3', 'sample_NJ4', 'sample_NJ5', 'sample_NJ6', 'sample_NJ7', 'sample_NJ8', 'sample_NJ9']\n"
     ]
    }
   ],
   "source": [
    "from collections import defaultdict\n",
    "\n",
    "ind_dict = defaultdict(dict)\n",
    "\n",
    "parse_popmap(popmap='../popmap', individual_dict=ind_dict)\n",
    "print sorted(ind_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def parse_taglist(tagfile):\n",
    "    \"\"\"\n",
    "    The function parses a text (i.e. a list of tags)\n",
    "    and puts the content of the first column of the \n",
    "    into a list\"\"\"\n",
    "    \n",
    "    import gzip\n",
    "    taglistt = []\n",
    "    \n",
    "    if tagfile.endswith('.gz'):\n",
    "        FH = gzip.open(tagfile,'rb')\n",
    "    else:\n",
    "        FH = open(tagfile,'r')\n",
    "        \n",
    "    for line in FH:\n",
    "        line = line.strip()\n",
    "        per_line = line.split('\\t')\n",
    "        taglistt.append(per_line[0])\n",
    "    \n",
    "    FH.close()\n",
    "    \n",
    "    return taglistt\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "taglist = []\n",
    "taglist = parse_taglist(tagfile = '../whitelist.txt')\n",
    "#print taglist"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "parse catalog\n",
    "optional keep only relevant tags"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def parse_catalog(catalog, tag_whitelist=[]):\n",
    "    \"\"\"\n",
    "    The function parses a stacks catalog\n",
    "    \"\"\"\n",
    "    import gzip\n",
    "    from collections import defaultdict\n",
    "    \n",
    "    catalog_dict_local = defaultdict(list)\n",
    "    whitelist = []\n",
    "    \n",
    "    if tag_whitelist:\n",
    "        print \"You have provided a whitelist of tags\\n\"\n",
    "        whitelist = tag_whitelist[:]\n",
    "    \n",
    "    if catalog.endswith('.gz'):\n",
    "        print \"open gzipped catalog\\n\"\n",
    "        FH_c = gzip.open(catalog,'rb')\n",
    "    else:\n",
    "        FH_c = open(catalog,'r')\n",
    "\n",
    "#    for i in range(100):\n",
    "#        line = FH_c.readline()\n",
    "    for line in FH_c:\n",
    "        line = line.strip()\n",
    "        per_line = line.split('\\t')\n",
    "        if not tag_whitelist:\n",
    "            catalog_dict_local[per_line[2]] = per_line[8].split(',')\n",
    "        else:\n",
    "            if whitelist:\n",
    "                if per_line[2] in whitelist:\n",
    "                    catalog_dict_local[int(per_line[2])] = per_line[8].split(',')\n",
    "                    whitelist.remove(per_line[2])\n",
    "                \n",
    "            else:\n",
    "                print \"Done parsing all relevant tags (n=%i)\\n\" %len(tag_whitelist)\n",
    "                break\n",
    "                \n",
    "    FH_c.close()\n",
    "    \n",
    "    if tag_whitelist:\n",
    "        print \"Parsed %i of %i desired tags\\n\" %(len(catalog_dict_local), len(tag_whitelist))\n",
    "    else:\n",
    "        print \"Parsed %i tags\\n\" %(len(catalog_dict_local))\n",
    "\n",
    "#    print sorted(catalog_dict)\n",
    "    \n",
    "    return catalog_dict_local\n",
    "        \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "open gzipped catalog\n",
      "\n",
      "Parsed 74001 tags\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#parse full catalog\n",
    "catalog_dict = {}\n",
    "catalog_dict = parse_catalog(catalog='../../../../stacks/ALL/mapping/excl_PCR_dupl/BWA-8MM/M_zebra/2-cstacks/m5/n0/data/batch_1.catalog.tags.tsv.gz')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "You have provided a whitelist of tags\n",
      "\n",
      "open gzipped catalog\n",
      "\n",
      "Done parsing all relevant tags (n=11786)\n",
      "\n",
      "Parsed 11786 of 11786 desired tags\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#parse catalog using a whitelist\n",
    "catalog_dict= {}\n",
    "#test_list = ['1','2','3','9']\n",
    "catalog_dict = parse_catalog(catalog='../../../../stacks/ALL/mapping/excl_PCR_dupl/BWA-8MM/M_zebra/2-cstacks/m5/n0/data/batch_1.catalog.tags.tsv.gz', tag_whitelist=taglist)\n",
    "#print sorted(catalog_dict)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "find stacks synonyms for sample names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def find_stacks_ids(ind_list, stacks_dir, individual_dict):\n",
    "    \"\"\"\n",
    "    The function finds the internal stacks ids for a list of sample names\n",
    "    \"\"\"\n",
    "    \n",
    "    import os.path\n",
    "    import glob\n",
    "    import gzip\n",
    "\n",
    "    if type(ind_list) is list: #check the value passed through ind_list is already a list\n",
    "        print \"IDs for individuals provided in list\"\n",
    "        sample_list = ind_list[:]\n",
    "    else: #if it's not a list check if it's a file and parse the first column\n",
    "        if not os.path.isfile(ind_list):\n",
    "            IOError('ind_list needs to be either a txt file or an existing list')\n",
    "        print \"parsing IDs for individuals from file\"\n",
    "        sample_list = []\n",
    "        sample_list = parse_taglist(ind_list)\n",
    "\n",
    "    \n",
    "    file_list = glob.glob(stacks_dir+'*tags*')\n",
    "    if len(file_list) == 0:\n",
    "        IOError('no tags files detected in the directory that you have specified')\n",
    "#    print file_list\n",
    "\n",
    "    #check if there is a file for all samples in the list\n",
    "    for sample in sorted(sample_list):\n",
    "#        print sample\n",
    "        found_tags_file = \"\"\n",
    "        if file_list:\n",
    "            for tags_file in sorted(file_list):\n",
    "                if sample in tags_file:\n",
    "#                    print tags_file\n",
    "                    found_tags_file = tags_file\n",
    "                    file_list.remove(tags_file)\n",
    "                    break\n",
    "        \n",
    "        if not found_tags_file:#individual_dict[sample].has_key('tags_file'):\n",
    "            raise IOError(\"did not find a tags file for sample %s\" %sample)\n",
    "        else:\n",
    "            individual_dict[sample]['tags_file'] = os.path.abspath(found_tags_file)\n",
    "#            print individual_dict[sample]['tags_file']\n",
    "            if found_tags_file.endswith('.gz'):\n",
    "                tags_FH = gzip.open(found_tags_file, 'rb')\n",
    "            else:\n",
    "                tags_FH = open(found_tags_file, 'r')\n",
    "            \n",
    "            ID = tags_FH.readline().split(\"\\t\")[1]\n",
    "            tags_FH.close()\n",
    "            individual_dict[sample]['catalog_id'] = ID\n",
    "            print \"%s -> id: %s\" %(sample,ID)\n",
    "            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "IDs for individuals provided in list\n",
      "sample_BBNB1 -> id: 149\n",
      "sample_BBNB10 -> id: 148\n",
      "sample_BBNB2 -> id: 150\n",
      "sample_BBNB3 -> id: 151\n",
      "sample_BBNB4 -> id: 152\n",
      "sample_BBNB5 -> id: 153\n",
      "sample_BBNB6 -> id: 154\n",
      "sample_BBNB7 -> id: 155\n",
      "sample_BBNB8 -> id: 156\n",
      "sample_BBNB9 -> id: 157\n",
      "sample_LBPNK2 -> id: 180\n",
      "sample_LBPNK3 -> id: 181\n",
      "sample_LBPNK4 -> id: 182\n",
      "sample_LBPNK5 -> id: 183\n",
      "sample_LBPNK6 -> id: 184\n",
      "sample_LBPNK7 -> id: 185\n",
      "sample_MONB11 -> id: 209\n",
      "sample_MONB2 -> id: 210\n",
      "sample_MONB3 -> id: 211\n",
      "sample_MONB4 -> id: 212\n",
      "sample_MONB5 -> id: 213\n",
      "sample_MONB7 -> id: 214\n",
      "sample_MONB8 -> id: 215\n",
      "sample_MONB9 -> id: 216\n",
      "sample_NJ1 -> id: 227\n",
      "sample_NJ10 -> id: 226\n",
      "sample_NJ2 -> id: 228\n",
      "sample_NJ3 -> id: 229\n",
      "sample_NJ4 -> id: 230\n",
      "sample_NJ5 -> id: 231\n",
      "sample_NJ6 -> id: 232\n",
      "sample_NJ7 -> id: 233\n",
      "sample_NJ8 -> id: 234\n",
      "sample_NJ9 -> id: 235\n"
     ]
    }
   ],
   "source": [
    "find_stacks_ids(ind_list=ind_dict.keys(), individual_dict=ind_dict, stacks_dir='../../../../stacks/ALL/mapping/excl_PCR_dupl/BWA-8MM/M_zebra/1-pstacks/m5/') #ind_dict.keys(), stacks_dir='../stacks_data/pstacks/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def find_tag_ids_per_individual(ind_list, individual_dict, catalog, catalog_tag_whitelist=[]):\n",
    "    \"\"\"\n",
    "    The function finds the read ids for all relevant\n",
    "    tags\n",
    "    \"\"\"\n",
    "    \n",
    "    import os.path\n",
    "    import gzip\n",
    "    \n",
    "    catalog_ind_ids = {}\n",
    "    \n",
    "    if type(ind_list) is list: #check the value passed through ind_list is already a list\n",
    "        print \"IDs for individuals provided in list\"\n",
    "        sample_list = ind_list[:]\n",
    "    else: #if it's not a list check if it's a file and parse the first column\n",
    "        if not os.path.isfile(ind_list):\n",
    "            IOError('ind_list needs to be either a txt file or an existing list')\n",
    "        print \"parsing IDs for individuals from file\"\n",
    "        sample_list = []\n",
    "        sample_list = parse_taglist(ind_list)\n",
    "\n",
    "    for ind in sample_list:\n",
    "        if not individual_dict.has_key(ind):\n",
    "            raise IOError('no data found for %s' %ind)\n",
    "        else:\n",
    "            print \"Found data for %s\" %ind\n",
    "            catalog_ind_ids[ind_dict[ind]['catalog_id']] = ind\n",
    "            ind_dict[ind]['tags'] = {}\n",
    "    \n",
    "    if not catalog_tag_whitelist:\n",
    "        catalog_tag_whitelist = [int(i) for i in sorted(catalog.keys()[:])] #sorted(catalog.keys()[:])\n",
    "    \n",
    "    print \"Screening through %i tag ids\" %len(catalog_tag_whitelist)\n",
    "    \n",
    "    for tag in catalog_tag_whitelist:\n",
    "#        print str(tag)+\":\"\n",
    "        for s in catalog[tag]:\n",
    "            split = s.split(\"_\")\n",
    "            if catalog_ind_ids.has_key(split[0]):\n",
    "                ind_dict[catalog_ind_ids[split[0]]]['tags'][int(tag)] = int(split[1]) #[int(split[1])] = int(tag)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "IDs for individuals provided in list\n",
      "Found data for sample_LBPNK3\n",
      "Found data for sample_LBPNK2\n",
      "Found data for sample_LBPNK5\n",
      "Found data for sample_LBPNK4\n",
      "Found data for sample_LBPNK7\n",
      "Found data for sample_MONB11\n",
      "Found data for sample_BBNB7\n",
      "Found data for sample_BBNB6\n",
      "Found data for sample_BBNB5\n",
      "Found data for sample_BBNB4\n",
      "Found data for sample_BBNB3\n",
      "Found data for sample_BBNB2\n",
      "Found data for sample_BBNB1\n",
      "Found data for sample_LBPNK6\n",
      "Found data for sample_BBNB9\n",
      "Found data for sample_BBNB8\n",
      "Found data for sample_MONB3\n",
      "Found data for sample_MONB2\n",
      "Found data for sample_MONB7\n",
      "Found data for sample_MONB5\n",
      "Found data for sample_MONB4\n",
      "Found data for sample_MONB9\n",
      "Found data for sample_MONB8\n",
      "Found data for sample_BBNB10\n",
      "Found data for sample_NJ10\n",
      "Found data for sample_NJ9\n",
      "Found data for sample_NJ8\n",
      "Found data for sample_NJ1\n",
      "Found data for sample_NJ3\n",
      "Found data for sample_NJ2\n",
      "Found data for sample_NJ5\n",
      "Found data for sample_NJ4\n",
      "Found data for sample_NJ7\n",
      "Found data for sample_NJ6\n",
      "Screening through 11786 tag ids\n"
     ]
    }
   ],
   "source": [
    "find_tag_ids_per_individual(ind_list=ind_dict.keys(), individual_dict=ind_dict, catalog=catalog_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['sample_LBPNK3', 'sample_LBPNK2', 'sample_LBPNK5', 'sample_LBPNK4', 'sample_LBPNK7', 'sample_MONB11', 'sample_BBNB7', 'sample_BBNB6', 'sample_BBNB5', 'sample_BBNB4', 'sample_BBNB3', 'sample_BBNB2', 'sample_BBNB1', 'sample_LBPNK6', 'sample_BBNB9', 'sample_BBNB8', 'sample_MONB3', 'sample_MONB2', 'sample_MONB7', 'sample_MONB5', 'sample_MONB4', 'sample_MONB9', 'sample_MONB8', 'sample_BBNB10', 'sample_NJ10', 'sample_NJ9', 'sample_NJ8', 'sample_NJ1', 'sample_NJ3', 'sample_NJ2', 'sample_NJ5', 'sample_NJ4', 'sample_NJ7', 'sample_NJ6']\n",
      "sample_BBNB1\n",
      "11747\n",
      "sample_BBNB10\n",
      "11722\n",
      "sample_BBNB2\n",
      "11286\n",
      "sample_BBNB3\n",
      "11770\n",
      "sample_BBNB4\n",
      "11763\n",
      "sample_BBNB5\n",
      "11764\n",
      "sample_BBNB6\n",
      "11698\n",
      "sample_BBNB7\n",
      "11784\n",
      "sample_BBNB8\n",
      "11768\n",
      "sample_BBNB9\n",
      "11767\n",
      "sample_LBPNK2\n",
      "11779\n",
      "sample_LBPNK3\n",
      "11388\n",
      "sample_LBPNK4\n",
      "10810\n",
      "sample_LBPNK5\n",
      "11725\n",
      "sample_LBPNK6\n",
      "11781\n",
      "sample_LBPNK7\n",
      "11773\n",
      "sample_MONB11\n",
      "11768\n",
      "sample_MONB2\n",
      "11776\n",
      "sample_MONB3\n",
      "10884\n",
      "sample_MONB4\n",
      "11776\n",
      "sample_MONB5\n",
      "11215\n",
      "sample_MONB7\n",
      "11393\n",
      "sample_MONB8\n",
      "11772\n",
      "sample_MONB9\n",
      "11446\n",
      "sample_NJ1\n",
      "11753\n",
      "sample_NJ10\n",
      "11780\n",
      "sample_NJ2\n",
      "11701\n",
      "sample_NJ3\n",
      "11455\n",
      "sample_NJ4\n",
      "11431\n",
      "sample_NJ5\n",
      "11726\n",
      "sample_NJ6\n",
      "11269\n",
      "sample_NJ7\n",
      "11775\n",
      "sample_NJ8\n",
      "11521\n",
      "sample_NJ9\n",
      "11569\n"
     ]
    }
   ],
   "source": [
    "print ind_dict.keys()\n",
    "for samp in sorted(ind_dict.keys()):\n",
    "    print samp\n",
    "    print len(ind_dict[samp]['tags'])\n",
    "#    print len(ind_dict[samp]['read_ids'].keys())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def find_reads_per_individual(ind_list, individual_dict, catalog, catalog_tag_whitelist=[]):\n",
    "    \"\"\"\n",
    "    The function finds read ids for each individual for each tag\n",
    "    \"\"\"\n",
    "    import os.path\n",
    "    import gzip\n",
    "    \n",
    "    catalog_ind_ids = {}\n",
    "    \n",
    "    if type(ind_list) is list: #check the value passed through ind_list is already a list\n",
    "        print \"IDs for individuals provided in list\"\n",
    "        sample_list = ind_list[:]\n",
    "    else: #if it's not a list check if it's a file and parse the first column\n",
    "        if not os.path.isfile(ind_list):\n",
    "            IOError('ind_list needs to be either a txt file or an existing list')\n",
    "        print \"parsing IDs for individuals from file\"\n",
    "        sample_list = []\n",
    "        sample_list = parse_taglist(ind_list)\n",
    "\n",
    "    if not catalog_tag_whitelist:\n",
    "        catalog_tag_whitelist = [int(i) for i in sorted(catalog.keys()[:])]\n",
    "    \n",
    "    for ind in sample_list:\n",
    "        if not individual_dict.has_key(ind):\n",
    "            raise IOError('no data found for %s' %ind)\n",
    "        else:\n",
    "            print \"Found data for %s\" %ind\n",
    "            catalog_ind_ids[ind_dict[ind]['catalog_id']] = ind\n",
    "#            ind_dict[ind]['tags'] = {}\n",
    "    \n",
    "    print \"Screening through %i tag ids\" %len(catalog_tag_whitelist)\n",
    "    \n",
    "    for ind in sorted(sample_list):\n",
    "        print \"#################\\n\"\n",
    "        print ind\n",
    "#        print ind_dict[ind]['tags']\n",
    "#        print sorted(ind_dict[ind]['tags'])\n",
    "        ind_catalog_tags_to_process = []\n",
    "        \n",
    "\n",
    "#        print ind_dict[ind]['tags'].keys()\n",
    "\n",
    "        for catalog_tag in catalog_tag_whitelist:\n",
    "            if ind_dict[ind]['tags'].has_key(catalog_tag):\n",
    "                ind_catalog_tags_to_process.append(ind_dict[ind]['tags'][catalog_tag])\n",
    "        \n",
    "        if not ind_catalog_tags_to_process: #if this individual contains any relevant tags\n",
    "            print \"nothing to process for this indidividual\"\n",
    "            continue #go to next individual\n",
    "            \n",
    "        print \"%s individual tags to process:\\n\" %len(ind_catalog_tags_to_process)\n",
    "        ind_dict[ind]['read_ids'] = {}\n",
    "        for to_process in ind_catalog_tags_to_process:\n",
    "            ind_dict[ind]['read_ids'][int(to_process)] = []\n",
    "            \n",
    "        #open file\n",
    "        if ind_dict[ind]['tags_file'].endswith('.gz'):\n",
    "            ind_tags_FH = gzip.open(ind_dict[ind]['tags_file'], 'rb')\n",
    "        else:\n",
    "            ind_tags_FH = open(ind_dict[ind]['tags_file'], 'r')\n",
    "            \n",
    "        add = False \n",
    "\n",
    "        for line in [line.strip() for line in ind_tags_FH]:\n",
    "            if not ind_catalog_tags_to_process: #if there is tags left to be processed\n",
    "                break\n",
    "\n",
    "            if 'consensus' in line:\n",
    "                if add: #that means that the previous tag was relevant\n",
    "#                    print \"done with tag %s - found %s reads - will be removed from list\" %(str(t), len(ind_dict[ind]['read_ids'][int(t)]))\n",
    "                    ind_catalog_tags_to_process.remove(int(t))\n",
    "#                    print \"still to process:\\n%s\" %ind_catalog_tags_to_process\n",
    "\n",
    "                add = False #per default a given tag is not to be added\n",
    "                t = line.split(\"\\t\")[2]\n",
    "#                print \"\\n\\nFound consensus for tag: %s\" %t\n",
    "                if int(t) in ind_catalog_tags_to_process:\n",
    "#                    print \"Tag %s is ON THE LIST\" %t\n",
    "                    ind_dict[ind]['read_ids'][int(t)] = []\n",
    "                    add = True\n",
    "            elif add and not 'model' in line:\n",
    "                read = line.split(\"\\t\")[8]\n",
    "#                print \"FOUND READ: %s - append to %s\" %(read, t)\n",
    "                ind_dict[ind]['read_ids'][int(t)].append(read)\n",
    "#                ind_dict[ind]['reads'][t].append(read)\n",
    "\n",
    "        if add and int(t) in ind_catalog_tags_to_process:\n",
    "            ind_catalog_tags_to_process.remove(int(t))\n",
    "        print \"\\nDONE WITH INDIVIDUAL %s\" %(ind)\n",
    "        print \"processed %i of %i tags\\n\" %((len(ind_dict[ind]['read_ids'].keys())-len(ind_catalog_tags_to_process)), len(ind_dict[ind]['read_ids'].keys()))\n",
    "        if ind_catalog_tags_to_process:\n",
    "            print \"not processed:\\n%s\" %ind_catalog_tags_to_process\n",
    "        ind_tags_FH.close()\n",
    "\n",
    "                    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "IDs for individuals provided in list\n",
      "Found data for sample_LBPNK3\n",
      "Found data for sample_LBPNK2\n",
      "Found data for sample_LBPNK5\n",
      "Found data for sample_LBPNK4\n",
      "Found data for sample_LBPNK7\n",
      "Found data for sample_MONB11\n",
      "Found data for sample_BBNB7\n",
      "Found data for sample_BBNB6\n",
      "Found data for sample_BBNB5\n",
      "Found data for sample_BBNB4\n",
      "Found data for sample_BBNB3\n",
      "Found data for sample_BBNB2\n",
      "Found data for sample_BBNB1\n",
      "Found data for sample_LBPNK6\n",
      "Found data for sample_BBNB9\n",
      "Found data for sample_BBNB8\n",
      "Found data for sample_MONB3\n",
      "Found data for sample_MONB2\n",
      "Found data for sample_MONB7\n",
      "Found data for sample_MONB5\n",
      "Found data for sample_MONB4\n",
      "Found data for sample_MONB9\n",
      "Found data for sample_MONB8\n",
      "Found data for sample_BBNB10\n",
      "Found data for sample_NJ10\n",
      "Found data for sample_NJ9\n",
      "Found data for sample_NJ8\n",
      "Found data for sample_NJ1\n",
      "Found data for sample_NJ3\n",
      "Found data for sample_NJ2\n",
      "Found data for sample_NJ5\n",
      "Found data for sample_NJ4\n",
      "Found data for sample_NJ7\n",
      "Found data for sample_NJ6\n",
      "Screening through 11786 tag ids\n",
      "#################\n",
      "\n",
      "sample_BBNB1\n",
      "11747 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_BBNB1\n",
      "processed 11747 of 11747 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_BBNB10\n",
      "11722 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_BBNB10\n",
      "processed 11722 of 11722 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_BBNB2\n",
      "11286 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_BBNB2\n",
      "processed 11286 of 11286 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_BBNB3\n",
      "11770 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_BBNB3\n",
      "processed 11770 of 11770 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_BBNB4\n",
      "11763 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_BBNB4\n",
      "processed 11763 of 11763 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_BBNB5\n",
      "11764 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_BBNB5\n",
      "processed 11764 of 11764 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_BBNB6\n",
      "11698 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_BBNB6\n",
      "processed 11698 of 11698 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_BBNB7\n",
      "11784 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_BBNB7\n",
      "processed 11784 of 11784 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_BBNB8\n",
      "11768 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_BBNB8\n",
      "processed 11768 of 11768 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_BBNB9\n",
      "11767 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_BBNB9\n",
      "processed 11767 of 11767 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_LBPNK2\n",
      "11779 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_LBPNK2\n",
      "processed 11779 of 11779 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_LBPNK3\n",
      "11388 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_LBPNK3\n",
      "processed 11388 of 11388 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_LBPNK4\n",
      "10810 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_LBPNK4\n",
      "processed 10810 of 10810 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_LBPNK5\n",
      "11725 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_LBPNK5\n",
      "processed 11725 of 11725 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_LBPNK6\n",
      "11781 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_LBPNK6\n",
      "processed 11781 of 11781 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_LBPNK7\n",
      "11773 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_LBPNK7\n",
      "processed 11773 of 11773 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_MONB11\n",
      "11768 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_MONB11\n",
      "processed 11768 of 11768 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_MONB2\n",
      "11776 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_MONB2\n",
      "processed 11776 of 11776 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_MONB3\n",
      "10884 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_MONB3\n",
      "processed 10884 of 10884 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_MONB4\n",
      "11776 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_MONB4\n",
      "processed 11776 of 11776 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_MONB5\n",
      "11215 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_MONB5\n",
      "processed 11215 of 11215 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_MONB7\n",
      "11393 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_MONB7\n",
      "processed 11393 of 11393 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_MONB8\n",
      "11772 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_MONB8\n",
      "processed 11772 of 11772 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_MONB9\n",
      "11446 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_MONB9\n",
      "processed 11446 of 11446 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_NJ1\n",
      "11753 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_NJ1\n",
      "processed 11753 of 11753 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_NJ10\n",
      "11780 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_NJ10\n",
      "processed 11780 of 11780 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_NJ2\n",
      "11701 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_NJ2\n",
      "processed 11701 of 11701 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_NJ3\n",
      "11455 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_NJ3\n",
      "processed 11455 of 11455 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_NJ4\n",
      "11431 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_NJ4\n",
      "processed 11431 of 11431 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_NJ5\n",
      "11726 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_NJ5\n",
      "processed 11726 of 11726 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_NJ6\n",
      "11269 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_NJ6\n",
      "processed 11269 of 11269 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_NJ7\n",
      "11775 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_NJ7\n",
      "processed 11775 of 11775 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_NJ8\n",
      "11521 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_NJ8\n",
      "processed 11521 of 11521 tags\n",
      "\n",
      "#################\n",
      "\n",
      "sample_NJ9\n",
      "11569 individual tags to process:\n",
      "\n",
      "\n",
      "DONE WITH INDIVIDUAL sample_NJ9\n",
      "processed 11569 of 11569 tags\n",
      "\n"
     ]
    }
   ],
   "source": [
    "find_reads_per_individual(ind_list=ind_dict.keys(), individual_dict=ind_dict, catalog=catalog_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def find_raw_reads(ind_list, individual_dict, raw_reads, pe=False):\n",
    "    \"\"\"\n",
    "    The function checks if there are raw read files present for each sample in the list\n",
    "    \"\"\"\n",
    "    import os.path\n",
    "    import gzip\n",
    "    import glob\n",
    "    \n",
    "#    catalog_ind_ids = {}\n",
    "    \n",
    "    if type(ind_list) is list: #check the value passed through ind_list is already a list\n",
    "        print \"IDs for individuals provided in list\"\n",
    "        sample_list = ind_list[:]\n",
    "    else: #if it's not a list check if it's a file and parse the first column\n",
    "        if not os.path.isfile(ind_list):\n",
    "            IOError('ind_list needs to be either a txt file or an existing list')\n",
    "        print \"parsing IDs for individuals from file\"\n",
    "        sample_list = []\n",
    "        sample_list = parse_taglist(ind_list)\n",
    "\n",
    "    file_list = glob.glob(raw_reads+'/*fastq*')\n",
    "    file_list = sorted(file_list)\n",
    "    if len(file_list) == 0:\n",
    "        IOError('no fastq files detected in the directory that you have specified')\n",
    "\n",
    "#    print \"Files present\\n\"\n",
    "#    print \"\\n\".join(file_list)\n",
    "    \n",
    "    for ind in sorted(sample_list):\n",
    "        print \"%s - checking for presence of reads in %s\" %(ind, raw_reads)\n",
    "    \n",
    "        found_fastq_files = []\n",
    "        if file_list:\n",
    "            for fastq_file in file_list:\n",
    "#                print \"testing %s\" %fastq_file\n",
    "                if ind in fastq_file:\n",
    "#                    print \"found: %s\" %fastq_file\n",
    "                    found_fastq_files.append(os.path.abspath(fastq_file))\n",
    "                    if not pe and len(found_fastq_files) == 1:\n",
    "                        break\n",
    "                    if pe and len(found_fastq_files) == 2:\n",
    "                        break\n",
    "        \n",
    "        if not found_fastq_files:\n",
    "            raise IOError(\"did not find any fastq file for sample %s\" %sample)\n",
    "        else:\n",
    "            print \"Found the following files for %s\\n%s\" %(ind, found_fastq_files)\n",
    "            individual_dict[ind]['fastq_files'] = found_fastq_files\n",
    "        \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "IDs for individuals provided in list\n",
      "sample_BBNB1 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_BBNB1\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB1.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB1.2.fastq.gz']\n",
      "sample_BBNB10 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_BBNB10\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB10.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB10.2.fastq.gz']\n",
      "sample_BBNB2 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_BBNB2\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB2.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB2.2.fastq.gz']\n",
      "sample_BBNB3 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_BBNB3\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB3.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB3.2.fastq.gz']\n",
      "sample_BBNB4 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_BBNB4\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB4.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB4.2.fastq.gz']\n",
      "sample_BBNB5 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_BBNB5\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB5.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB5.2.fastq.gz']\n",
      "sample_BBNB6 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_BBNB6\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB6.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB6.2.fastq.gz']\n",
      "sample_BBNB7 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_BBNB7\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB7.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB7.2.fastq.gz']\n",
      "sample_BBNB8 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_BBNB8\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB8.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB8.2.fastq.gz']\n",
      "sample_BBNB9 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_BBNB9\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB9.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB9.2.fastq.gz']\n",
      "sample_LBPNK2 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_LBPNK2\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_LBPNK2.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_LBPNK2.2.fastq.gz']\n",
      "sample_LBPNK3 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_LBPNK3\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_LBPNK3.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_LBPNK3.2.fastq.gz']\n",
      "sample_LBPNK4 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_LBPNK4\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_LBPNK4.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_LBPNK4.2.fastq.gz']\n",
      "sample_LBPNK5 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_LBPNK5\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_LBPNK5.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_LBPNK5.2.fastq.gz']\n",
      "sample_LBPNK6 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_LBPNK6\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_LBPNK6.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_LBPNK6.2.fastq.gz']\n",
      "sample_LBPNK7 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_LBPNK7\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_LBPNK7.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_LBPNK7.2.fastq.gz']\n",
      "sample_MONB11 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_MONB11\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB11.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB11.2.fastq.gz']\n",
      "sample_MONB2 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_MONB2\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB2.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB2.2.fastq.gz']\n",
      "sample_MONB3 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_MONB3\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB3.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB3.2.fastq.gz']\n",
      "sample_MONB4 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_MONB4\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB4.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB4.2.fastq.gz']\n",
      "sample_MONB5 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_MONB5\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB5.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB5.2.fastq.gz']\n",
      "sample_MONB7 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_MONB7\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB7.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB7.2.fastq.gz']\n",
      "sample_MONB8 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_MONB8\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB8.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB8.2.fastq.gz']\n",
      "sample_MONB9 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_MONB9\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB9.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB9.2.fastq.gz']\n",
      "sample_NJ1 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_NJ1\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ1.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ1.2.fastq.gz']\n",
      "sample_NJ10 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_NJ10\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ10.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ10.2.fastq.gz']\n",
      "sample_NJ2 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_NJ2\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ2.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ2.2.fastq.gz']\n",
      "sample_NJ3 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_NJ3\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ3.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ3.2.fastq.gz']\n",
      "sample_NJ4 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_NJ4\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ4.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ4.2.fastq.gz']\n",
      "sample_NJ5 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_NJ5\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ5.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ5.2.fastq.gz']\n",
      "sample_NJ6 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_NJ6\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ6.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ6.2.fastq.gz']\n",
      "sample_NJ7 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_NJ7\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ7.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ7.2.fastq.gz']\n",
      "sample_NJ8 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_NJ8\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ8.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ8.2.fastq.gz']\n",
      "sample_NJ9 - checking for presence of reads in ../raw_reads/\n",
      "Found the following files for sample_NJ9\n",
      "['/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ9.1.fastq.gz', '/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ9.2.fastq.gz']\n"
     ]
    }
   ],
   "source": [
    "find_raw_reads(ind_list=ind_dict.keys(), individual_dict=ind_dict, raw_reads=\"../raw_reads/\", pe=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def extract_reads(ind_list, individual_dict, catalog, out_dir=\"./\", catalog_tag_whitelist=[], pe=False, per_tag=False):\n",
    "    \"\"\"\n",
    "    The function extracts reads for each individual\n",
    "    into either one file or one file each per tag.\n",
    "    Optional it extracts the paired end reads as well\"\"\"\n",
    "    \n",
    "    import os.path\n",
    "    import gzip\n",
    "    from Bio import SeqIO\n",
    "    \n",
    "    catalog_ind_ids = {}\n",
    "    \n",
    "    if type(ind_list) is list: #check the value passed through ind_list is already a list\n",
    "        print \"IDs for individuals provided in list\"\n",
    "        sample_list = ind_list[:]\n",
    "    else: #if it's not a list check if it's a file and parse the first column\n",
    "        if not os.path.isfile(ind_list):\n",
    "            IOError('ind_list needs to be either a txt file or an existing list')\n",
    "        print \"parsing IDs for individuals from file\"\n",
    "        sample_list = []\n",
    "        sample_list = parse_taglist(ind_list)\n",
    "\n",
    "    if not catalog_tag_whitelist:\n",
    "        catalog_tag_whitelist = [int(i) for i in sorted(catalog.keys()[:])]\n",
    "    \n",
    "    for ind in sample_list:\n",
    "        if not individual_dict.has_key(ind):\n",
    "            raise IOError('no data found for %s' %ind)\n",
    "        else:\n",
    "            if not individual_dict[ind]['fastq_files']:\n",
    "                raise IOError('no fastq files for %s' %s)\n",
    "            catalog_ind_ids[ind_dict[ind]['catalog_id']] = ind\n",
    "\n",
    "    sample_list = sorted(sample_list)\n",
    "    \n",
    "    print \"Fetching reads for %i tags\" %len(catalog_tag_whitelist)\n",
    "    \n",
    "    for ind in sample_list:\n",
    "        print \"#################\\n\"\n",
    "        print ind\n",
    "#        print ind_dict[ind]['tags']\n",
    "#        print sorted(ind_dict[ind]['tags'])\n",
    "        \n",
    "        if not pe and len(individual_dict[ind]['fastq_files']) <= 2:\n",
    "            fq_files = [individual_dict[ind]['fastq_files'][0]]\n",
    "        elif pe and len(individual_dict[ind]['fastq_files']) == 2:\n",
    "            fq_files = individual_dict[ind]['fastq_files'][:]\n",
    "        else:\n",
    "            raise IOError('something wrong with the fastq files for %s' %ind)\n",
    "        \n",
    "\n",
    "        for i in range(len(fq_files)):\n",
    "            the_file = fq_files[i]\n",
    "            file_id = i+1\n",
    "            print \"processing file %i: %s\\n\" %(file_id, the_file)\n",
    "                        \n",
    "            #open file\n",
    "#            the_file = individual_dict[ind]['fastq_files'][0]\n",
    "            if the_file.endswith('.gz'):\n",
    "                reads_FH = gzip.open(the_file, 'rb')\n",
    "            else:\n",
    "                reads_FH = open(the_file, 'r')\n",
    "        \n",
    "            ind_catalog_tags_to_process = {}\n",
    "            files_to_write = []\n",
    "            reads_to_process = {}\n",
    "            \n",
    "            for catalog_tag in catalog_tag_whitelist:\n",
    "                if ind_dict[ind]['tags'].has_key(catalog_tag):\n",
    "                    if not per_tag:\n",
    "                        ind_catalog_tags_to_process[int(ind_dict[ind]['tags'][catalog_tag])] = [os.path.abspath(out_dir)+'/'+ind+'/'+ind+'.'+str(file_id)+'.fastq'] #.append(ind_dict[ind]['tags'][catalog_tag])\n",
    "                        if not os.path.exists(out_dir+\"/\"+ind):\n",
    "                            os.makedirs(out_dir+\"/\"+ind)\n",
    "                        if not ind_catalog_tags_to_process[int(ind_dict[ind]['tags'][catalog_tag])] in files_to_write:\n",
    "                            open(ind_catalog_tags_to_process[int(ind_dict[ind]['tags'][catalog_tag])][0], 'w').close()\n",
    "        \n",
    "            if not ind_catalog_tags_to_process: #if this individual contains any relevant tags\n",
    "                print \"nothing to process for this indidividual\"\n",
    "                continue #go to next individual\n",
    "            \n",
    "            print \"%s individual tags to process\\n\" %len(ind_catalog_tags_to_process)\n",
    "            if not os.path.exists(out_dir+\"/\"+ind):\n",
    "                os.makedirs(out_dir+\"/\"+ind)\n",
    "\n",
    "            for tag in individual_dict[ind]['read_ids'].keys(): #the keys here are the individual tag ids\n",
    "                for read in individual_dict[ind]['read_ids'][tag]: #the list accessed via the individual tag are the read ids\n",
    "                    reads_to_process[read[:-1]+str(file_id)] = int(tag)\n",
    "\n",
    "            complete = len(reads_to_process.keys())\n",
    "            print (\"total of %i reads\" %complete),\n",
    "        \n",
    "        \n",
    "\n",
    "            for record in SeqIO.parse(reads_FH, 'fastq'):\n",
    "#                print record.id\n",
    "\n",
    "                if reads_to_process.has_key(record.id):        \n",
    "                    ind_catalog_tags_to_process[int(reads_to_process[record.id])].append(record)\n",
    "\n",
    "                \n",
    "                    if (len(ind_catalog_tags_to_process[int(reads_to_process[record.id])])-1) == len(individual_dict[ind]['read_ids'][int(reads_to_process[record.id])]):\n",
    "                        OUT = open(ind_catalog_tags_to_process[int(reads_to_process[record.id])][0], 'a')\n",
    "                        SeqIO.write(ind_catalog_tags_to_process[int(reads_to_process[record.id])][1:], OUT, \"fastq\")\n",
    "                        OUT.close()\n",
    "                        del ind_catalog_tags_to_process[int(reads_to_process[record.id])]\n",
    "\n",
    "                    del reads_to_process[record.id]\n",
    "\n",
    "                    \n",
    "                if not len(reads_to_process) > 0:\n",
    "                    break\n",
    "\n",
    "            reads_FH.close()\n",
    "            if len(reads_to_process) == 0:\n",
    "                print \" - done!\\n\"\n",
    "            else:\n",
    "                print \"did not find the expected number of %i reads\\n\" %complete\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Tue Feb 16 14:35:18 2016\n",
      "IDs for individuals provided in list\n",
      "Fetching reads for 11786 tags\n",
      "#################\n",
      "\n",
      "sample_BBNB1\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB1.1.fastq.gz\n",
      "\n",
      "11747 individual tags to process\n",
      "\n",
      "total of 318498 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB1.2.fastq.gz\n",
      "\n",
      "11747 individual tags to process\n",
      "\n",
      "total of 318498 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_BBNB10\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB10.1.fastq.gz\n",
      "\n",
      "11722 individual tags to process\n",
      "\n",
      "total of 243146 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB10.2.fastq.gz\n",
      "\n",
      "11722 individual tags to process\n",
      "\n",
      "total of 243146 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_BBNB2\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB2.1.fastq.gz\n",
      "\n",
      "11286 individual tags to process\n",
      "\n",
      "total of 151412 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB2.2.fastq.gz\n",
      "\n",
      "11286 individual tags to process\n",
      "\n",
      "total of 151412 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_BBNB3\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB3.1.fastq.gz\n",
      "\n",
      "11770 individual tags to process\n",
      "\n",
      "total of 549079 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB3.2.fastq.gz\n",
      "\n",
      "11770 individual tags to process\n",
      "\n",
      "total of 549079 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_BBNB4\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB4.1.fastq.gz\n",
      "\n",
      "11763 individual tags to process\n",
      "\n",
      "total of 384669 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB4.2.fastq.gz\n",
      "\n",
      "11763 individual tags to process\n",
      "\n",
      "total of 384669 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_BBNB5\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB5.1.fastq.gz\n",
      "\n",
      "11764 individual tags to process\n",
      "\n",
      "total of 485516 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB5.2.fastq.gz\n",
      "\n",
      "11764 individual tags to process\n",
      "\n",
      "total of 485516 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_BBNB6\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB6.1.fastq.gz\n",
      "\n",
      "11698 individual tags to process\n",
      "\n",
      "total of 254143 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB6.2.fastq.gz\n",
      "\n",
      "11698 individual tags to process\n",
      "\n",
      "total of 254143 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_BBNB7\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB7.1.fastq.gz\n",
      "\n",
      "11784 individual tags to process\n",
      "\n",
      "total of 539920 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB7.2.fastq.gz\n",
      "\n",
      "11784 individual tags to process\n",
      "\n",
      "total of 539920 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_BBNB8\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB8.1.fastq.gz\n",
      "\n",
      "11768 individual tags to process\n",
      "\n",
      "total of 651332 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB8.2.fastq.gz\n",
      "\n",
      "11768 individual tags to process\n",
      "\n",
      "total of 651332 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_BBNB9\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB9.1.fastq.gz\n",
      "\n",
      "11767 individual tags to process\n",
      "\n",
      "total of 402388 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_BBNB9.2.fastq.gz\n",
      "\n",
      "11767 individual tags to process\n",
      "\n",
      "total of 402388 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_LBPNK2\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_LBPNK2.1.fastq.gz\n",
      "\n",
      "11779 individual tags to process\n",
      "\n",
      "total of 899504 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_LBPNK2.2.fastq.gz\n",
      "\n",
      "11779 individual tags to process\n",
      "\n",
      "total of 899504 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_LBPNK3\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_LBPNK3.1.fastq.gz\n",
      "\n",
      "11388 individual tags to process\n",
      "\n",
      "total of 156024 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_LBPNK3.2.fastq.gz\n",
      "\n",
      "11388 individual tags to process\n",
      "\n",
      "total of 156024 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_LBPNK4\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_LBPNK4.1.fastq.gz\n",
      "\n",
      "10810 individual tags to process\n",
      "\n",
      "total of 115187 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_LBPNK4.2.fastq.gz\n",
      "\n",
      "10810 individual tags to process\n",
      "\n",
      "total of 115187 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_LBPNK5\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_LBPNK5.1.fastq.gz\n",
      "\n",
      "11725 individual tags to process\n",
      "\n",
      "total of 206189 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_LBPNK5.2.fastq.gz\n",
      "\n",
      "11725 individual tags to process\n",
      "\n",
      "total of 206189 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_LBPNK6\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_LBPNK6.1.fastq.gz\n",
      "\n",
      "11781 individual tags to process\n",
      "\n",
      "total of 353254 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_LBPNK6.2.fastq.gz\n",
      "\n",
      "11781 individual tags to process\n",
      "\n",
      "total of 353254 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_LBPNK7\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_LBPNK7.1.fastq.gz\n",
      "\n",
      "11773 individual tags to process\n",
      "\n",
      "total of 382859 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_LBPNK7.2.fastq.gz\n",
      "\n",
      "11773 individual tags to process\n",
      "\n",
      "total of 382859 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_MONB11\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB11.1.fastq.gz\n",
      "\n",
      "11768 individual tags to process\n",
      "\n",
      "total of 237120 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB11.2.fastq.gz\n",
      "\n",
      "11768 individual tags to process\n",
      "\n",
      "total of 237120 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_MONB2\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB2.1.fastq.gz\n",
      "\n",
      "11776 individual tags to process\n",
      "\n",
      "total of 396898 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB2.2.fastq.gz\n",
      "\n",
      "11776 individual tags to process\n",
      "\n",
      "total of 396898 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_MONB3\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB3.1.fastq.gz\n",
      "\n",
      "10884 individual tags to process\n",
      "\n",
      "total of 115733 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB3.2.fastq.gz\n",
      "\n",
      "10884 individual tags to process\n",
      "\n",
      "total of 115733 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_MONB4\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB4.1.fastq.gz\n",
      "\n",
      "11776 individual tags to process\n",
      "\n",
      "total of 367028 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB4.2.fastq.gz\n",
      "\n",
      "11776 individual tags to process\n",
      "\n",
      "total of 367028 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_MONB5\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB5.1.fastq.gz\n",
      "\n",
      "11215 individual tags to process\n",
      "\n",
      "total of 132266 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB5.2.fastq.gz\n",
      "\n",
      "11215 individual tags to process\n",
      "\n",
      "total of 132266 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_MONB7\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB7.1.fastq.gz\n",
      "\n",
      "11393 individual tags to process\n",
      "\n",
      "total of 137251 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB7.2.fastq.gz\n",
      "\n",
      "11393 individual tags to process\n",
      "\n",
      "total of 137251 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_MONB8\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB8.1.fastq.gz\n",
      "\n",
      "11772 individual tags to process\n",
      "\n",
      "total of 283032 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB8.2.fastq.gz\n",
      "\n",
      "11772 individual tags to process\n",
      "\n",
      "total of 283032 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_MONB9\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB9.1.fastq.gz\n",
      "\n",
      "11446 individual tags to process\n",
      "\n",
      "total of 145806 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_MONB9.2.fastq.gz\n",
      "\n",
      "11446 individual tags to process\n",
      "\n",
      "total of 145806 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_NJ1\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ1.1.fastq.gz\n",
      "\n",
      "11753 individual tags to process\n",
      "\n",
      "total of 246015 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ1.2.fastq.gz\n",
      "\n",
      "11753 individual tags to process\n",
      "\n",
      "total of 246015 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_NJ10\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ10.1.fastq.gz\n",
      "\n",
      "11780 individual tags to process\n",
      "\n",
      "total of 679927 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ10.2.fastq.gz\n",
      "\n",
      "11780 individual tags to process\n",
      "\n",
      "total of 679927 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_NJ2\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ2.1.fastq.gz\n",
      "\n",
      "11701 individual tags to process\n",
      "\n",
      "total of 206264 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ2.2.fastq.gz\n",
      "\n",
      "11701 individual tags to process\n",
      "\n",
      "total of 206264 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_NJ3\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ3.1.fastq.gz\n",
      "\n",
      "11455 individual tags to process\n",
      "\n",
      "total of 145822 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ3.2.fastq.gz\n",
      "\n",
      "11455 individual tags to process\n",
      "\n",
      "total of 145822 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_NJ4\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ4.1.fastq.gz\n",
      "\n",
      "11431 individual tags to process\n",
      "\n",
      "total of 150180 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ4.2.fastq.gz\n",
      "\n",
      "11431 individual tags to process\n",
      "\n",
      "total of 150180 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_NJ5\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ5.1.fastq.gz\n",
      "\n",
      "11726 individual tags to process\n",
      "\n",
      "total of 211635 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ5.2.fastq.gz\n",
      "\n",
      "11726 individual tags to process\n",
      "\n",
      "total of 211635 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_NJ6\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ6.1.fastq.gz\n",
      "\n",
      "11269 individual tags to process\n",
      "\n",
      "total of 137536 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ6.2.fastq.gz\n",
      "\n",
      "11269 individual tags to process\n",
      "\n",
      "total of 137536 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_NJ7\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ7.1.fastq.gz\n",
      "\n",
      "11775 individual tags to process\n",
      "\n",
      "total of 551932 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ7.2.fastq.gz\n",
      "\n",
      "11775 individual tags to process\n",
      "\n",
      "total of 551932 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_NJ8\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ8.1.fastq.gz\n",
      "\n",
      "11521 individual tags to process\n",
      "\n",
      "total of 158219 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ8.2.fastq.gz\n",
      "\n",
      "11521 individual tags to process\n",
      "\n",
      "total of 158219 reads  - done!\n",
      "\n",
      "#################\n",
      "\n",
      "sample_NJ9\n",
      "processing file 1: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ9.1.fastq.gz\n",
      "\n",
      "11569 individual tags to process\n",
      "\n",
      "total of 158855 reads  - done!\n",
      "\n",
      "processing file 2: /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/raw_reads/sample_NJ9.2.fastq.gz\n",
      "\n",
      "11569 individual tags to process\n",
      "\n",
      "total of 158855 reads  - done!\n",
      "\n",
      "Tue Feb 16 16:04:17 2016\n"
     ]
    }
   ],
   "source": [
    "import time\n",
    "print time.asctime( time.localtime(time.time()) )\n",
    "extract_reads(ind_list=ind_dict.keys(), individual_dict=ind_dict, catalog=catalog_dict, out_dir=\"./reads/\", pe=True)\n",
    "print time.asctime( time.localtime(time.time()) )"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "run BWA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "!mkdir ./mapping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating index\n",
      "\n",
      "‘/media/chrishah/STORAGE/DATA/Cichlids/reference_data/M_zebra/MetZeb1.1_prescreen/M_zebra_v0.assembly.fasta’ -> ‘M_zebra_v1.1.fasta’\n",
      "Tue Feb 16 20:30:50 GMT 2016\n",
      "\n",
      "processing sample sample_BBNB1\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_BBNB1.bam’\n",
      "Tue Feb 16 20:31:25 GMT 2016\n",
      "\n",
      "processing sample sample_BBNB10\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_BBNB10.bam’\n",
      "Tue Feb 16 20:31:49 GMT 2016\n",
      "\n",
      "processing sample sample_BBNB2\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_BBNB2.bam’\n",
      "Tue Feb 16 20:32:05 GMT 2016\n",
      "\n",
      "processing sample sample_BBNB3\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_BBNB3.bam’\n",
      "Tue Feb 16 20:33:01 GMT 2016\n",
      "\n",
      "processing sample sample_BBNB4\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_BBNB4.bam’\n",
      "Tue Feb 16 20:33:38 GMT 2016\n",
      "\n",
      "processing sample sample_BBNB5\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_BBNB5.bam’\n",
      "Tue Feb 16 20:34:26 GMT 2016\n",
      "\n",
      "processing sample sample_BBNB6\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_BBNB6.bam’\n",
      "Tue Feb 16 20:34:53 GMT 2016\n",
      "\n",
      "processing sample sample_BBNB7\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_BBNB7.bam’\n",
      "Tue Feb 16 20:35:48 GMT 2016\n",
      "\n",
      "processing sample sample_BBNB8\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_BBNB8.bam’\n",
      "Tue Feb 16 20:36:59 GMT 2016\n",
      "\n",
      "processing sample sample_BBNB9\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_BBNB9.bam’\n",
      "Tue Feb 16 20:37:42 GMT 2016\n",
      "\n",
      "processing sample sample_LBPNK2\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_LBPNK2.bam’\n",
      "Tue Feb 16 20:39:15 GMT 2016\n",
      "\n",
      "processing sample sample_LBPNK3\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_LBPNK3.bam’\n",
      "Tue Feb 16 20:39:34 GMT 2016\n",
      "\n",
      "processing sample sample_LBPNK4\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_LBPNK4.bam’\n",
      "Tue Feb 16 20:39:48 GMT 2016\n",
      "\n",
      "processing sample sample_LBPNK5\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_LBPNK5.bam’\n",
      "Tue Feb 16 20:40:11 GMT 2016\n",
      "\n",
      "processing sample sample_LBPNK6\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_LBPNK6.bam’\n",
      "Tue Feb 16 20:40:58 GMT 2016\n",
      "\n",
      "processing sample sample_LBPNK7\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_LBPNK7.bam’\n",
      "Tue Feb 16 20:41:49 GMT 2016\n",
      "\n",
      "processing sample sample_MONB11\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_MONB11.bam’\n",
      "Tue Feb 16 20:42:16 GMT 2016\n",
      "\n",
      "processing sample sample_MONB2\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_MONB2.bam’\n",
      "Tue Feb 16 20:42:59 GMT 2016\n",
      "\n",
      "processing sample sample_MONB3\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_MONB3.bam’\n",
      "Tue Feb 16 20:43:13 GMT 2016\n",
      "\n",
      "processing sample sample_MONB4\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_MONB4.bam’\n",
      "Tue Feb 16 20:43:53 GMT 2016\n",
      "\n",
      "processing sample sample_MONB5\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_MONB5.bam’\n",
      "Tue Feb 16 20:44:08 GMT 2016\n",
      "\n",
      "processing sample sample_MONB7\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_MONB7.bam’\n",
      "Tue Feb 16 20:44:23 GMT 2016\n",
      "\n",
      "processing sample sample_MONB8\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_MONB8.bam’\n",
      "Tue Feb 16 20:44:54 GMT 2016\n",
      "\n",
      "processing sample sample_MONB9\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_MONB9.bam’\n",
      "Tue Feb 16 20:45:10 GMT 2016\n",
      "\n",
      "processing sample sample_NJ1\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_NJ1.bam’\n",
      "Tue Feb 16 20:45:37 GMT 2016\n",
      "\n",
      "processing sample sample_NJ10\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_NJ10.bam’\n",
      "Tue Feb 16 20:47:02 GMT 2016\n",
      "\n",
      "processing sample sample_NJ2\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_NJ2.bam’\n",
      "Tue Feb 16 20:47:26 GMT 2016\n",
      "\n",
      "processing sample sample_NJ3\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_NJ3.bam’\n",
      "Tue Feb 16 20:47:42 GMT 2016\n",
      "\n",
      "processing sample sample_NJ4\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_NJ4.bam’\n",
      "Tue Feb 16 20:48:00 GMT 2016\n",
      "\n",
      "processing sample sample_NJ5\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_NJ5.bam’\n",
      "Tue Feb 16 20:48:24 GMT 2016\n",
      "\n",
      "processing sample sample_NJ6\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_NJ6.bam’\n",
      "Tue Feb 16 20:48:41 GMT 2016\n",
      "\n",
      "processing sample sample_NJ7\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_NJ7.bam’\n",
      "Tue Feb 16 20:49:42 GMT 2016\n",
      "\n",
      "processing sample sample_NJ8\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_NJ8.bam’\n",
      "Tue Feb 16 20:50:00 GMT 2016\n",
      "\n",
      "processing sample sample_NJ9\n",
      "\n",
      "removed ‘forward.sai’\n",
      "removed ‘sample_NJ9.bam’\n",
      "Tue Feb 16 20:50:19 GMT 2016\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[bwa_index] Pack FASTA... 8.41 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[bwa_index] 551.97 seconds elapse.\n",
      "[bwa_index] Update BWT... 5.03 sec\n",
      "[bwa_index] Pack forward-only FASTA... 5.71 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 205.17 sec\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa index -p M_zebra_v1.1 -a is M_zebra_v1.1.fasta\n",
      "[main] Real time: 794.800 sec; CPU: 776.298 sec\n",
      "[bwa_aln_core] calculate SA coordinate... 40.03 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] calculate SA coordinate... 10.84 sec\n",
      "[bwa_aln_core] write to the disk... 0.01 sec\n",
      "[bwa_aln_core] 318498 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_BBNB1/sample_BBNB1.1.fastq\n",
      "[main] Real time: 26.662 sec; CPU: 51.564 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.93 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.58 sec\n",
      "[bwa_aln_core] print alignments... 0.75 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.44 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.21 sec\n",
      "[bwa_aln_core] print alignments... 0.16 sec\n",
      "[bwa_aln_core] 318498 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_BBNB1\\tSM:sample_BBNB1 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_BBNB1/sample_BBNB1.1.fastq\n",
      "[main] Real time: 4.298 sec; CPU: 3.589 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 34.72 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 243146 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_BBNB10/sample_BBNB10.1.fastq\n",
      "[main] Real time: 18.160 sec; CPU: 35.273 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.83 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.40 sec\n",
      "[bwa_aln_core] print alignments... 0.44 sec\n",
      "[bwa_aln_core] 243146 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_BBNB10\\tSM:sample_BBNB10 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_BBNB10/sample_BBNB10.1.fastq\n",
      "[main] Real time: 2.588 sec; CPU: 2.057 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 21.57 sec\n",
      "[bwa_aln_core] write to the disk... 0.02 sec\n",
      "[bwa_aln_core] 151412 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_BBNB2/sample_BBNB2.1.fastq\n",
      "[main] Real time: 11.357 sec; CPU: 21.961 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.60 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.29 sec\n",
      "[bwa_aln_core] print alignments... 0.27 sec\n",
      "[bwa_aln_core] 151412 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_BBNB2\\tSM:sample_BBNB2 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_BBNB2/sample_BBNB2.1.fastq\n",
      "[main] Real time: 1.745 sec; CPU: 1.352 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 30.13 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] calculate SA coordinate... 43.58 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 524288 sequences have been processed.\n",
      "[bwa_aln_core] calculate SA coordinate... 5.08 sec\n",
      "[bwa_aln_core] write to the disk... 0.00 sec\n",
      "[bwa_aln_core] 549079 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_BBNB3/sample_BBNB3.1.fastq\n",
      "[main] Real time: 41.296 sec; CPU: 79.768 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.87 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.48 sec\n",
      "[bwa_aln_core] print alignments... 0.74 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.90 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.72 sec\n",
      "[bwa_aln_core] print alignments... 0.74 sec\n",
      "[bwa_aln_core] 524288 sequences have been processed.\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.37 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.15 sec\n",
      "[bwa_aln_core] print alignments... 0.07 sec\n",
      "[bwa_aln_core] 549079 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_BBNB3\\tSM:sample_BBNB3 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_BBNB3/sample_BBNB3.1.fastq\n",
      "[main] Real time: 7.265 sec; CPU: 5.888 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 32.87 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] calculate SA coordinate... 20.79 sec\n",
      "[bwa_aln_core] write to the disk... 0.01 sec\n",
      "[bwa_aln_core] 384669 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_BBNB4/sample_BBNB4.1.fastq\n",
      "[main] Real time: 28.325 sec; CPU: 54.347 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.83 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.47 sec\n",
      "[bwa_aln_core] print alignments... 0.75 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.51 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.28 sec\n",
      "[bwa_aln_core] print alignments... 0.26 sec\n",
      "[bwa_aln_core] 384669 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_BBNB4\\tSM:sample_BBNB4 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_BBNB4/sample_BBNB4.1.fastq\n",
      "[main] Real time: 4.568 sec; CPU: 3.587 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 35.37 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] calculate SA coordinate... 32.91 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 485516 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_BBNB5/sample_BBNB5.1.fastq\n",
      "[main] Real time: 35.712 sec; CPU: 69.208 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.91 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.53 sec\n",
      "[bwa_aln_core] print alignments... 0.74 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.82 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.58 sec\n",
      "[bwa_aln_core] print alignments... 0.43 sec\n",
      "[bwa_aln_core] 485516 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_BBNB5\\tSM:sample_BBNB5 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_BBNB5/sample_BBNB5.1.fastq\n",
      "[main] Real time: 5.979 sec; CPU: 4.751 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 37.14 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 254143 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_BBNB6/sample_BBNB6.1.fastq\n",
      "[main] Real time: 19.831 sec; CPU: 37.730 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.88 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.61 sec\n",
      "[bwa_aln_core] print alignments... 0.74 sec\n",
      "[bwa_aln_core] 254143 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_BBNB6\\tSM:sample_BBNB6 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_BBNB6/sample_BBNB6.1.fastq\n",
      "[main] Real time: 3.288 sec; CPU: 2.662 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 35.87 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] calculate SA coordinate... 39.83 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 524288 sequences have been processed.\n",
      "[bwa_aln_core] calculate SA coordinate... 3.11 sec\n",
      "[bwa_aln_core] write to the disk... 0.00 sec\n",
      "[bwa_aln_core] 539920 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_BBNB7/sample_BBNB7.1.fastq\n",
      "[main] Real time: 41.400 sec; CPU: 79.710 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.77 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.41 sec\n",
      "[bwa_aln_core] print alignments... 0.61 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.91 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.60 sec\n",
      "[bwa_aln_core] print alignments... 0.73 sec\n",
      "[bwa_aln_core] 524288 sequences have been processed.\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.33 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.12 sec\n",
      "[bwa_aln_core] print alignments... 0.05 sec\n",
      "[bwa_aln_core] 539920 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_BBNB7\\tSM:sample_BBNB7 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_BBNB7/sample_BBNB7.1.fastq\n",
      "[main] Real time: 6.508 sec; CPU: 5.334 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 38.62 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] calculate SA coordinate... 38.12 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 524288 sequences have been processed.\n",
      "[bwa_aln_core] calculate SA coordinate... 24.69 sec\n",
      "[bwa_aln_core] write to the disk... 0.02 sec\n",
      "[bwa_aln_core] 651332 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_BBNB8/sample_BBNB8.1.fastq\n",
      "[main] Real time: 53.351 sec; CPU: 102.603 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.90 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.53 sec\n",
      "[bwa_aln_core] print alignments... 0.73 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.88 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.65 sec\n",
      "[bwa_aln_core] print alignments... 0.62 sec\n",
      "[bwa_aln_core] 524288 sequences have been processed.\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.59 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.41 sec\n",
      "[bwa_aln_core] print alignments... 0.36 sec\n",
      "[bwa_aln_core] 651332 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_BBNB8\\tSM:sample_BBNB8 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_BBNB8/sample_BBNB8.1.fastq\n",
      "[main] Real time: 8.086 sec; CPU: 6.607 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 33.91 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] calculate SA coordinate... 30.62 sec\n",
      "[bwa_aln_core] write to the disk... 0.02 sec\n",
      "[bwa_aln_core] 402388 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_BBNB9/sample_BBNB9.1.fastq\n",
      "[main] Real time: 33.756 sec; CPU: 65.164 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.82 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.54 sec\n",
      "[bwa_aln_core] print alignments... 0.65 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.58 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.31 sec\n",
      "[bwa_aln_core] print alignments... 0.25 sec\n",
      "[bwa_aln_core] 402388 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_BBNB9\\tSM:sample_BBNB9 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_BBNB9/sample_BBNB9.1.fastq\n",
      "[main] Real time: 4.573 sec; CPU: 3.677 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 33.70 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] calculate SA coordinate... 37.70 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 524288 sequences have been processed.\n",
      "[bwa_aln_core] calculate SA coordinate... 37.07 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 786432 sequences have been processed.\n",
      "[bwa_aln_core] calculate SA coordinate... 24.66 sec\n",
      "[bwa_aln_core] write to the disk... 0.01 sec\n",
      "[bwa_aln_core] 899504 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_LBPNK2/sample_LBPNK2.1.fastq\n",
      "[main] Real time: 69.877 sec; CPU: 134.609 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.82 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.49 sec\n",
      "[bwa_aln_core] print alignments... 0.74 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.88 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.60 sec\n",
      "[bwa_aln_core] print alignments... 0.74 sec\n",
      "[bwa_aln_core] 524288 sequences have been processed.\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.91 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.73 sec\n",
      "[bwa_aln_core] print alignments... 0.74 sec\n",
      "[bwa_aln_core] 786432 sequences have been processed.\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.58 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.40 sec\n",
      "[bwa_aln_core] print alignments... 0.27 sec\n",
      "[bwa_aln_core] 899504 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_LBPNK2\\tSM:sample_LBPNK2 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_LBPNK2/sample_LBPNK2.1.fastq\n",
      "[main] Real time: 11.270 sec; CPU: 9.249 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 27.95 sec\n",
      "[bwa_aln_core] write to the disk... 0.02 sec\n",
      "[bwa_aln_core] 156024 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_LBPNK3/sample_LBPNK3.1.fastq\n",
      "[main] Real time: 14.717 sec; CPU: 28.337 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.60 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.35 sec\n",
      "[bwa_aln_core] print alignments... 0.33 sec\n",
      "[bwa_aln_core] 156024 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_LBPNK3\\tSM:sample_LBPNK3 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_LBPNK3/sample_LBPNK3.1.fastq\n",
      "[main] Real time: 1.941 sec; CPU: 1.555 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 19.77 sec\n",
      "[bwa_aln_core] write to the disk... 0.01 sec\n",
      "[bwa_aln_core] 115187 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_LBPNK4/sample_LBPNK4.1.fastq\n",
      "[main] Real time: 10.572 sec; CPU: 20.158 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.53 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.33 sec\n",
      "[bwa_aln_core] print alignments... 0.35 sec\n",
      "[bwa_aln_core] 115187 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_LBPNK4\\tSM:sample_LBPNK4 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_LBPNK4/sample_LBPNK4.1.fastq\n",
      "[main] Real time: 1.701 sec; CPU: 1.412 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 33.41 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 206189 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_LBPNK5/sample_LBPNK5.1.fastq\n",
      "[main] Real time: 17.398 sec; CPU: 33.939 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.69 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.36 sec\n",
      "[bwa_aln_core] print alignments... 0.58 sec\n",
      "[bwa_aln_core] 206189 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_LBPNK5\\tSM:sample_LBPNK5 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_LBPNK5/sample_LBPNK5.1.fastq\n",
      "[main] Real time: 2.357 sec; CPU: 1.891 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 50.03 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] calculate SA coordinate... 22.67 sec\n",
      "[bwa_aln_core] write to the disk... 0.01 sec\n",
      "[bwa_aln_core] 353254 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_LBPNK6/sample_LBPNK6.1.fastq\n",
      "[main] Real time: 37.801 sec; CPU: 73.428 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.92 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.68 sec\n",
      "[bwa_aln_core] print alignments... 0.74 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.43 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.33 sec\n",
      "[bwa_aln_core] print alignments... 0.27 sec\n",
      "[bwa_aln_core] 353254 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_LBPNK6\\tSM:sample_LBPNK6 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_LBPNK6/sample_LBPNK6.1.fastq\n",
      "[main] Real time: 4.788 sec; CPU: 3.891 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 47.96 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] calculate SA coordinate... 29.08 sec\n",
      "[bwa_aln_core] write to the disk... 0.01 sec\n",
      "[bwa_aln_core] 382859 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_LBPNK7/sample_LBPNK7.1.fastq\n",
      "[main] Real time: 40.205 sec; CPU: 77.794 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.80 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.56 sec\n",
      "[bwa_aln_core] print alignments... 0.61 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.58 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.40 sec\n",
      "[bwa_aln_core] print alignments... 0.34 sec\n",
      "[bwa_aln_core] 382859 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_LBPNK7\\tSM:sample_LBPNK7 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_LBPNK7/sample_LBPNK7.1.fastq\n",
      "[main] Real time: 4.819 sec; CPU: 3.889 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 40.44 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 237120 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_MONB11/sample_MONB11.1.fastq\n",
      "[main] Real time: 21.356 sec; CPU: 41.011 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.73 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.41 sec\n",
      "[bwa_aln_core] print alignments... 0.41 sec\n",
      "[bwa_aln_core] 237120 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_MONB11\\tSM:sample_MONB11 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_MONB11/sample_MONB11.1.fastq\n",
      "[main] Real time: 2.447 sec; CPU: 1.892 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 37.16 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] calculate SA coordinate... 23.55 sec\n",
      "[bwa_aln_core] write to the disk... 0.01 sec\n",
      "[bwa_aln_core] 396898 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_MONB2/sample_MONB2.1.fastq\n",
      "[main] Real time: 32.586 sec; CPU: 61.497 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.81 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.46 sec\n",
      "[bwa_aln_core] print alignments... 0.47 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.53 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.40 sec\n",
      "[bwa_aln_core] print alignments... 0.40 sec\n",
      "[bwa_aln_core] 396898 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_MONB2\\tSM:sample_MONB2 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_MONB2/sample_MONB2.1.fastq\n",
      "[main] Real time: 4.587 sec; CPU: 3.636 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 20.40 sec\n",
      "[bwa_aln_core] write to the disk... 0.01 sec\n",
      "[bwa_aln_core] 115733 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_MONB3/sample_MONB3.1.fastq\n",
      "[main] Real time: 10.946 sec; CPU: 20.785 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.55 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.28 sec\n",
      "[bwa_aln_core] print alignments... 0.22 sec\n",
      "[bwa_aln_core] 115733 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_MONB3\\tSM:sample_MONB3 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_MONB3/sample_MONB3.1.fastq\n",
      "[main] Real time: 1.527 sec; CPU: 1.245 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 33.05 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] calculate SA coordinate... 23.55 sec\n",
      "[bwa_aln_core] write to the disk... 0.01 sec\n",
      "[bwa_aln_core] 367028 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_MONB4/sample_MONB4.1.fastq\n",
      "[main] Real time: 30.009 sec; CPU: 57.295 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.91 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.56 sec\n",
      "[bwa_aln_core] print alignments... 0.74 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.56 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.37 sec\n",
      "[bwa_aln_core] print alignments... 0.30 sec\n",
      "[bwa_aln_core] 367028 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_MONB4\\tSM:sample_MONB4 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_MONB4/sample_MONB4.1.fastq\n",
      "[main] Real time: 4.957 sec; CPU: 4.021 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 21.50 sec\n",
      "[bwa_aln_core] write to the disk... 0.02 sec\n",
      "[bwa_aln_core] 132266 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_MONB5/sample_MONB5.1.fastq\n",
      "[main] Real time: 11.301 sec; CPU: 21.834 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.62 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.29 sec\n",
      "[bwa_aln_core] print alignments... 0.29 sec\n",
      "[bwa_aln_core] 132266 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_MONB5\\tSM:sample_MONB5 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_MONB5/sample_MONB5.1.fastq\n",
      "[main] Real time: 1.701 sec; CPU: 1.380 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 21.82 sec\n",
      "[bwa_aln_core] write to the disk... 0.02 sec\n",
      "[bwa_aln_core] 137251 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_MONB7/sample_MONB7.1.fastq\n",
      "[main] Real time: 11.618 sec; CPU: 22.187 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.64 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.37 sec\n",
      "[bwa_aln_core] print alignments... 0.39 sec\n",
      "[bwa_aln_core] 137251 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_MONB7\\tSM:sample_MONB7 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_MONB7/sample_MONB7.1.fastq\n",
      "[main] Real time: 1.963 sec; CPU: 1.620 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 40.64 sec\n",
      "[bwa_aln_core] write to the disk... 0.02 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] calculate SA coordinate... 3.62 sec\n",
      "[bwa_aln_core] write to the disk... 0.00 sec\n",
      "[bwa_aln_core] 283032 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_MONB8/sample_MONB8.1.fastq\n",
      "[main] Real time: 23.275 sec; CPU: 44.863 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.76 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.44 sec\n",
      "[bwa_aln_core] print alignments... 0.60 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.33 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.12 sec\n",
      "[bwa_aln_core] print alignments... 0.06 sec\n",
      "[bwa_aln_core] 283032 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_MONB8\\tSM:sample_MONB8 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_MONB8/sample_MONB8.1.fastq\n",
      "[main] Real time: 3.350 sec; CPU: 2.654 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 23.76 sec\n",
      "[bwa_aln_core] write to the disk... 0.02 sec\n",
      "[bwa_aln_core] 145806 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_MONB9/sample_MONB9.1.fastq\n",
      "[main] Real time: 12.527 sec; CPU: 24.135 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.67 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.41 sec\n",
      "[bwa_aln_core] print alignments... 0.42 sec\n",
      "[bwa_aln_core] 145806 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_MONB9\\tSM:sample_MONB9 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_MONB9/sample_MONB9.1.fastq\n",
      "[main] Real time: 2.143 sec; CPU: 1.754 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 38.52 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 246015 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_NJ1/sample_NJ1.1.fastq\n",
      "[main] Real time: 20.598 sec; CPU: 39.098 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.74 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.52 sec\n",
      "[bwa_aln_core] print alignments... 0.65 sec\n",
      "[bwa_aln_core] 246015 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_NJ1\\tSM:sample_NJ1 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_NJ1/sample_NJ1.1.fastq\n",
      "[main] Real time: 2.930 sec; CPU: 2.333 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 40.38 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] calculate SA coordinate... 55.59 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 524288 sequences have been processed.\n",
      "[bwa_aln_core] calculate SA coordinate... 34.58 sec\n",
      "[bwa_aln_core] write to the disk... 0.02 sec\n",
      "[bwa_aln_core] 679927 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_NJ10/sample_NJ10.1.fastq\n",
      "[main] Real time: 67.078 sec; CPU: 131.759 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.91 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.59 sec\n",
      "[bwa_aln_core] print alignments... 0.73 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.90 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.70 sec\n",
      "[bwa_aln_core] print alignments... 0.74 sec\n",
      "[bwa_aln_core] 524288 sequences have been processed.\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.66 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.52 sec\n",
      "[bwa_aln_core] print alignments... 0.44 sec\n",
      "[bwa_aln_core] 679927 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_NJ10\\tSM:sample_NJ10 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_NJ10/sample_NJ10.1.fastq\n",
      "[main] Real time: 8.696 sec; CPU: 7.220 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 34.49 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 206264 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_NJ2/sample_NJ2.1.fastq\n",
      "[main] Real time: 18.213 sec; CPU: 35.030 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.72 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.37 sec\n",
      "[bwa_aln_core] print alignments... 0.47 sec\n",
      "[bwa_aln_core] 206264 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_NJ2\\tSM:sample_NJ2 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_NJ2/sample_NJ2.1.fastq\n",
      "[main] Real time: 2.406 sec; CPU: 1.904 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 23.41 sec\n",
      "[bwa_aln_core] write to the disk... 0.02 sec\n",
      "[bwa_aln_core] 145822 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_NJ3/sample_NJ3.1.fastq\n",
      "[main] Real time: 12.379 sec; CPU: 23.834 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.59 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.28 sec\n",
      "[bwa_aln_core] print alignments... 0.37 sec\n",
      "[bwa_aln_core] 145822 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_NJ3\\tSM:sample_NJ3 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_NJ3/sample_NJ3.1.fastq\n",
      "[main] Real time: 1.884 sec; CPU: 1.501 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 25.96 sec\n",
      "[bwa_aln_core] write to the disk... 0.01 sec\n",
      "[bwa_aln_core] 150180 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_NJ4/sample_NJ4.1.fastq\n",
      "[main] Real time: 13.774 sec; CPU: 26.397 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.67 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.40 sec\n",
      "[bwa_aln_core] print alignments... 0.43 sec\n",
      "[bwa_aln_core] 150180 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_NJ4\\tSM:sample_NJ4 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_NJ4/sample_NJ4.1.fastq\n",
      "[main] Real time: 2.143 sec; CPU: 1.776 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 32.57 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 211635 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_NJ5/sample_NJ5.1.fastq\n",
      "[main] Real time: 18.222 sec; CPU: 33.103 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.81 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.53 sec\n",
      "[bwa_aln_core] print alignments... 0.59 sec\n",
      "[bwa_aln_core] 211635 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_NJ5\\tSM:sample_NJ5 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_NJ5/sample_NJ5.1.fastq\n",
      "[main] Real time: 2.773 sec; CPU: 2.215 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 23.92 sec\n",
      "[bwa_aln_core] write to the disk... 0.02 sec\n",
      "[bwa_aln_core] 137536 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_NJ6/sample_NJ6.1.fastq\n",
      "[main] Real time: 12.647 sec; CPU: 24.321 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.59 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.37 sec\n",
      "[bwa_aln_core] print alignments... 0.34 sec\n",
      "[bwa_aln_core] 137536 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_NJ6\\tSM:sample_NJ6 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_NJ6/sample_NJ6.1.fastq\n",
      "[main] Real time: 1.818 sec; CPU: 1.509 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 39.99 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] calculate SA coordinate... 43.67 sec\n",
      "[bwa_aln_core] write to the disk... 0.03 sec\n",
      "[bwa_aln_core] 524288 sequences have been processed.\n",
      "[bwa_aln_core] calculate SA coordinate... 6.06 sec\n",
      "[bwa_aln_core] write to the disk... 0.00 sec\n",
      "[bwa_aln_core] 551932 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_NJ7/sample_NJ7.1.fastq\n",
      "[main] Real time: 47.150 sec; CPU: 90.625 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.77 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.48 sec\n",
      "[bwa_aln_core] print alignments... 0.74 sec\n",
      "[bwa_aln_core] 262144 sequences have been processed.\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.81 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.71 sec\n",
      "[bwa_aln_core] print alignments... 0.68 sec\n",
      "[bwa_aln_core] 524288 sequences have been processed.\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.32 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.12 sec\n",
      "[bwa_aln_core] print alignments... 0.08 sec\n",
      "[bwa_aln_core] 551932 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_NJ7\\tSM:sample_NJ7 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_NJ7/sample_NJ7.1.fastq\n",
      "[main] Real time: 6.708 sec; CPU: 5.420 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 25.78 sec\n",
      "[bwa_aln_core] write to the disk... 0.02 sec\n",
      "[bwa_aln_core] 158219 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_NJ8/sample_NJ8.1.fastq\n",
      "[main] Real time: 13.714 sec; CPU: 26.158 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.61 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.31 sec\n",
      "[bwa_aln_core] print alignments... 0.33 sec\n",
      "[bwa_aln_core] 158219 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_NJ8\\tSM:sample_NJ8 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_NJ8/sample_NJ8.1.fastq\n",
      "[main] Real time: 1.934 sec; CPU: 1.520 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n",
      "[bwa_aln_core] calculate SA coordinate... 26.75 sec\n",
      "[bwa_aln_core] write to the disk... 0.01 sec\n",
      "[bwa_aln_core] 158855 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa aln -n 8 -t 2 ./index/M_zebra_v1.1 /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_NJ9/sample_NJ9.1.fastq\n",
      "[main] Real time: 14.248 sec; CPU: 27.137 sec\n",
      "[bwa_aln_core] convert to sequence coordinate... 0.67 sec\n",
      "[bwa_aln_core] refine gapped alignments... 0.42 sec\n",
      "[bwa_aln_core] print alignments... 0.46 sec\n",
      "[bwa_aln_core] 158855 sequences have been processed.\n",
      "[main] Version: 0.7.10-r789\n",
      "[main] CMD: bwa samse -r @RG\\tID:sample_NJ9\\tSM:sample_NJ9 ./index/M_zebra_v1.1 forward.sai /media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads/sample_NJ9/sample_NJ9.1.fastq\n",
      "[main] Real time: 2.264 sec; CPU: 1.822 sec\n",
      "rm: cannot remove ‘reverse.sai’: No such file or directory\n",
      "[samopen] SAM header is present: 3750 sequences.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "working_dir=/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/mapping\n",
    "data_dir=/media/chrishah/STORAGE/RAD/popgen/Freebayes/Diplotaxodon_4pop_for_paper/test/reads\n",
    "reference=/media/chrishah/STORAGE/DATA/Cichlids/reference_data/M_zebra/MetZeb1.1_prescreen/M_zebra_v0.assembly.fasta\n",
    "#prefix=M_zebra_v1.1\n",
    "index=M_zebra_v1.1\n",
    "MM=8\n",
    "threads=2\n",
    "\n",
    "## Do some work:\n",
    "cd $working_dir\n",
    "\n",
    "#preparing\n",
    "mkdir index/; mkdir sam/; mkdir bam/\n",
    "\n",
    "##builing indexfile\n",
    "echo -e \"Creating index\\n\"\n",
    "cd index\n",
    "cp -v $reference $index.fasta\n",
    "bwa index -p $index -a is $index.fasta\n",
    "cd ..\n",
    "        \n",
    "date\n",
    "\n",
    "for sample in $(ls -1 $data_dir)\n",
    "do \n",
    "        echo -e \"\\nprocessing sample $sample\\n\"\n",
    "        \n",
    "        bwa aln -n $MM -t $threads ./index/$index $data_dir/$sample/$sample.1.fastq > forward.sai\n",
    "#        bwa aln -t $threads ./index/$index $data_dir/$sample/$sample.2.fastq > reverse.sai\n",
    "        RG=\"@RG\\tID:$sample\\tSM:$sample\"\n",
    "        bwa samse -r $RG ./index/$index forward.sai $data_dir/$sample/$sample.1.fastq > sam/$sample.sam\n",
    "#        bwa sampe -r $RG $index_pre forward.sai reverse.sai data/$prefix\\_1.fastq data/$prefix\\_2.fastq > sam/$prefix.sam\n",
    "        rm -v forward.sai reverse.sai \n",
    "\n",
    "        cd bam/\n",
    "        samtools view -@ $threads -bS -o $sample.bam ../sam/$sample.sam\n",
    "        samtools sort -@ $threads $sample.bam $sample.sorted\n",
    "        samtools index $sample.sorted.bam\n",
    "#        samtools idxstats $sample.sorted.bam\n",
    "        rm -v $sample.bam\n",
    "        cd ../\n",
    "#        gzip -v sam/*.sam\n",
    "#        rm -rfv ./sam\n",
    "        date\n",
    "done\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Do SNPcalling"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "!mkdir ./SNP-calling/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "cd SNP-calling/\n",
    "for sample in $(cut -f 1 ../../popmap); do ls -1 ../mapping/bam/$sample.* |grep \"bam$\"; done > bamlist.txt\n",
    "\n",
    "freebayes -L bamlist.txt -f ../mapping/index/M_zebra_v1.1.fasta -v full.vcf --report-monomorphic --populations ../../popmap --no-complex\n",
    "cd .."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "extract fasta inclusive reference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ignoring indels: scaffold_0 937795\n",
      "ignoring indels: scaffold_0 959661\n",
      "ignoring > bi-allelic SNPs: scaffold_0 973178\n",
      "ignoring > bi-allelic SNPs: scaffold_0 1985535\n",
      "ignoring indels: scaffold_0 2290805\n",
      "ignoring > bi-allelic SNPs: scaffold_0 2339302\n",
      "ignoring indels: scaffold_0 2558792\n",
      "ignoring indels: scaffold_0 3566065\n",
      "ignoring indels: scaffold_0 4883311\n",
      "ignoring > bi-allelic SNPs: scaffold_0 5562771\n",
      "ignoring indels: scaffold_0 5570042\n",
      "ignoring indels: scaffold_0 6248672\n",
      "ignoring indels: scaffold_0 6283735\n",
      "ignoring indels: scaffold_0 6689464\n",
      "ignoring indels: scaffold_0 6723865\n",
      "ignoring > bi-allelic SNPs: scaffold_0 6844277\n",
      "ignoring indels: scaffold_0 6844319\n",
      "ignoring indels: scaffold_0 7079686\n",
      "ignoring indels: scaffold_0 7591476\n",
      "ignoring indels: scaffold_0 8203013\n",
      "ignoring indels: scaffold_0 8281824\n",
      "ignoring indels: scaffold_0 9069215\n",
      "ignoring indels: scaffold_0 9280032\n",
      "ignoring indels: scaffold_0 10060904\n",
      "ignoring indels: scaffold_0 11143410\n",
      "ignoring > bi-allelic SNPs: scaffold_0 12032826\n",
      "ignoring indels: scaffold_0 14715096\n",
      "ignoring indels: scaffold_0 15585053\n",
      "ignoring indels: scaffold_0 15585081\n",
      "ignoring indels: scaffold_0 15818309\n",
      "ignoring > bi-allelic SNPs: scaffold_0 15902794\n",
      "ignoring indels: scaffold_0 16312081\n",
      "ignoring indels: scaffold_0 17463915\n",
      "ignoring > bi-allelic SNPs: scaffold_1 1277129\n",
      "ignoring indels: scaffold_1 1292331\n",
      "ignoring > bi-allelic SNPs: scaffold_1 2232197\n",
      "ignoring indels: scaffold_1 2250683\n",
      "ignoring indels: scaffold_1 2412990\n",
      "ignoring indels: scaffold_1 2576158\n",
      "ignoring indels: scaffold_1 2623107\n",
      "ignoring indels: scaffold_1 2630513\n",
      "ignoring indels: scaffold_1 2927983\n",
      "ignoring > bi-allelic SNPs: scaffold_1 3002086\n",
      "ignoring > bi-allelic SNPs: scaffold_1 3058520\n",
      "ignoring indels: scaffold_1 3058527\n",
      "ignoring indels: scaffold_1 3058559\n",
      "ignoring > bi-allelic SNPs: scaffold_1 3547460\n",
      "ignoring > bi-allelic SNPs: scaffold_1 3639272\n",
      "ignoring > bi-allelic SNPs: scaffold_1 3779217\n",
      "ignoring > bi-allelic SNPs: scaffold_1 3840418\n",
      "ignoring indels: scaffold_1 4803787\n",
      "ignoring > bi-allelic SNPs: scaffold_1 6229080\n",
      "ignoring > bi-allelic SNPs: scaffold_1 7472291\n",
      "ignoring > bi-allelic SNPs: scaffold_1 7593324\n",
      "ignoring > bi-allelic SNPs: scaffold_1 8464313\n",
      "ignoring > bi-allelic SNPs: scaffold_1 8464326\n",
      "ignoring > bi-allelic SNPs: scaffold_1 8555957\n",
      "ignoring indels: scaffold_1 8807214\n",
      "ignoring indels: scaffold_1 8899505\n",
      "ignoring indels: scaffold_1 9047385\n",
      "ignoring > bi-allelic SNPs: scaffold_1 9047388\n",
      "ignoring > bi-allelic SNPs: scaffold_1 9381169\n",
      "ignoring indels: scaffold_1 9381208\n",
      "ignoring indels: scaffold_1 10769609\n",
      "ignoring > bi-allelic SNPs: scaffold_1 11064370\n",
      "ignoring indels: scaffold_1 11064387\n",
      "ignoring > bi-allelic SNPs: scaffold_1 11617397\n",
      "ignoring indels: scaffold_1 12529349\n",
      "ignoring indels: scaffold_1 12529377\n",
      "ignoring indels: scaffold_1 12807130\n",
      "ignoring indels: scaffold_1 12955634\n",
      "ignoring > bi-allelic SNPs: scaffold_1 13080686\n",
      "ignoring > bi-allelic SNPs: scaffold_1 13543235\n",
      "ignoring indels: scaffold_1 13543244\n",
      "ignoring > bi-allelic SNPs: scaffold_1 13592391\n",
      "ignoring indels: scaffold_1 13592408\n",
      "ignoring > bi-allelic SNPs: scaffold_1 13957289\n",
      "ignoring indels: scaffold_1 14803402\n",
      "ignoring indels: scaffold_2 1766968\n",
      "ignoring indels: scaffold_2 2233385\n",
      "ignoring > bi-allelic SNPs: scaffold_2 2247798\n",
      "ignoring > bi-allelic SNPs: scaffold_2 3766033\n",
      "ignoring > bi-allelic SNPs: scaffold_2 3891575\n",
      "ignoring indels: scaffold_2 4361780\n",
      "ignoring > bi-allelic SNPs: scaffold_2 4970112\n",
      "ignoring > bi-allelic SNPs: scaffold_2 5014399\n",
      "ignoring > bi-allelic SNPs: scaffold_2 5627104\n",
      "ignoring indels: scaffold_2 6496319\n",
      "ignoring > bi-allelic SNPs: scaffold_2 7246998\n",
      "ignoring indels: scaffold_2 7994482\n",
      "ignoring > bi-allelic SNPs: scaffold_2 9027299\n",
      "ignoring indels: scaffold_2 9791431\n",
      "ignoring > bi-allelic SNPs: scaffold_2 9878299\n",
      "ignoring indels: scaffold_2 9995175\n",
      "ignoring indels: scaffold_2 10460739\n",
      "ignoring > bi-allelic SNPs: scaffold_2 10572997\n",
      "ignoring > bi-allelic SNPs: scaffold_2 11429317\n",
      "ignoring indels: scaffold_2 11529407\n",
      "ignoring > bi-allelic SNPs: scaffold_2 11536836\n",
      "ignoring > bi-allelic SNPs: scaffold_2 11792120\n",
      "ignoring > bi-allelic SNPs: scaffold_2 11792121\n",
      "ignoring > bi-allelic SNPs: scaffold_2 11792122\n",
      "ignoring > bi-allelic SNPs: scaffold_2 11792123\n",
      "ignoring > bi-allelic SNPs: scaffold_2 11792124\n",
      "ignoring indels: scaffold_2 11896290\n",
      "ignoring > bi-allelic SNPs: scaffold_2 11909296\n",
      "ignoring > bi-allelic SNPs: scaffold_2 11909297\n",
      "ignoring indels: scaffold_2 11954419\n",
      "ignoring indels: scaffold_2 12766628\n",
      "ignoring > bi-allelic SNPs: scaffold_2 12766631\n",
      "ignoring > bi-allelic SNPs: scaffold_2 12883848\n",
      "ignoring indels: scaffold_2 13189916\n",
      "ignoring indels: scaffold_2 13257559\n",
      "ignoring indels: scaffold_2 13496818\n",
      "ignoring indels: scaffold_2 13995097\n",
      "ignoring indels: scaffold_2 14154519\n",
      "ignoring indels: scaffold_2 14556846\n",
      "ignoring > bi-allelic SNPs: scaffold_3 46989\n",
      "ignoring indels: scaffold_3 82646\n",
      "ignoring > bi-allelic SNPs: scaffold_3 120634\n",
      "ignoring > bi-allelic SNPs: scaffold_3 120718\n",
      "ignoring indels: scaffold_3 655510\n",
      "ignoring > bi-allelic SNPs: scaffold_3 673955\n",
      "ignoring > bi-allelic SNPs: scaffold_3 1362438\n",
      "ignoring indels: scaffold_3 1601259\n",
      "ignoring indels: scaffold_3 2152369\n",
      "ignoring indels: scaffold_3 2523428\n",
      "ignoring indels: scaffold_3 2744501\n",
      "ignoring > bi-allelic SNPs: scaffold_3 3283275\n",
      "ignoring indels: scaffold_3 3283291\n",
      "ignoring indels: scaffold_3 3968828\n",
      "ignoring indels: scaffold_3 5642314\n",
      "ignoring > bi-allelic SNPs: scaffold_3 5642408\n",
      "ignoring > bi-allelic SNPs: scaffold_3 5659017\n",
      "ignoring indels: scaffold_3 6052863\n",
      "ignoring indels: scaffold_3 6458456\n",
      "ignoring > bi-allelic SNPs: scaffold_3 6618391\n",
      "ignoring indels: scaffold_3 6632182\n",
      "ignoring indels: scaffold_3 7052783\n",
      "ignoring indels: scaffold_3 7613617\n",
      "ignoring > bi-allelic SNPs: scaffold_3 7916421\n",
      "ignoring indels: scaffold_3 8782728\n",
      "ignoring indels: scaffold_3 9970248\n",
      "ignoring indels: scaffold_3 10283997\n",
      "ignoring > bi-allelic SNPs: scaffold_3 10284010\n",
      "ignoring > bi-allelic SNPs: scaffold_3 10415237\n",
      "ignoring indels: scaffold_3 10967227\n",
      "ignoring indels: scaffold_3 10967313\n",
      "ignoring > bi-allelic SNPs: scaffold_3 11916596\n",
      "ignoring > bi-allelic SNPs: scaffold_3 12282421\n",
      "ignoring indels: scaffold_3 12508872\n",
      "ignoring indels: scaffold_3 12797847\n",
      "ignoring indels: scaffold_3 12797866\n",
      "ignoring > bi-allelic SNPs: scaffold_3 12797869\n",
      "ignoring indels: scaffold_3 12797881\n",
      "ignoring > bi-allelic SNPs: scaffold_3 12797884\n",
      "ignoring indels: scaffold_3 12951973\n",
      "ignoring indels: scaffold_3 13370079\n",
      "ignoring > bi-allelic SNPs: scaffold_3 13386849\n",
      "ignoring > bi-allelic SNPs: scaffold_3 13535399\n",
      "ignoring indels: scaffold_4 1079583\n",
      "ignoring indels: scaffold_4 1613549\n",
      "ignoring indels: scaffold_4 2504733\n",
      "ignoring indels: scaffold_4 2552581\n",
      "ignoring > bi-allelic SNPs: scaffold_4 2638107\n",
      "ignoring indels: scaffold_4 2836403\n",
      "ignoring > bi-allelic SNPs: scaffold_4 3002199\n",
      "ignoring > bi-allelic SNPs: scaffold_4 3263224\n",
      "ignoring indels: scaffold_4 3385564\n",
      "ignoring > bi-allelic SNPs: scaffold_4 3651437\n",
      "ignoring indels: scaffold_4 5367146\n",
      "ignoring indels: scaffold_4 5367181\n",
      "ignoring > bi-allelic SNPs: scaffold_4 5367197\n",
      "ignoring indels: scaffold_4 6675416\n",
      "ignoring indels: scaffold_4 7029619\n",
      "ignoring indels: scaffold_4 8261539\n",
      "ignoring > bi-allelic SNPs: scaffold_4 8669255\n",
      "ignoring > bi-allelic SNPs: scaffold_4 8669276\n",
      "ignoring > bi-allelic SNPs: scaffold_4 8669277\n",
      "ignoring > bi-allelic SNPs: scaffold_4 8669278\n",
      "ignoring > bi-allelic SNPs: scaffold_4 8770984\n",
      "ignoring indels: scaffold_4 11324815\n",
      "ignoring indels: scaffold_5 64962\n",
      "ignoring indels: scaffold_5 230448\n",
      "ignoring > bi-allelic SNPs: scaffold_5 343827\n",
      "ignoring indels: scaffold_5 910243\n",
      "ignoring indels: scaffold_5 977262\n",
      "ignoring indels: scaffold_5 1225222\n",
      "ignoring > bi-allelic SNPs: scaffold_5 3067509\n",
      "ignoring indels: scaffold_5 3291963\n",
      "ignoring > bi-allelic SNPs: scaffold_5 3872160\n",
      "ignoring > bi-allelic SNPs: scaffold_5 4388802\n",
      "ignoring indels: scaffold_5 4567634\n",
      "ignoring > bi-allelic SNPs: scaffold_5 4873289\n",
      "ignoring indels: scaffold_5 5082018\n",
      "ignoring > bi-allelic SNPs: scaffold_5 5918544\n",
      "ignoring > bi-allelic SNPs: scaffold_5 5918545\n",
      "ignoring > bi-allelic SNPs: scaffold_5 6208942\n",
      "ignoring indels: scaffold_5 6675726\n",
      "ignoring indels: scaffold_5 7075764\n",
      "ignoring indels: scaffold_5 7621469\n",
      "ignoring > bi-allelic SNPs: scaffold_5 8354473\n",
      "ignoring indels: scaffold_5 9547433\n",
      "ignoring > bi-allelic SNPs: scaffold_5 10262677\n",
      "ignoring indels: scaffold_5 10531135\n",
      "ignoring indels: scaffold_5 10531142\n",
      "ignoring > bi-allelic SNPs: scaffold_5 10670214\n",
      "ignoring > bi-allelic SNPs: scaffold_5 11095258\n",
      "ignoring > bi-allelic SNPs: scaffold_5 11478684\n",
      "ignoring > bi-allelic SNPs: scaffold_6 1124626\n",
      "ignoring > bi-allelic SNPs: scaffold_6 2268208\n",
      "ignoring indels: scaffold_6 3215784\n",
      "something is wrong with the reference allele at scaffold_6 4554270: AT\n",
      "ignoring > bi-allelic SNPs: scaffold_6 6069703\n",
      "ignoring > bi-allelic SNPs: scaffold_6 6079580\n",
      "ignoring indels: scaffold_6 6105240\n",
      "ignoring > bi-allelic SNPs: scaffold_6 6862442\n",
      "ignoring indels: scaffold_6 6862481\n",
      "ignoring indels: scaffold_6 6862550\n",
      "ignoring indels: scaffold_6 8490574\n",
      "ignoring > bi-allelic SNPs: scaffold_6 8842005\n",
      "ignoring indels: scaffold_6 9116152\n",
      "ignoring > bi-allelic SNPs: scaffold_6 9116181\n",
      "ignoring indels: scaffold_6 10355277\n",
      "ignoring indels: scaffold_7 45904\n",
      "ignoring indels: scaffold_7 1484189\n",
      "ignoring indels: scaffold_7 1772603\n",
      "ignoring > bi-allelic SNPs: scaffold_7 2474384\n",
      "ignoring > bi-allelic SNPs: scaffold_7 2474385\n",
      "ignoring > bi-allelic SNPs: scaffold_7 2474386\n",
      "ignoring indels: scaffold_7 3007197\n",
      "ignoring > bi-allelic SNPs: scaffold_7 3119654\n",
      "ignoring indels: scaffold_7 3277557\n",
      "ignoring > bi-allelic SNPs: scaffold_7 3498578\n",
      "ignoring > bi-allelic SNPs: scaffold_7 4966693\n",
      "ignoring > bi-allelic SNPs: scaffold_7 5043858\n",
      "ignoring indels: scaffold_7 5397584\n",
      "ignoring > bi-allelic SNPs: scaffold_7 5758762\n",
      "ignoring indels: scaffold_7 6585673\n",
      "ignoring indels: scaffold_7 6711091\n",
      "ignoring indels: scaffold_7 7293170\n",
      "ignoring indels: scaffold_7 8653796\n",
      "ignoring > bi-allelic SNPs: scaffold_7 9498880\n",
      "ignoring indels: scaffold_7 10658895\n",
      "ignoring indels: scaffold_8 207716\n",
      "ignoring indels: scaffold_8 1564076\n",
      "ignoring indels: scaffold_8 1942427\n",
      "ignoring > bi-allelic SNPs: scaffold_8 3577751\n",
      "ignoring > bi-allelic SNPs: scaffold_8 3726026\n",
      "ignoring > bi-allelic SNPs: scaffold_8 3739519\n",
      "ignoring indels: scaffold_8 4117920\n",
      "ignoring > bi-allelic SNPs: scaffold_8 4379263\n",
      "ignoring indels: scaffold_8 4390239\n",
      "ignoring indels: scaffold_8 6918787\n",
      "ignoring indels: scaffold_8 8017354\n",
      "ignoring indels: scaffold_8 8164878\n",
      "ignoring > bi-allelic SNPs: scaffold_8 9841611\n",
      "ignoring indels: scaffold_8 10385907\n",
      "ignoring > bi-allelic SNPs: scaffold_8 10445949\n",
      "ignoring indels: scaffold_9 2312316\n",
      "ignoring indels: scaffold_9 3137295\n",
      "ignoring > bi-allelic SNPs: scaffold_9 3808718\n",
      "ignoring > bi-allelic SNPs: scaffold_9 4887448\n",
      "ignoring indels: scaffold_9 5029197\n",
      "ignoring indels: scaffold_9 5214230\n",
      "ignoring > bi-allelic SNPs: scaffold_9 5732118\n",
      "ignoring > bi-allelic SNPs: scaffold_9 7602189\n",
      "ignoring indels: scaffold_9 7958531\n",
      "ignoring indels: scaffold_9 8405917\n",
      "ignoring > bi-allelic SNPs: scaffold_9 8422837\n",
      "ignoring > bi-allelic SNPs: scaffold_9 8422838\n",
      "ignoring > bi-allelic SNPs: scaffold_9 8422843\n",
      "ignoring indels: scaffold_9 8422856\n",
      "ignoring indels: scaffold_9 8422880\n",
      "ignoring > bi-allelic SNPs: scaffold_9 8422885\n",
      "ignoring indels: scaffold_9 8446574\n",
      "ignoring > bi-allelic SNPs: scaffold_9 9164898\n",
      "ignoring > bi-allelic SNPs: scaffold_9 9634801\n",
      "ignoring indels: scaffold_9 9634813\n",
      "ignoring > bi-allelic SNPs: scaffold_9 9985982\n",
      "ignoring indels: scaffold_9 10175521\n",
      "ignoring indels: scaffold_9 10321420\n",
      "ignoring > bi-allelic SNPs: scaffold_10 204589\n",
      "ignoring > bi-allelic SNPs: scaffold_10 258321\n",
      "ignoring indels: scaffold_10 578452\n",
      "ignoring indels: scaffold_10 829120\n",
      "ignoring indels: scaffold_10 905093\n",
      "ignoring indels: scaffold_10 946469\n",
      "ignoring > bi-allelic SNPs: scaffold_10 1047531\n",
      "ignoring > bi-allelic SNPs: scaffold_10 1047532\n",
      "ignoring > bi-allelic SNPs: scaffold_10 1047534\n",
      "ignoring > bi-allelic SNPs: scaffold_10 1047542\n",
      "ignoring > bi-allelic SNPs: scaffold_10 1261654\n",
      "ignoring indels: scaffold_10 1673056\n",
      "ignoring > bi-allelic SNPs: scaffold_10 1698058\n",
      "ignoring > bi-allelic SNPs: scaffold_10 2499908\n",
      "ignoring > bi-allelic SNPs: scaffold_10 2913172\n",
      "ignoring indels: scaffold_10 3025894\n",
      "ignoring indels: scaffold_10 3498333\n",
      "ignoring indels: scaffold_10 4170577\n",
      "ignoring indels: scaffold_10 4170652\n",
      "ignoring indels: scaffold_10 4860358\n",
      "ignoring > bi-allelic SNPs: scaffold_10 6392828\n",
      "ignoring indels: scaffold_10 7033353\n",
      "ignoring indels: scaffold_10 7122990\n",
      "ignoring > bi-allelic SNPs: scaffold_10 8735590\n",
      "ignoring indels: scaffold_10 8846685\n",
      "ignoring > bi-allelic SNPs: scaffold_11 61477\n",
      "ignoring > bi-allelic SNPs: scaffold_11 1171782\n",
      "ignoring > bi-allelic SNPs: scaffold_11 1589882\n",
      "ignoring indels: scaffold_11 1589906\n",
      "ignoring indels: scaffold_11 4809442\n",
      "ignoring > bi-allelic SNPs: scaffold_11 4931225\n",
      "ignoring > bi-allelic SNPs: scaffold_11 5089837\n",
      "ignoring > bi-allelic SNPs: scaffold_11 5089841\n",
      "ignoring indels: scaffold_11 5876562\n",
      "ignoring indels: scaffold_11 5885441\n",
      "ignoring > bi-allelic SNPs: scaffold_11 5930439\n",
      "ignoring indels: scaffold_11 7140394\n",
      "ignoring > bi-allelic SNPs: scaffold_11 7272555\n",
      "ignoring > bi-allelic SNPs: scaffold_11 7272583\n",
      "ignoring indels: scaffold_12 536027\n",
      "ignoring indels: scaffold_12 814272\n",
      "ignoring > bi-allelic SNPs: scaffold_12 1384160\n",
      "ignoring > bi-allelic SNPs: scaffold_12 1384166\n",
      "ignoring > bi-allelic SNPs: scaffold_12 1384167\n",
      "ignoring indels: scaffold_12 1837254\n",
      "ignoring indels: scaffold_12 5055140\n",
      "ignoring indels: scaffold_12 6022393\n",
      "ignoring > bi-allelic SNPs: scaffold_12 7536601\n",
      "ignoring > bi-allelic SNPs: scaffold_12 7536605\n",
      "ignoring indels: scaffold_12 7587617\n",
      "ignoring indels: scaffold_12 7678803\n",
      "ignoring indels: scaffold_12 7829528\n",
      "ignoring indels: scaffold_12 7993505\n",
      "ignoring indels: scaffold_12 8114997\n",
      "ignoring > bi-allelic SNPs: scaffold_12 8426865\n",
      "ignoring indels: scaffold_13 178655\n",
      "ignoring > bi-allelic SNPs: scaffold_13 1058085\n",
      "ignoring indels: scaffold_13 1341315\n",
      "ignoring indels: scaffold_13 2502843\n",
      "ignoring indels: scaffold_13 2502850\n",
      "ignoring > bi-allelic SNPs: scaffold_13 4026005\n",
      "ignoring indels: scaffold_13 4026050\n",
      "ignoring indels: scaffold_13 4383258\n",
      "ignoring indels: scaffold_13 4950941\n",
      "ignoring indels: scaffold_13 5210949\n",
      "ignoring > bi-allelic SNPs: scaffold_13 5746665\n",
      "ignoring indels: scaffold_13 5960707\n",
      "ignoring indels: scaffold_13 6225677\n",
      "ignoring indels: scaffold_13 6882401\n",
      "ignoring > bi-allelic SNPs: scaffold_13 6882405\n",
      "ignoring indels: scaffold_13 7821443\n",
      "ignoring indels: scaffold_13 7897318\n",
      "ignoring indels: scaffold_13 8498411\n",
      "ignoring indels: scaffold_13 8671953\n",
      "ignoring indels: scaffold_13 8671963\n",
      "ignoring indels: scaffold_14 992414\n",
      "ignoring > bi-allelic SNPs: scaffold_14 1153773\n",
      "ignoring > bi-allelic SNPs: scaffold_14 1474795\n",
      "ignoring > bi-allelic SNPs: scaffold_14 2243811\n",
      "ignoring indels: scaffold_14 2270052\n",
      "ignoring indels: scaffold_14 2603332\n",
      "ignoring indels: scaffold_14 3676995\n",
      "ignoring indels: scaffold_14 3797739\n",
      "ignoring indels: scaffold_14 4032454\n",
      "ignoring > bi-allelic SNPs: scaffold_14 4730706\n",
      "ignoring > bi-allelic SNPs: scaffold_14 4730707\n",
      "ignoring > bi-allelic SNPs: scaffold_14 4730708\n",
      "ignoring > bi-allelic SNPs: scaffold_14 4730709\n",
      "ignoring indels: scaffold_14 5373270\n",
      "ignoring > bi-allelic SNPs: scaffold_14 5725432\n",
      "ignoring > bi-allelic SNPs: scaffold_14 6195446\n",
      "ignoring indels: scaffold_14 6359916\n",
      "ignoring indels: scaffold_14 6667641\n",
      "ignoring indels: scaffold_14 7027135\n",
      "ignoring indels: scaffold_14 8371731\n",
      "ignoring > bi-allelic SNPs: scaffold_15 85226\n",
      "ignoring indels: scaffold_15 362585\n",
      "ignoring > bi-allelic SNPs: scaffold_15 569063\n",
      "ignoring indels: scaffold_15 724375\n",
      "ignoring indels: scaffold_15 1025848\n",
      "ignoring indels: scaffold_15 1203930\n",
      "ignoring > bi-allelic SNPs: scaffold_15 1655167\n",
      "ignoring > bi-allelic SNPs: scaffold_15 2435859\n",
      "ignoring indels: scaffold_15 2462238\n",
      "ignoring indels: scaffold_15 2515509\n",
      "ignoring indels: scaffold_15 4327964\n",
      "ignoring > bi-allelic SNPs: scaffold_15 4702312\n",
      "ignoring indels: scaffold_15 4714726\n",
      "ignoring > bi-allelic SNPs: scaffold_15 4915515\n",
      "ignoring indels: scaffold_15 5353081\n",
      "ignoring > bi-allelic SNPs: scaffold_15 6127161\n",
      "ignoring indels: scaffold_15 6509070\n",
      "ignoring > bi-allelic SNPs: scaffold_15 6993223\n",
      "ignoring > bi-allelic SNPs: scaffold_15 7157044\n",
      "ignoring indels: scaffold_15 7292973\n",
      "ignoring > bi-allelic SNPs: scaffold_15 7909406\n",
      "ignoring > bi-allelic SNPs: scaffold_15 8067606\n",
      "ignoring indels: scaffold_16 100501\n",
      "ignoring indels: scaffold_16 557318\n",
      "ignoring > bi-allelic SNPs: scaffold_16 1190447\n",
      "ignoring indels: scaffold_16 1728752\n",
      "ignoring indels: scaffold_16 4481860\n",
      "ignoring indels: scaffold_16 4612424\n",
      "ignoring indels: scaffold_16 4612449\n",
      "ignoring indels: scaffold_16 4682953\n",
      "ignoring > bi-allelic SNPs: scaffold_16 5146508\n",
      "ignoring indels: scaffold_16 5323021\n",
      "ignoring indels: scaffold_16 5323038\n",
      "ignoring indels: scaffold_16 5918285\n",
      "ignoring indels: scaffold_16 6136168\n",
      "ignoring > bi-allelic SNPs: scaffold_16 7150272\n",
      "ignoring indels: scaffold_16 7150318\n",
      "ignoring indels: scaffold_16 7265428\n",
      "ignoring indels: scaffold_17 404462\n",
      "ignoring > bi-allelic SNPs: scaffold_17 422484\n",
      "ignoring indels: scaffold_17 1039644\n",
      "ignoring indels: scaffold_17 1266841\n",
      "something is wrong with the reference allele at scaffold_17 1266857: TC\n",
      "ignoring indels: scaffold_17 1278454\n",
      "ignoring indels: scaffold_17 1278625\n",
      "ignoring > bi-allelic SNPs: scaffold_17 1565691\n",
      "ignoring indels: scaffold_17 2630351\n",
      "ignoring indels: scaffold_17 3104970\n",
      "ignoring indels: scaffold_17 3480817\n",
      "ignoring indels: scaffold_17 4202938\n",
      "ignoring indels: scaffold_17 4501416\n",
      "ignoring > bi-allelic SNPs: scaffold_17 4788872\n",
      "ignoring indels: scaffold_17 6151206\n",
      "ignoring > bi-allelic SNPs: scaffold_17 6195507\n",
      "ignoring indels: scaffold_17 6195510\n",
      "ignoring indels: scaffold_17 6383684\n",
      "ignoring indels: scaffold_17 6888303\n",
      "ignoring indels: scaffold_17 6888312\n",
      "ignoring indels: scaffold_17 6888328\n",
      "ignoring > bi-allelic SNPs: scaffold_17 6939058\n",
      "ignoring > bi-allelic SNPs: scaffold_17 7204994\n",
      "ignoring indels: scaffold_17 7301726\n",
      "ignoring indels: scaffold_18 2406651\n",
      "ignoring indels: scaffold_18 2406759\n",
      "ignoring > bi-allelic SNPs: scaffold_18 2506937\n",
      "ignoring indels: scaffold_18 3160470\n",
      "ignoring indels: scaffold_18 4596433\n",
      "ignoring > bi-allelic SNPs: scaffold_18 6224421\n",
      "ignoring indels: scaffold_18 6226295\n",
      "ignoring indels: scaffold_19 197496\n",
      "ignoring > bi-allelic SNPs: scaffold_19 412990\n",
      "ignoring indels: scaffold_19 947336\n",
      "ignoring indels: scaffold_19 1052137\n",
      "ignoring indels: scaffold_19 1052174\n",
      "ignoring > bi-allelic SNPs: scaffold_19 1144805\n",
      "ignoring > bi-allelic SNPs: scaffold_19 1144815\n",
      "ignoring > bi-allelic SNPs: scaffold_19 1252848\n",
      "ignoring indels: scaffold_19 1252869\n",
      "ignoring > bi-allelic SNPs: scaffold_19 1481290\n",
      "ignoring indels: scaffold_19 1873880\n",
      "ignoring indels: scaffold_19 2215762\n",
      "ignoring > bi-allelic SNPs: scaffold_19 2286557\n",
      "ignoring indels: scaffold_19 2358064\n",
      "ignoring > bi-allelic SNPs: scaffold_19 2850521\n",
      "ignoring > bi-allelic SNPs: scaffold_19 3169227\n",
      "ignoring indels: scaffold_19 3495711\n",
      "ignoring indels: scaffold_19 4420867\n",
      "ignoring > bi-allelic SNPs: scaffold_19 5219146\n",
      "ignoring > bi-allelic SNPs: scaffold_19 5457930\n",
      "ignoring indels: scaffold_19 6376379\n",
      "ignoring > bi-allelic SNPs: scaffold_19 7068849\n",
      "ignoring > bi-allelic SNPs: scaffold_19 7305803\n",
      "ignoring indels: scaffold_19 7305877\n",
      "ignoring > bi-allelic SNPs: scaffold_20 148336\n",
      "ignoring indels: scaffold_20 435934\n",
      "ignoring > bi-allelic SNPs: scaffold_20 1135040\n",
      "ignoring > bi-allelic SNPs: scaffold_20 1256009\n",
      "ignoring > bi-allelic SNPs: scaffold_20 1256025\n",
      "ignoring > bi-allelic SNPs: scaffold_20 1510109\n",
      "ignoring > bi-allelic SNPs: scaffold_20 1698982\n",
      "ignoring > bi-allelic SNPs: scaffold_20 1740725\n",
      "ignoring indels: scaffold_20 1934938\n",
      "ignoring > bi-allelic SNPs: scaffold_20 2229280\n",
      "ignoring indels: scaffold_20 2281582\n",
      "ignoring indels: scaffold_20 2855050\n",
      "ignoring indels: scaffold_20 3212074\n",
      "ignoring indels: scaffold_20 3212084\n",
      "ignoring indels: scaffold_20 3449001\n",
      "ignoring > bi-allelic SNPs: scaffold_20 3866610\n",
      "ignoring indels: scaffold_20 3973073\n",
      "ignoring > bi-allelic SNPs: scaffold_20 4693021\n",
      "ignoring > bi-allelic SNPs: scaffold_20 4857386\n",
      "ignoring > bi-allelic SNPs: scaffold_20 4975165\n",
      "ignoring indels: scaffold_20 5372436\n",
      "ignoring indels: scaffold_20 5803682\n",
      "ignoring indels: scaffold_20 7245225\n",
      "ignoring indels: scaffold_21 2243670\n",
      "ignoring indels: scaffold_21 2366684\n",
      "ignoring indels: scaffold_21 2859958\n",
      "ignoring > bi-allelic SNPs: scaffold_21 2859983\n",
      "ignoring indels: scaffold_21 3447806\n",
      "ignoring indels: scaffold_21 4477392\n",
      "ignoring indels: scaffold_21 4518690\n",
      "ignoring indels: scaffold_21 4766366\n",
      "ignoring indels: scaffold_21 4879155\n",
      "ignoring > bi-allelic SNPs: scaffold_21 4994931\n",
      "ignoring > bi-allelic SNPs: scaffold_21 4994932\n",
      "ignoring > bi-allelic SNPs: scaffold_21 5227391\n",
      "ignoring indels: scaffold_21 5227422\n",
      "ignoring indels: scaffold_21 5258712\n",
      "ignoring indels: scaffold_21 5276706\n",
      "ignoring indels: scaffold_21 5807056\n",
      "ignoring indels: scaffold_21 5807076\n",
      "ignoring > bi-allelic SNPs: scaffold_21 6234240\n",
      "ignoring indels: scaffold_21 6500578\n",
      "ignoring indels: scaffold_21 6543283\n",
      "ignoring indels: scaffold_21 6896373\n",
      "ignoring > bi-allelic SNPs: scaffold_22 688498\n",
      "ignoring indels: scaffold_22 909466\n",
      "ignoring > bi-allelic SNPs: scaffold_22 2538321\n",
      "ignoring > bi-allelic SNPs: scaffold_22 3096191\n",
      "ignoring > bi-allelic SNPs: scaffold_22 3324016\n",
      "ignoring indels: scaffold_22 4212118\n",
      "ignoring indels: scaffold_22 4747876\n",
      "ignoring > bi-allelic SNPs: scaffold_22 4747893\n",
      "ignoring indels: scaffold_23 27852\n",
      "ignoring indels: scaffold_23 1303287\n",
      "ignoring indels: scaffold_23 1729803\n",
      "ignoring indels: scaffold_23 1729810\n",
      "ignoring indels: scaffold_23 2891971\n",
      "ignoring indels: scaffold_23 2891992\n",
      "ignoring > bi-allelic SNPs: scaffold_23 2891996\n",
      "ignoring > bi-allelic SNPs: scaffold_23 2952256\n",
      "ignoring indels: scaffold_23 3101056\n",
      "ignoring indels: scaffold_23 3146818\n",
      "ignoring > bi-allelic SNPs: scaffold_23 3855859\n",
      "ignoring > bi-allelic SNPs: scaffold_23 4370470\n",
      "ignoring > bi-allelic SNPs: scaffold_23 4370474\n",
      "ignoring > bi-allelic SNPs: scaffold_23 4428322\n",
      "ignoring indels: scaffold_23 4567697\n",
      "ignoring > bi-allelic SNPs: scaffold_23 5096181\n",
      "ignoring indels: scaffold_23 5096233\n",
      "ignoring > bi-allelic SNPs: scaffold_23 5568914\n",
      "ignoring indels: scaffold_23 6344118\n",
      "ignoring indels: scaffold_24 479097\n",
      "ignoring > bi-allelic SNPs: scaffold_24 669971\n",
      "ignoring indels: scaffold_24 907146\n",
      "ignoring indels: scaffold_24 1416643\n",
      "ignoring indels: scaffold_24 1416649\n",
      "ignoring > bi-allelic SNPs: scaffold_24 2021705\n",
      "ignoring > bi-allelic SNPs: scaffold_24 4227798\n",
      "ignoring indels: scaffold_24 4431709\n",
      "ignoring indels: scaffold_24 4522271\n",
      "ignoring indels: scaffold_24 4522288\n",
      "ignoring indels: scaffold_24 4883317\n",
      "ignoring indels: scaffold_24 6106537\n",
      "ignoring > bi-allelic SNPs: scaffold_25 1064707\n",
      "ignoring > bi-allelic SNPs: scaffold_25 1138515\n",
      "ignoring > bi-allelic SNPs: scaffold_25 1587616\n",
      "ignoring indels: scaffold_25 1891563\n",
      "ignoring indels: scaffold_25 1962471\n",
      "ignoring indels: scaffold_25 1991638\n",
      "ignoring indels: scaffold_25 2190594\n",
      "ignoring indels: scaffold_25 2312528\n",
      "ignoring indels: scaffold_25 2312530\n",
      "ignoring indels: scaffold_25 2719578\n",
      "ignoring indels: scaffold_25 2719608\n",
      "ignoring indels: scaffold_25 2738621\n",
      "ignoring > bi-allelic SNPs: scaffold_25 2738638\n",
      "ignoring > bi-allelic SNPs: scaffold_25 2913159\n",
      "ignoring indels: scaffold_25 3518270\n",
      "ignoring > bi-allelic SNPs: scaffold_25 4323870\n",
      "ignoring indels: scaffold_25 4323878\n",
      "ignoring indels: scaffold_25 4323896\n",
      "ignoring > bi-allelic SNPs: scaffold_25 4649472\n",
      "ignoring indels: scaffold_25 4812039\n",
      "ignoring > bi-allelic SNPs: scaffold_25 5319393\n",
      "ignoring > bi-allelic SNPs: scaffold_25 5508296\n",
      "ignoring indels: scaffold_25 6114784\n",
      "ignoring > bi-allelic SNPs: scaffold_26 23014\n",
      "ignoring > bi-allelic SNPs: scaffold_26 23018\n",
      "ignoring > bi-allelic SNPs: scaffold_26 23036\n",
      "ignoring indels: scaffold_26 144663\n",
      "ignoring > bi-allelic SNPs: scaffold_26 651957\n",
      "ignoring indels: scaffold_26 662751\n",
      "ignoring indels: scaffold_26 751679\n",
      "ignoring indels: scaffold_26 896503\n",
      "ignoring > bi-allelic SNPs: scaffold_26 1442834\n",
      "ignoring > bi-allelic SNPs: scaffold_26 2167182\n",
      "ignoring indels: scaffold_26 2326796\n",
      "ignoring > bi-allelic SNPs: scaffold_26 2326881\n",
      "ignoring indels: scaffold_26 2696003\n",
      "ignoring indels: scaffold_26 3012724\n",
      "ignoring indels: scaffold_26 4543443\n",
      "ignoring indels: scaffold_26 4814160\n",
      "ignoring indels: scaffold_26 6125806\n",
      "ignoring indels: scaffold_26 6125816\n",
      "ignoring > bi-allelic SNPs: scaffold_26 6407447\n",
      "ignoring > bi-allelic SNPs: scaffold_26 6407460\n",
      "ignoring indels: scaffold_27 76258\n",
      "ignoring indels: scaffold_27 249669\n",
      "ignoring > bi-allelic SNPs: scaffold_27 487951\n",
      "ignoring > bi-allelic SNPs: scaffold_27 754749\n",
      "ignoring indels: scaffold_27 754752\n",
      "ignoring indels: scaffold_27 754759\n",
      "ignoring > bi-allelic SNPs: scaffold_27 754764\n",
      "ignoring indels: scaffold_27 754785\n",
      "ignoring indels: scaffold_27 1092810\n",
      "ignoring indels: scaffold_27 1710687\n",
      "ignoring indels: scaffold_27 2363418\n",
      "something is wrong with the reference allele at scaffold_27 2363423: TT\n",
      "ignoring indels: scaffold_27 2363425\n",
      "ignoring > bi-allelic SNPs: scaffold_27 2363435\n",
      "ignoring > bi-allelic SNPs: scaffold_27 4496328\n",
      "ignoring indels: scaffold_27 4638708\n",
      "ignoring > bi-allelic SNPs: scaffold_27 4638713\n",
      "ignoring indels: scaffold_27 4638756\n",
      "ignoring indels: scaffold_27 5497195\n",
      "ignoring indels: scaffold_27 5746272\n",
      "ignoring indels: scaffold_27 5975553\n",
      "ignoring > bi-allelic SNPs: scaffold_27 6121876\n",
      "ignoring > bi-allelic SNPs: scaffold_28 257524\n",
      "ignoring > bi-allelic SNPs: scaffold_28 1064576\n",
      "ignoring indels: scaffold_28 3517629\n",
      "ignoring > bi-allelic SNPs: scaffold_28 3649836\n",
      "ignoring indels: scaffold_28 3649908\n",
      "ignoring indels: scaffold_28 3951734\n",
      "ignoring indels: scaffold_28 4206212\n",
      "ignoring > bi-allelic SNPs: scaffold_28 4206223\n",
      "ignoring indels: scaffold_28 6518569\n",
      "ignoring > bi-allelic SNPs: scaffold_29 373767\n",
      "ignoring indels: scaffold_29 431509\n",
      "ignoring > bi-allelic SNPs: scaffold_29 1854798\n",
      "ignoring indels: scaffold_29 3129413\n",
      "ignoring indels: scaffold_29 5320422\n",
      "ignoring indels: scaffold_30 903092\n",
      "ignoring > bi-allelic SNPs: scaffold_30 2323975\n",
      "ignoring indels: scaffold_30 2762572\n",
      "ignoring indels: scaffold_30 5247478\n",
      "ignoring indels: scaffold_30 5273986\n",
      "ignoring > bi-allelic SNPs: scaffold_30 6007987\n",
      "ignoring indels: scaffold_30 6109146\n",
      "ignoring indels: scaffold_30 6109215\n",
      "ignoring indels: scaffold_31 666607\n",
      "ignoring > bi-allelic SNPs: scaffold_31 1222092\n",
      "ignoring > bi-allelic SNPs: scaffold_31 1771581\n",
      "ignoring indels: scaffold_31 1786463\n",
      "ignoring > bi-allelic SNPs: scaffold_31 2151719\n",
      "ignoring > bi-allelic SNPs: scaffold_31 2229173\n",
      "ignoring > bi-allelic SNPs: scaffold_31 2732453\n",
      "ignoring indels: scaffold_31 2798701\n",
      "ignoring indels: scaffold_31 2835882\n",
      "ignoring indels: scaffold_31 2963487\n",
      "ignoring > bi-allelic SNPs: scaffold_31 3100979\n",
      "ignoring > bi-allelic SNPs: scaffold_31 3122184\n",
      "ignoring indels: scaffold_31 4066300\n",
      "ignoring indels: scaffold_31 4112582\n",
      "ignoring indels: scaffold_31 5061815\n",
      "ignoring indels: scaffold_31 6237683\n",
      "ignoring > bi-allelic SNPs: scaffold_32 2786974\n",
      "ignoring indels: scaffold_32 2786986\n",
      "ignoring > bi-allelic SNPs: scaffold_32 4880011\n",
      "ignoring indels: scaffold_32 5888351\n",
      "ignoring > bi-allelic SNPs: scaffold_32 5925580\n",
      "ignoring > bi-allelic SNPs: scaffold_32 6067772\n",
      "ignoring indels: scaffold_32 6149633\n",
      "ignoring > bi-allelic SNPs: scaffold_33 65062\n",
      "ignoring > bi-allelic SNPs: scaffold_33 244732\n",
      "ignoring indels: scaffold_33 1414960\n",
      "ignoring indels: scaffold_33 1540713\n",
      "ignoring indels: scaffold_33 2042408\n",
      "ignoring indels: scaffold_33 2476813\n",
      "ignoring > bi-allelic SNPs: scaffold_33 2973972\n",
      "ignoring > bi-allelic SNPs: scaffold_33 2973984\n",
      "ignoring > bi-allelic SNPs: scaffold_33 4322478\n",
      "ignoring indels: scaffold_33 5279860\n",
      "ignoring indels: scaffold_34 211987\n",
      "ignoring indels: scaffold_34 681914\n",
      "ignoring > bi-allelic SNPs: scaffold_34 828673\n",
      "ignoring indels: scaffold_34 1214034\n",
      "ignoring > bi-allelic SNPs: scaffold_34 1255318\n",
      "ignoring indels: scaffold_34 1408795\n",
      "ignoring > bi-allelic SNPs: scaffold_34 1722801\n",
      "ignoring indels: scaffold_34 1922544\n",
      "ignoring indels: scaffold_34 1934514\n",
      "ignoring indels: scaffold_34 2315515\n",
      "ignoring indels: scaffold_34 3491909\n",
      "ignoring > bi-allelic SNPs: scaffold_34 3491939\n",
      "ignoring > bi-allelic SNPs: scaffold_34 3731394\n",
      "ignoring indels: scaffold_34 3985842\n",
      "ignoring > bi-allelic SNPs: scaffold_34 4708334\n",
      "ignoring > bi-allelic SNPs: scaffold_34 4708337\n",
      "ignoring indels: scaffold_34 5282267\n",
      "ignoring indels: scaffold_35 34810\n",
      "ignoring indels: scaffold_35 38167\n",
      "ignoring > bi-allelic SNPs: scaffold_35 326543\n",
      "ignoring indels: scaffold_35 514072\n",
      "ignoring indels: scaffold_35 1081575\n",
      "ignoring > bi-allelic SNPs: scaffold_35 1289996\n",
      "ignoring indels: scaffold_35 1907731\n",
      "ignoring indels: scaffold_35 2382271\n",
      "ignoring indels: scaffold_35 2386602\n",
      "ignoring > bi-allelic SNPs: scaffold_35 2766520\n",
      "ignoring > bi-allelic SNPs: scaffold_35 2766521\n",
      "ignoring > bi-allelic SNPs: scaffold_35 2766527\n",
      "something is wrong with the reference allele at scaffold_35 3048150: AG\n",
      "ignoring indels: scaffold_35 3249301\n",
      "something is wrong with the reference allele at scaffold_35 3623647: CT\n",
      "ignoring indels: scaffold_35 4714083\n",
      "ignoring indels: scaffold_35 4714134\n",
      "ignoring > bi-allelic SNPs: scaffold_35 5026003\n",
      "ignoring indels: scaffold_35 5053897\n",
      "ignoring > bi-allelic SNPs: scaffold_36 740682\n",
      "ignoring > bi-allelic SNPs: scaffold_36 740686\n",
      "ignoring > bi-allelic SNPs: scaffold_36 915010\n",
      "ignoring indels: scaffold_36 1378303\n",
      "ignoring indels: scaffold_36 1650461\n",
      "ignoring > bi-allelic SNPs: scaffold_36 2003758\n",
      "ignoring indels: scaffold_36 2005850\n",
      "ignoring > bi-allelic SNPs: scaffold_36 2359983\n",
      "ignoring > bi-allelic SNPs: scaffold_36 2359984\n",
      "ignoring indels: scaffold_36 2519388\n",
      "ignoring indels: scaffold_36 3200443\n",
      "ignoring indels: scaffold_36 3506771\n",
      "ignoring indels: scaffold_36 4382408\n",
      "ignoring indels: scaffold_36 4605204\n",
      "ignoring > bi-allelic SNPs: scaffold_36 4605212\n",
      "ignoring > bi-allelic SNPs: scaffold_36 4687607\n",
      "ignoring > bi-allelic SNPs: scaffold_36 5154425\n",
      "ignoring > bi-allelic SNPs: scaffold_36 5276017\n",
      "ignoring > bi-allelic SNPs: scaffold_36 5276018\n",
      "ignoring > bi-allelic SNPs: scaffold_36 5276019\n",
      "ignoring > bi-allelic SNPs: scaffold_36 5276020\n",
      "ignoring > bi-allelic SNPs: scaffold_36 5276024\n",
      "ignoring > bi-allelic SNPs: scaffold_37 279897\n",
      "ignoring > bi-allelic SNPs: scaffold_37 1013317\n",
      "ignoring indels: scaffold_37 1191889\n",
      "ignoring indels: scaffold_37 1191985\n",
      "ignoring > bi-allelic SNPs: scaffold_37 3976259\n",
      "ignoring indels: scaffold_37 4332525\n",
      "ignoring indels: scaffold_37 4345691\n",
      "ignoring > bi-allelic SNPs: scaffold_37 4402602\n",
      "ignoring > bi-allelic SNPs: scaffold_37 4992459\n",
      "ignoring > bi-allelic SNPs: scaffold_38 323325\n",
      "ignoring > bi-allelic SNPs: scaffold_38 357219\n",
      "ignoring > bi-allelic SNPs: scaffold_38 1159197\n",
      "ignoring indels: scaffold_38 2614250\n",
      "ignoring indels: scaffold_38 4016258\n",
      "ignoring indels: scaffold_38 4347877\n",
      "ignoring > bi-allelic SNPs: scaffold_38 4804425\n",
      "ignoring > bi-allelic SNPs: scaffold_39 1277417\n",
      "ignoring indels: scaffold_39 1561870\n",
      "ignoring > bi-allelic SNPs: scaffold_39 1561872\n",
      "ignoring indels: scaffold_39 1561888\n",
      "ignoring > bi-allelic SNPs: scaffold_39 2215857\n",
      "ignoring indels: scaffold_39 2671936\n",
      "ignoring > bi-allelic SNPs: scaffold_39 2974078\n",
      "ignoring > bi-allelic SNPs: scaffold_39 2979383\n",
      "ignoring indels: scaffold_39 3162712\n",
      "ignoring indels: scaffold_39 3166872\n",
      "ignoring > bi-allelic SNPs: scaffold_39 3253789\n",
      "ignoring > bi-allelic SNPs: scaffold_39 3444257\n",
      "ignoring indels: scaffold_39 3444306\n",
      "ignoring indels: scaffold_39 3945784\n",
      "ignoring > bi-allelic SNPs: scaffold_39 4626498\n",
      "ignoring > bi-allelic SNPs: scaffold_40 274157\n",
      "ignoring indels: scaffold_40 1355008\n",
      "ignoring indels: scaffold_40 1355287\n",
      "ignoring indels: scaffold_40 2137544\n",
      "ignoring > bi-allelic SNPs: scaffold_40 2221432\n",
      "ignoring indels: scaffold_40 2452920\n",
      "ignoring indels: scaffold_40 2967593\n",
      "ignoring > bi-allelic SNPs: scaffold_40 3022678\n",
      "ignoring indels: scaffold_40 4405839\n",
      "ignoring indels: scaffold_40 4406001\n",
      "ignoring indels: scaffold_41 997716\n",
      "ignoring indels: scaffold_41 1268303\n",
      "ignoring indels: scaffold_41 2079013\n",
      "ignoring indels: scaffold_41 2444756\n",
      "ignoring indels: scaffold_41 3700974\n",
      "ignoring indels: scaffold_41 3727635\n",
      "ignoring > bi-allelic SNPs: scaffold_41 3748188\n",
      "ignoring indels: scaffold_41 4658475\n",
      "ignoring > bi-allelic SNPs: scaffold_42 496103\n",
      "ignoring indels: scaffold_42 846597\n",
      "ignoring indels: scaffold_42 999461\n",
      "ignoring indels: scaffold_42 1057483\n",
      "ignoring indels: scaffold_42 1673449\n",
      "ignoring indels: scaffold_42 1730765\n",
      "ignoring > bi-allelic SNPs: scaffold_42 1730769\n",
      "ignoring > bi-allelic SNPs: scaffold_42 2626413\n",
      "ignoring > bi-allelic SNPs: scaffold_42 4132705\n",
      "ignoring indels: scaffold_42 4374973\n",
      "ignoring indels: scaffold_42 4375003\n",
      "ignoring indels: scaffold_43 96498\n",
      "ignoring indels: scaffold_43 239027\n",
      "ignoring indels: scaffold_43 304036\n",
      "ignoring indels: scaffold_43 641288\n",
      "ignoring > bi-allelic SNPs: scaffold_43 1406265\n",
      "ignoring indels: scaffold_43 1406280\n",
      "ignoring indels: scaffold_43 1410279\n",
      "ignoring > bi-allelic SNPs: scaffold_43 1538307\n",
      "ignoring indels: scaffold_43 1546350\n",
      "ignoring > bi-allelic SNPs: scaffold_43 1582731\n",
      "ignoring indels: scaffold_43 2355557\n",
      "ignoring > bi-allelic SNPs: scaffold_43 2418879\n",
      "ignoring > bi-allelic SNPs: scaffold_43 2620361\n",
      "ignoring > bi-allelic SNPs: scaffold_43 2766542\n",
      "ignoring indels: scaffold_43 2862951\n",
      "ignoring indels: scaffold_43 2871123\n",
      "ignoring indels: scaffold_43 3332536\n",
      "ignoring indels: scaffold_43 3695977\n",
      "ignoring indels: scaffold_43 4078757\n",
      "ignoring indels: scaffold_43 4314075\n",
      "ignoring indels: scaffold_43 4377188\n",
      "ignoring > bi-allelic SNPs: scaffold_44 234597\n",
      "ignoring indels: scaffold_44 338841\n",
      "ignoring indels: scaffold_44 500511\n",
      "ignoring indels: scaffold_44 500548\n",
      "ignoring > bi-allelic SNPs: scaffold_44 626569\n",
      "ignoring > bi-allelic SNPs: scaffold_44 626583\n",
      "ignoring > bi-allelic SNPs: scaffold_44 1266273\n",
      "ignoring > bi-allelic SNPs: scaffold_44 1410908\n",
      "ignoring > bi-allelic SNPs: scaffold_44 1908207\n",
      "ignoring > bi-allelic SNPs: scaffold_44 2540482\n",
      "ignoring > bi-allelic SNPs: scaffold_44 2687070\n",
      "ignoring indels: scaffold_44 3009663\n",
      "ignoring indels: scaffold_44 3357209\n",
      "ignoring > bi-allelic SNPs: scaffold_44 4218611\n",
      "ignoring > bi-allelic SNPs: scaffold_45 357463\n",
      "ignoring > bi-allelic SNPs: scaffold_45 357465\n",
      "ignoring > bi-allelic SNPs: scaffold_45 357476\n",
      "ignoring > bi-allelic SNPs: scaffold_45 357500\n",
      "ignoring indels: scaffold_45 393150\n",
      "ignoring indels: scaffold_45 637420\n",
      "ignoring indels: scaffold_45 2874403\n",
      "ignoring > bi-allelic SNPs: scaffold_45 3890516\n",
      "ignoring > bi-allelic SNPs: scaffold_46 171295\n",
      "ignoring indels: scaffold_46 214253\n",
      "ignoring indels: scaffold_46 270720\n",
      "ignoring indels: scaffold_46 506396\n",
      "ignoring indels: scaffold_46 565998\n",
      "ignoring > bi-allelic SNPs: scaffold_46 566005\n",
      "ignoring > bi-allelic SNPs: scaffold_46 589106\n",
      "ignoring > bi-allelic SNPs: scaffold_46 589113\n",
      "ignoring indels: scaffold_46 589162\n",
      "ignoring indels: scaffold_46 1241312\n",
      "ignoring indels: scaffold_46 1625739\n",
      "ignoring indels: scaffold_46 1911529\n",
      "ignoring indels: scaffold_46 2221320\n",
      "ignoring > bi-allelic SNPs: scaffold_46 2730881\n",
      "ignoring > bi-allelic SNPs: scaffold_46 2730900\n",
      "ignoring indels: scaffold_46 2878068\n",
      "ignoring > bi-allelic SNPs: scaffold_46 2878091\n",
      "ignoring > bi-allelic SNPs: scaffold_46 3150603\n",
      "ignoring indels: scaffold_46 3228341\n",
      "ignoring > bi-allelic SNPs: scaffold_46 3394042\n",
      "ignoring > bi-allelic SNPs: scaffold_46 3394045\n",
      "ignoring indels: scaffold_46 3394063\n",
      "ignoring indels: scaffold_47 222976\n",
      "ignoring indels: scaffold_47 2235542\n",
      "ignoring indels: scaffold_47 2846929\n",
      "ignoring > bi-allelic SNPs: scaffold_47 2861060\n",
      "ignoring indels: scaffold_47 2869111\n",
      "ignoring indels: scaffold_47 3122361\n",
      "ignoring > bi-allelic SNPs: scaffold_47 3754723\n",
      "ignoring indels: scaffold_47 4061354\n",
      "ignoring indels: scaffold_47 4061447\n",
      "ignoring indels: scaffold_47 4120942\n",
      "ignoring indels: scaffold_47 4126333\n",
      "ignoring indels: scaffold_48 102772\n",
      "something is wrong with the reference allele at scaffold_48 171811: GT\n",
      "ignoring indels: scaffold_48 171836\n",
      "ignoring > bi-allelic SNPs: scaffold_48 557679\n",
      "ignoring > bi-allelic SNPs: scaffold_48 686386\n",
      "ignoring indels: scaffold_48 1695657\n",
      "ignoring > bi-allelic SNPs: scaffold_48 1752625\n",
      "ignoring > bi-allelic SNPs: scaffold_48 1773670\n",
      "ignoring indels: scaffold_48 1872996\n",
      "ignoring indels: scaffold_48 1901666\n",
      "ignoring > bi-allelic SNPs: scaffold_48 2486254\n",
      "ignoring > bi-allelic SNPs: scaffold_48 2674369\n",
      "ignoring indels: scaffold_48 3607561\n",
      "ignoring indels: scaffold_49 381210\n",
      "ignoring indels: scaffold_49 615052\n",
      "ignoring > bi-allelic SNPs: scaffold_49 828452\n",
      "ignoring > bi-allelic SNPs: scaffold_49 828455\n",
      "ignoring indels: scaffold_49 874200\n",
      "ignoring indels: scaffold_49 874243\n",
      "ignoring indels: scaffold_49 1468985\n",
      "ignoring > bi-allelic SNPs: scaffold_49 1496432\n",
      "ignoring indels: scaffold_49 1496445\n",
      "ignoring indels: scaffold_49 2495141\n",
      "ignoring > bi-allelic SNPs: scaffold_49 2680137\n",
      "ignoring indels: scaffold_49 2680181\n",
      "ignoring indels: scaffold_49 3491883\n",
      "ignoring > bi-allelic SNPs: scaffold_49 4067547\n",
      "ignoring indels: scaffold_50 60128\n",
      "ignoring > bi-allelic SNPs: scaffold_50 141647\n",
      "ignoring > bi-allelic SNPs: scaffold_50 423364\n",
      "ignoring > bi-allelic SNPs: scaffold_50 1598954\n",
      "ignoring > bi-allelic SNPs: scaffold_50 1598964\n",
      "ignoring > bi-allelic SNPs: scaffold_50 3247130\n",
      "ignoring > bi-allelic SNPs: scaffold_50 3479567\n",
      "ignoring indels: scaffold_50 3840587\n",
      "ignoring > bi-allelic SNPs: scaffold_51 1287903\n",
      "ignoring indels: scaffold_51 1707338\n",
      "ignoring indels: scaffold_51 1707351\n",
      "ignoring indels: scaffold_51 2070801\n",
      "ignoring indels: scaffold_51 2133067\n",
      "ignoring > bi-allelic SNPs: scaffold_51 3416434\n",
      "ignoring > bi-allelic SNPs: scaffold_51 3698804\n",
      "ignoring > bi-allelic SNPs: scaffold_51 3901633\n",
      "ignoring > bi-allelic SNPs: scaffold_51 4082687\n",
      "ignoring > bi-allelic SNPs: scaffold_52 21726\n",
      "ignoring > bi-allelic SNPs: scaffold_52 2001794\n",
      "ignoring > bi-allelic SNPs: scaffold_52 2136369\n",
      "ignoring indels: scaffold_52 2141829\n",
      "ignoring > bi-allelic SNPs: scaffold_52 2141862\n",
      "ignoring indels: scaffold_52 2955961\n",
      "ignoring > bi-allelic SNPs: scaffold_52 3757859\n",
      "ignoring indels: scaffold_52 3757915\n",
      "ignoring indels: scaffold_52 3773550\n",
      "ignoring > bi-allelic SNPs: scaffold_53 1449618\n",
      "ignoring > bi-allelic SNPs: scaffold_53 1772212\n",
      "ignoring > bi-allelic SNPs: scaffold_53 2014823\n",
      "ignoring > bi-allelic SNPs: scaffold_53 2014830\n",
      "ignoring indels: scaffold_53 2014845\n",
      "ignoring indels: scaffold_53 2014858\n",
      "ignoring > bi-allelic SNPs: scaffold_53 2305458\n",
      "ignoring indels: scaffold_53 2305494\n",
      "ignoring > bi-allelic SNPs: scaffold_54 9974\n",
      "ignoring > bi-allelic SNPs: scaffold_54 2249638\n",
      "ignoring indels: scaffold_54 2249765\n",
      "ignoring > bi-allelic SNPs: scaffold_54 2427932\n",
      "ignoring > bi-allelic SNPs: scaffold_54 2486019\n",
      "ignoring indels: scaffold_55 277797\n",
      "ignoring > bi-allelic SNPs: scaffold_55 333218\n",
      "ignoring indels: scaffold_55 405016\n",
      "ignoring > bi-allelic SNPs: scaffold_55 489791\n",
      "ignoring > bi-allelic SNPs: scaffold_55 900881\n",
      "ignoring indels: scaffold_55 1867408\n",
      "ignoring indels: scaffold_55 1867431\n",
      "ignoring > bi-allelic SNPs: scaffold_55 2387545\n",
      "ignoring > bi-allelic SNPs: scaffold_55 3244910\n",
      "ignoring indels: scaffold_55 3280421\n",
      "ignoring indels: scaffold_56 305183\n",
      "ignoring indels: scaffold_56 305233\n",
      "ignoring > bi-allelic SNPs: scaffold_56 1657165\n",
      "ignoring > bi-allelic SNPs: scaffold_56 1657166\n",
      "ignoring > bi-allelic SNPs: scaffold_56 1657167\n",
      "ignoring > bi-allelic SNPs: scaffold_56 1657168\n",
      "ignoring > bi-allelic SNPs: scaffold_56 1657170\n",
      "ignoring > bi-allelic SNPs: scaffold_56 1657171\n",
      "ignoring > bi-allelic SNPs: scaffold_56 2289867\n",
      "ignoring indels: scaffold_56 2442135\n",
      "ignoring indels: scaffold_56 2515136\n",
      "ignoring indels: scaffold_56 2928122\n",
      "ignoring indels: scaffold_56 3021741\n",
      "ignoring indels: scaffold_56 3104363\n",
      "ignoring indels: scaffold_56 3282325\n",
      "ignoring > bi-allelic SNPs: scaffold_57 193880\n",
      "ignoring indels: scaffold_57 288475\n",
      "ignoring > bi-allelic SNPs: scaffold_57 445605\n",
      "ignoring > bi-allelic SNPs: scaffold_57 445606\n",
      "ignoring > bi-allelic SNPs: scaffold_57 445607\n",
      "ignoring > bi-allelic SNPs: scaffold_57 445608\n",
      "ignoring > bi-allelic SNPs: scaffold_57 445609\n",
      "ignoring > bi-allelic SNPs: scaffold_57 445611\n",
      "ignoring indels: scaffold_57 556817\n",
      "ignoring > bi-allelic SNPs: scaffold_57 556834\n",
      "ignoring indels: scaffold_57 1064071\n",
      "ignoring indels: scaffold_57 1665027\n",
      "ignoring > bi-allelic SNPs: scaffold_57 1665030\n",
      "ignoring > bi-allelic SNPs: scaffold_57 1723676\n",
      "ignoring indels: scaffold_57 2170280\n",
      "ignoring indels: scaffold_57 2261280\n",
      "ignoring indels: scaffold_57 3037407\n",
      "ignoring indels: scaffold_57 3589582\n",
      "ignoring indels: scaffold_58 599070\n",
      "ignoring indels: scaffold_58 762434\n",
      "ignoring indels: scaffold_58 1183353\n",
      "ignoring > bi-allelic SNPs: scaffold_58 1478866\n",
      "ignoring indels: scaffold_58 1634052\n",
      "ignoring > bi-allelic SNPs: scaffold_58 2435001\n",
      "ignoring > bi-allelic SNPs: scaffold_58 2435047\n",
      "ignoring indels: scaffold_58 2516689\n",
      "ignoring indels: scaffold_58 2849541\n",
      "ignoring indels: scaffold_58 2861345\n",
      "ignoring > bi-allelic SNPs: scaffold_58 3143230\n",
      "ignoring > bi-allelic SNPs: scaffold_58 3260213\n",
      "ignoring indels: scaffold_58 3553492\n",
      "ignoring indels: scaffold_58 3754282\n",
      "ignoring indels: scaffold_59 566261\n",
      "ignoring > bi-allelic SNPs: scaffold_59 1031133\n",
      "ignoring > bi-allelic SNPs: scaffold_59 1052263\n",
      "ignoring indels: scaffold_59 2818678\n",
      "ignoring indels: scaffold_59 2818730\n",
      "ignoring indels: scaffold_59 3586255\n",
      "ignoring > bi-allelic SNPs: scaffold_59 3740805\n",
      "ignoring indels: scaffold_60 1253928\n",
      "ignoring indels: scaffold_60 1765232\n",
      "ignoring indels: scaffold_60 1924590\n",
      "ignoring > bi-allelic SNPs: scaffold_60 2031886\n",
      "ignoring indels: scaffold_60 2324062\n",
      "ignoring indels: scaffold_60 2324067\n",
      "ignoring indels: scaffold_60 2350155\n",
      "ignoring indels: scaffold_60 2363088\n",
      "ignoring indels: scaffold_60 3623382\n",
      "ignoring > bi-allelic SNPs: scaffold_61 113309\n",
      "ignoring indels: scaffold_61 440979\n",
      "ignoring > bi-allelic SNPs: scaffold_61 522275\n",
      "ignoring indels: scaffold_61 1611650\n",
      "ignoring > bi-allelic SNPs: scaffold_61 1774647\n",
      "ignoring > bi-allelic SNPs: scaffold_61 1833183\n",
      "ignoring > bi-allelic SNPs: scaffold_61 1833212\n",
      "ignoring > bi-allelic SNPs: scaffold_61 1833217\n",
      "ignoring > bi-allelic SNPs: scaffold_61 1833218\n",
      "ignoring > bi-allelic SNPs: scaffold_61 1833219\n",
      "ignoring > bi-allelic SNPs: scaffold_61 1833220\n",
      "ignoring indels: scaffold_61 2109423\n",
      "ignoring > bi-allelic SNPs: scaffold_61 2128458\n",
      "ignoring > bi-allelic SNPs: scaffold_61 2259336\n",
      "ignoring > bi-allelic SNPs: scaffold_61 2259338\n",
      "ignoring > bi-allelic SNPs: scaffold_61 2259377\n",
      "ignoring > bi-allelic SNPs: scaffold_61 2259428\n",
      "ignoring > bi-allelic SNPs: scaffold_61 2259467\n",
      "ignoring > bi-allelic SNPs: scaffold_61 2259495\n",
      "ignoring > bi-allelic SNPs: scaffold_61 2290494\n",
      "ignoring > bi-allelic SNPs: scaffold_61 2290554\n",
      "ignoring > bi-allelic SNPs: scaffold_61 2402862\n",
      "ignoring indels: scaffold_61 2650878\n",
      "ignoring indels: scaffold_61 2732521\n",
      "ignoring > bi-allelic SNPs: scaffold_61 2781020\n",
      "ignoring > bi-allelic SNPs: scaffold_61 3058305\n",
      "ignoring > bi-allelic SNPs: scaffold_62 329148\n",
      "ignoring indels: scaffold_62 380098\n",
      "ignoring indels: scaffold_62 1122169\n",
      "ignoring indels: scaffold_62 1223968\n",
      "ignoring > bi-allelic SNPs: scaffold_62 3539631\n",
      "ignoring indels: scaffold_63 432385\n",
      "ignoring > bi-allelic SNPs: scaffold_63 1130091\n",
      "ignoring > bi-allelic SNPs: scaffold_63 1475426\n",
      "ignoring indels: scaffold_63 1475440\n",
      "ignoring > bi-allelic SNPs: scaffold_63 1475442\n",
      "ignoring indels: scaffold_63 2201156\n",
      "ignoring indels: scaffold_63 2344701\n",
      "ignoring indels: scaffold_63 2344741\n",
      "ignoring > bi-allelic SNPs: scaffold_63 2973711\n",
      "ignoring > bi-allelic SNPs: scaffold_63 3325735\n",
      "ignoring > bi-allelic SNPs: scaffold_63 3325741\n",
      "ignoring indels: scaffold_63 3486476\n",
      "ignoring indels: scaffold_64 194311\n",
      "ignoring indels: scaffold_64 1049241\n",
      "ignoring indels: scaffold_64 1064281\n",
      "ignoring > bi-allelic SNPs: scaffold_64 1064314\n",
      "ignoring > bi-allelic SNPs: scaffold_64 1064384\n",
      "ignoring indels: scaffold_64 1270201\n",
      "ignoring indels: scaffold_64 1482171\n",
      "ignoring indels: scaffold_64 3191615\n",
      "ignoring > bi-allelic SNPs: scaffold_64 3191633\n",
      "ignoring > bi-allelic SNPs: scaffold_65 606033\n",
      "ignoring > bi-allelic SNPs: scaffold_65 606086\n",
      "ignoring > bi-allelic SNPs: scaffold_65 1071877\n",
      "ignoring > bi-allelic SNPs: scaffold_65 1454245\n",
      "ignoring indels: scaffold_65 1750804\n",
      "ignoring indels: scaffold_65 1750814\n",
      "ignoring > bi-allelic SNPs: scaffold_65 2756690\n",
      "ignoring indels: scaffold_66 749461\n",
      "ignoring indels: scaffold_66 1941416\n",
      "ignoring > bi-allelic SNPs: scaffold_66 1999892\n",
      "ignoring indels: scaffold_66 3320686\n",
      "ignoring > bi-allelic SNPs: scaffold_67 266430\n",
      "ignoring indels: scaffold_67 792400\n",
      "ignoring indels: scaffold_67 1129932\n",
      "ignoring indels: scaffold_67 1634875\n",
      "ignoring > bi-allelic SNPs: scaffold_67 2734966\n",
      "ignoring > bi-allelic SNPs: scaffold_67 2910783\n",
      "ignoring > bi-allelic SNPs: scaffold_67 3100958\n",
      "ignoring indels: scaffold_68 230878\n",
      "ignoring > bi-allelic SNPs: scaffold_68 340322\n",
      "ignoring indels: scaffold_68 569077\n",
      "ignoring indels: scaffold_68 582950\n",
      "ignoring > bi-allelic SNPs: scaffold_68 582959\n",
      "ignoring indels: scaffold_68 734075\n",
      "ignoring indels: scaffold_68 1359932\n",
      "ignoring > bi-allelic SNPs: scaffold_68 1573407\n",
      "ignoring indels: scaffold_68 1847370\n",
      "ignoring > bi-allelic SNPs: scaffold_68 1847408\n",
      "ignoring > bi-allelic SNPs: scaffold_68 3139773\n",
      "ignoring > bi-allelic SNPs: scaffold_68 3139777\n",
      "ignoring indels: scaffold_68 3283565\n",
      "ignoring > bi-allelic SNPs: scaffold_69 325497\n",
      "ignoring indels: scaffold_69 468055\n",
      "ignoring indels: scaffold_69 1822920\n",
      "ignoring indels: scaffold_69 2243242\n",
      "ignoring indels: scaffold_69 2243244\n",
      "ignoring > bi-allelic SNPs: scaffold_69 2429468\n",
      "ignoring > bi-allelic SNPs: scaffold_69 2656804\n",
      "ignoring indels: scaffold_70 388829\n",
      "ignoring > bi-allelic SNPs: scaffold_70 457826\n",
      "ignoring indels: scaffold_70 457882\n",
      "ignoring indels: scaffold_70 457886\n",
      "ignoring indels: scaffold_70 2226685\n",
      "ignoring indels: scaffold_71 1452584\n",
      "ignoring > bi-allelic SNPs: scaffold_71 1459507\n",
      "ignoring indels: scaffold_71 1481999\n",
      "ignoring indels: scaffold_71 1511632\n",
      "ignoring > bi-allelic SNPs: scaffold_71 1835949\n",
      "ignoring indels: scaffold_71 3225748\n",
      "ignoring indels: scaffold_72 647696\n",
      "ignoring indels: scaffold_72 1037046\n",
      "ignoring > bi-allelic SNPs: scaffold_72 1084305\n",
      "ignoring indels: scaffold_72 1186245\n",
      "ignoring indels: scaffold_72 2214414\n",
      "ignoring indels: scaffold_72 2421326\n",
      "ignoring > bi-allelic SNPs: scaffold_72 2453282\n",
      "ignoring indels: scaffold_72 2705588\n",
      "ignoring indels: scaffold_72 2705721\n",
      "ignoring indels: scaffold_72 2864335\n",
      "ignoring > bi-allelic SNPs: scaffold_73 628283\n",
      "ignoring > bi-allelic SNPs: scaffold_73 1402382\n",
      "ignoring indels: scaffold_73 1977124\n",
      "ignoring indels: scaffold_73 1983771\n",
      "ignoring > bi-allelic SNPs: scaffold_73 1983773\n",
      "ignoring indels: scaffold_73 2163537\n",
      "ignoring indels: scaffold_73 2355216\n",
      "ignoring > bi-allelic SNPs: scaffold_73 2451683\n",
      "ignoring indels: scaffold_73 2646286\n",
      "ignoring indels: scaffold_73 2693406\n",
      "ignoring indels: scaffold_73 2906737\n",
      "ignoring > bi-allelic SNPs: scaffold_74 464836\n",
      "ignoring indels: scaffold_74 1287387\n",
      "ignoring indels: scaffold_74 1315324\n",
      "ignoring indels: scaffold_74 1403834\n",
      "ignoring indels: scaffold_74 1438088\n",
      "ignoring > bi-allelic SNPs: scaffold_74 1532917\n",
      "ignoring indels: scaffold_74 1569141\n",
      "ignoring indels: scaffold_74 1569144\n",
      "ignoring indels: scaffold_74 2020350\n",
      "ignoring indels: scaffold_75 160323\n",
      "ignoring indels: scaffold_75 1371665\n",
      "ignoring indels: scaffold_75 1391992\n",
      "ignoring indels: scaffold_75 1597837\n",
      "ignoring indels: scaffold_75 1831530\n",
      "ignoring indels: scaffold_75 2279540\n",
      "ignoring indels: scaffold_75 2417331\n",
      "ignoring > bi-allelic SNPs: scaffold_75 2468178\n",
      "ignoring indels: scaffold_75 2661525\n",
      "ignoring indels: scaffold_76 136212\n",
      "ignoring > bi-allelic SNPs: scaffold_76 515617\n",
      "ignoring indels: scaffold_76 858873\n",
      "ignoring indels: scaffold_76 1028134\n",
      "ignoring > bi-allelic SNPs: scaffold_76 1161786\n",
      "ignoring indels: scaffold_76 1322300\n",
      "ignoring indels: scaffold_76 1878366\n",
      "ignoring > bi-allelic SNPs: scaffold_76 2023804\n",
      "ignoring indels: scaffold_76 2304615\n",
      "ignoring indels: scaffold_76 2333536\n",
      "ignoring indels: scaffold_76 2406626\n",
      "ignoring > bi-allelic SNPs: scaffold_76 2647383\n",
      "ignoring indels: scaffold_76 2647395\n",
      "ignoring indels: scaffold_76 2647437\n",
      "ignoring > bi-allelic SNPs: scaffold_77 679320\n",
      "ignoring indels: scaffold_78 249313\n",
      "ignoring indels: scaffold_78 249328\n",
      "ignoring indels: scaffold_78 343054\n",
      "ignoring indels: scaffold_78 770240\n",
      "ignoring > bi-allelic SNPs: scaffold_78 1721287\n",
      "ignoring > bi-allelic SNPs: scaffold_79 160277\n",
      "ignoring indels: scaffold_79 239775\n",
      "ignoring > bi-allelic SNPs: scaffold_79 617137\n",
      "ignoring > bi-allelic SNPs: scaffold_79 1384093\n",
      "ignoring indels: scaffold_79 1769358\n",
      "ignoring > bi-allelic SNPs: scaffold_79 2527373\n",
      "ignoring indels: scaffold_79 2527390\n",
      "ignoring > bi-allelic SNPs: scaffold_79 2527633\n",
      "ignoring indels: scaffold_80 835381\n",
      "ignoring indels: scaffold_80 912745\n",
      "ignoring indels: scaffold_81 109843\n",
      "ignoring indels: scaffold_81 304009\n",
      "ignoring indels: scaffold_81 316284\n",
      "ignoring > bi-allelic SNPs: scaffold_81 335631\n",
      "ignoring indels: scaffold_81 791968\n",
      "ignoring indels: scaffold_81 1151825\n",
      "ignoring > bi-allelic SNPs: scaffold_81 1151829\n",
      "ignoring indels: scaffold_81 1265462\n",
      "ignoring indels: scaffold_81 2334420\n",
      "ignoring indels: scaffold_81 2491874\n",
      "ignoring > bi-allelic SNPs: scaffold_81 2880267\n",
      "ignoring indels: scaffold_82 11621\n",
      "ignoring > bi-allelic SNPs: scaffold_82 52662\n",
      "ignoring indels: scaffold_82 292685\n",
      "ignoring > bi-allelic SNPs: scaffold_82 292689\n",
      "something is wrong with the reference allele at scaffold_82 1614974: AT\n",
      "ignoring indels: scaffold_82 1853939\n",
      "ignoring indels: scaffold_82 2813413\n",
      "ignoring > bi-allelic SNPs: scaffold_82 2861943\n",
      "ignoring > bi-allelic SNPs: scaffold_82 2861944\n",
      "ignoring > bi-allelic SNPs: scaffold_82 2861947\n",
      "ignoring indels: scaffold_82 2981180\n",
      "ignoring > bi-allelic SNPs: scaffold_83 1060\n",
      "ignoring indels: scaffold_83 192501\n",
      "ignoring > bi-allelic SNPs: scaffold_83 505130\n",
      "ignoring > bi-allelic SNPs: scaffold_83 524643\n",
      "ignoring > bi-allelic SNPs: scaffold_83 706628\n",
      "ignoring > bi-allelic SNPs: scaffold_83 1035158\n",
      "ignoring indels: scaffold_83 2258553\n",
      "ignoring indels: scaffold_83 2384122\n",
      "ignoring indels: scaffold_83 2384124\n",
      "ignoring indels: scaffold_83 2394964\n",
      "ignoring indels: scaffold_83 2395254\n",
      "ignoring indels: scaffold_83 2449837\n",
      "ignoring indels: scaffold_83 2777247\n",
      "ignoring indels: scaffold_83 2817946\n",
      "ignoring indels: scaffold_84 146\n",
      "ignoring indels: scaffold_84 188786\n",
      "ignoring > bi-allelic SNPs: scaffold_84 749325\n",
      "ignoring indels: scaffold_84 749351\n",
      "ignoring > bi-allelic SNPs: scaffold_84 1298209\n",
      "ignoring > bi-allelic SNPs: scaffold_84 1298249\n",
      "ignoring > bi-allelic SNPs: scaffold_84 1982758\n",
      "ignoring indels: scaffold_84 2368274\n",
      "ignoring indels: scaffold_84 2950482\n",
      "ignoring indels: scaffold_84 2985082\n",
      "ignoring > bi-allelic SNPs: scaffold_85 887489\n",
      "ignoring indels: scaffold_85 1716810\n",
      "ignoring indels: scaffold_85 1793146\n",
      "ignoring indels: scaffold_85 2143689\n",
      "ignoring indels: scaffold_85 2519948\n",
      "ignoring indels: scaffold_85 2700534\n",
      "ignoring indels: scaffold_86 95422\n",
      "ignoring > bi-allelic SNPs: scaffold_86 842588\n",
      "ignoring indels: scaffold_86 929270\n",
      "ignoring indels: scaffold_86 1212951\n",
      "ignoring indels: scaffold_86 2073600\n",
      "ignoring indels: scaffold_86 2349104\n",
      "ignoring indels: scaffold_87 1279792\n",
      "ignoring indels: scaffold_87 1279849\n",
      "ignoring > bi-allelic SNPs: scaffold_87 2139581\n",
      "ignoring indels: scaffold_87 2258642\n",
      "ignoring indels: scaffold_87 2371386\n",
      "ignoring indels: scaffold_87 2371397\n",
      "ignoring indels: scaffold_87 2371589\n",
      "ignoring > bi-allelic SNPs: scaffold_87 2723401\n",
      "ignoring indels: scaffold_88 258357\n",
      "ignoring indels: scaffold_88 673180\n",
      "ignoring indels: scaffold_89 171792\n",
      "ignoring > bi-allelic SNPs: scaffold_89 697290\n",
      "ignoring indels: scaffold_89 1049353\n",
      "ignoring indels: scaffold_89 2227969\n",
      "ignoring indels: scaffold_89 2271372\n",
      "ignoring > bi-allelic SNPs: scaffold_90 490075\n",
      "ignoring indels: scaffold_90 510810\n",
      "ignoring indels: scaffold_90 1296845\n",
      "ignoring indels: scaffold_90 1675984\n",
      "ignoring > bi-allelic SNPs: scaffold_90 2251729\n",
      "ignoring > bi-allelic SNPs: scaffold_91 323635\n",
      "ignoring indels: scaffold_91 2092641\n",
      "ignoring indels: scaffold_92 437666\n",
      "ignoring indels: scaffold_92 2422340\n",
      "ignoring indels: scaffold_93 226791\n",
      "ignoring indels: scaffold_93 675462\n",
      "ignoring indels: scaffold_93 1013354\n",
      "ignoring > bi-allelic SNPs: scaffold_93 2538678\n",
      "ignoring > bi-allelic SNPs: scaffold_94 458533\n",
      "ignoring > bi-allelic SNPs: scaffold_94 1052081\n",
      "ignoring > bi-allelic SNPs: scaffold_94 1052087\n",
      "ignoring indels: scaffold_94 1052097\n",
      "ignoring indels: scaffold_94 2038521\n",
      "ignoring indels: scaffold_94 2628455\n",
      "ignoring indels: scaffold_94 2628478\n",
      "ignoring indels: scaffold_94 2722847\n",
      "ignoring indels: scaffold_95 930857\n",
      "ignoring indels: scaffold_95 1429577\n",
      "ignoring > bi-allelic SNPs: scaffold_95 2253729\n",
      "ignoring indels: scaffold_96 786618\n",
      "ignoring indels: scaffold_96 1243727\n",
      "ignoring > bi-allelic SNPs: scaffold_96 1243741\n",
      "ignoring indels: scaffold_96 2042856\n",
      "ignoring indels: scaffold_96 2419494\n",
      "something is wrong with the reference allele at scaffold_97 450019: GA\n",
      "ignoring indels: scaffold_97 2078149\n",
      "ignoring indels: scaffold_97 2209659\n",
      "ignoring indels: scaffold_97 2516601\n",
      "ignoring indels: scaffold_97 2524894\n",
      "ignoring indels: scaffold_97 2554986\n",
      "ignoring indels: scaffold_97 2610494\n",
      "ignoring indels: scaffold_98 1100368\n",
      "ignoring indels: scaffold_99 162576\n",
      "ignoring > bi-allelic SNPs: scaffold_99 813860\n",
      "ignoring indels: scaffold_99 1686730\n",
      "ignoring indels: scaffold_99 2549611\n",
      "ignoring > bi-allelic SNPs: scaffold_100 460544\n",
      "ignoring indels: scaffold_100 1827468\n",
      "ignoring indels: scaffold_100 2132728\n",
      "ignoring > bi-allelic SNPs: scaffold_101 569604\n",
      "ignoring indels: scaffold_101 1004728\n",
      "ignoring indels: scaffold_101 2300915\n",
      "ignoring > bi-allelic SNPs: scaffold_101 2322038\n",
      "ignoring indels: scaffold_101 2396811\n",
      "ignoring indels: scaffold_102 1066776\n",
      "ignoring indels: scaffold_102 1114173\n",
      "ignoring indels: scaffold_102 1152292\n",
      "ignoring > bi-allelic SNPs: scaffold_102 1152420\n",
      "ignoring indels: scaffold_102 1562976\n",
      "ignoring indels: scaffold_103 144042\n",
      "ignoring indels: scaffold_103 388529\n",
      "ignoring indels: scaffold_103 388553\n",
      "ignoring indels: scaffold_103 579357\n",
      "ignoring indels: scaffold_103 1904254\n",
      "ignoring indels: scaffold_104 1799258\n",
      "ignoring indels: scaffold_104 2211402\n",
      "ignoring indels: scaffold_105 729515\n",
      "ignoring indels: scaffold_105 1219140\n",
      "ignoring > bi-allelic SNPs: scaffold_105 1526135\n",
      "ignoring > bi-allelic SNPs: scaffold_105 1706055\n",
      "ignoring indels: scaffold_105 2198507\n",
      "ignoring indels: scaffold_105 2382349\n",
      "ignoring indels: scaffold_106 135563\n",
      "ignoring > bi-allelic SNPs: scaffold_106 714371\n",
      "ignoring indels: scaffold_106 841057\n",
      "ignoring indels: scaffold_106 933130\n",
      "ignoring > bi-allelic SNPs: scaffold_106 1051281\n",
      "ignoring indels: scaffold_106 1685093\n",
      "ignoring indels: scaffold_106 1750673\n",
      "ignoring indels: scaffold_106 2416256\n",
      "ignoring indels: scaffold_106 2432038\n",
      "ignoring indels: scaffold_106 2443079\n",
      "ignoring > bi-allelic SNPs: scaffold_107 1957734\n",
      "ignoring indels: scaffold_107 2020163\n",
      "ignoring > bi-allelic SNPs: scaffold_108 41327\n",
      "ignoring indels: scaffold_108 1470074\n",
      "ignoring indels: scaffold_109 88353\n",
      "ignoring indels: scaffold_109 186768\n",
      "ignoring > bi-allelic SNPs: scaffold_109 1076944\n",
      "ignoring > bi-allelic SNPs: scaffold_109 1120972\n",
      "ignoring indels: scaffold_109 1225119\n",
      "ignoring > bi-allelic SNPs: scaffold_110 452587\n",
      "ignoring > bi-allelic SNPs: scaffold_110 1483144\n",
      "ignoring indels: scaffold_111 1108240\n",
      "ignoring indels: scaffold_111 1708826\n",
      "ignoring indels: scaffold_111 1877151\n",
      "ignoring indels: scaffold_111 1878999\n",
      "ignoring > bi-allelic SNPs: scaffold_111 2357345\n",
      "ignoring indels: scaffold_112 20924\n",
      "ignoring > bi-allelic SNPs: scaffold_112 910406\n",
      "ignoring indels: scaffold_112 983001\n",
      "ignoring > bi-allelic SNPs: scaffold_113 658753\n",
      "ignoring indels: scaffold_113 818048\n",
      "ignoring indels: scaffold_113 1102720\n",
      "ignoring indels: scaffold_113 1102835\n",
      "ignoring indels: scaffold_113 1511966\n",
      "ignoring > bi-allelic SNPs: scaffold_113 1511969\n",
      "ignoring > bi-allelic SNPs: scaffold_113 1522563\n",
      "ignoring indels: scaffold_113 2046038\n",
      "ignoring > bi-allelic SNPs: scaffold_113 2055849\n",
      "ignoring indels: scaffold_113 2055871\n",
      "ignoring indels: scaffold_113 2178878\n",
      "ignoring indels: scaffold_114 317860\n",
      "ignoring indels: scaffold_114 341785\n",
      "ignoring indels: scaffold_114 391650\n",
      "ignoring indels: scaffold_114 542468\n",
      "ignoring indels: scaffold_114 690743\n",
      "ignoring > bi-allelic SNPs: scaffold_114 759037\n",
      "ignoring indels: scaffold_114 1532678\n",
      "ignoring indels: scaffold_114 1587989\n",
      "ignoring indels: scaffold_114 2036159\n",
      "ignoring indels: scaffold_114 2156118\n",
      "ignoring indels: scaffold_114 2353355\n",
      "ignoring > bi-allelic SNPs: scaffold_115 509484\n",
      "ignoring indels: scaffold_115 609678\n",
      "ignoring indels: scaffold_115 977663\n",
      "ignoring indels: scaffold_115 1192058\n",
      "ignoring > bi-allelic SNPs: scaffold_115 1907416\n",
      "ignoring > bi-allelic SNPs: scaffold_115 2110459\n",
      "ignoring indels: scaffold_115 2263546\n",
      "ignoring indels: scaffold_116 185308\n",
      "ignoring indels: scaffold_116 1068925\n",
      "ignoring indels: scaffold_116 1700162\n",
      "ignoring indels: scaffold_117 69680\n",
      "ignoring indels: scaffold_118 447570\n",
      "ignoring > bi-allelic SNPs: scaffold_118 455074\n",
      "ignoring > bi-allelic SNPs: scaffold_118 2038431\n",
      "ignoring > bi-allelic SNPs: scaffold_119 421420\n",
      "ignoring > bi-allelic SNPs: scaffold_119 622329\n",
      "ignoring indels: scaffold_119 793585\n",
      "ignoring > bi-allelic SNPs: scaffold_119 852541\n",
      "ignoring indels: scaffold_119 922256\n",
      "ignoring indels: scaffold_119 1711749\n",
      "ignoring indels: scaffold_119 2149278\n",
      "ignoring indels: scaffold_120 160619\n",
      "ignoring > bi-allelic SNPs: scaffold_120 703649\n",
      "ignoring indels: scaffold_120 971275\n",
      "ignoring indels: scaffold_120 971282\n",
      "ignoring indels: scaffold_120 1096599\n",
      "ignoring indels: scaffold_120 2053601\n",
      "ignoring indels: scaffold_120 2211158\n",
      "ignoring > bi-allelic SNPs: scaffold_121 387881\n",
      "ignoring > bi-allelic SNPs: scaffold_121 387882\n",
      "ignoring > bi-allelic SNPs: scaffold_121 387899\n",
      "ignoring > bi-allelic SNPs: scaffold_121 387912\n",
      "ignoring indels: scaffold_121 541164\n",
      "ignoring indels: scaffold_121 757406\n",
      "ignoring indels: scaffold_121 944257\n",
      "ignoring indels: scaffold_121 944264\n",
      "ignoring > bi-allelic SNPs: scaffold_121 1925805\n",
      "ignoring > bi-allelic SNPs: scaffold_121 1925838\n",
      "ignoring indels: scaffold_122 146365\n",
      "ignoring indels: scaffold_122 989713\n",
      "ignoring indels: scaffold_122 1258291\n",
      "ignoring indels: scaffold_122 1482197\n",
      "ignoring indels: scaffold_123 414156\n",
      "ignoring indels: scaffold_123 1469319\n",
      "ignoring indels: scaffold_124 79528\n",
      "ignoring > bi-allelic SNPs: scaffold_124 136075\n",
      "ignoring indels: scaffold_124 865935\n",
      "ignoring indels: scaffold_125 29141\n",
      "ignoring indels: scaffold_125 1219288\n",
      "ignoring > bi-allelic SNPs: scaffold_125 1356706\n",
      "ignoring indels: scaffold_126 547846\n",
      "ignoring > bi-allelic SNPs: scaffold_126 547953\n",
      "ignoring > bi-allelic SNPs: scaffold_126 696142\n",
      "ignoring > bi-allelic SNPs: scaffold_126 839098\n",
      "ignoring indels: scaffold_126 839118\n",
      "ignoring > bi-allelic SNPs: scaffold_126 1098498\n",
      "ignoring > bi-allelic SNPs: scaffold_126 1153804\n",
      "ignoring indels: scaffold_126 1195716\n",
      "ignoring > bi-allelic SNPs: scaffold_126 1258100\n",
      "ignoring indels: scaffold_126 1704965\n",
      "ignoring indels: scaffold_126 2050538\n",
      "ignoring > bi-allelic SNPs: scaffold_127 664346\n",
      "ignoring indels: scaffold_127 664371\n",
      "ignoring indels: scaffold_127 701480\n",
      "ignoring indels: scaffold_127 1892743\n",
      "ignoring indels: scaffold_128 243327\n",
      "ignoring indels: scaffold_128 509916\n",
      "ignoring indels: scaffold_129 219680\n",
      "ignoring indels: scaffold_129 933261\n",
      "ignoring > bi-allelic SNPs: scaffold_129 1105685\n",
      "ignoring indels: scaffold_129 1742744\n",
      "ignoring indels: scaffold_130 228809\n",
      "ignoring > bi-allelic SNPs: scaffold_130 375838\n",
      "ignoring > bi-allelic SNPs: scaffold_130 878474\n",
      "ignoring indels: scaffold_130 917412\n",
      "ignoring indels: scaffold_131 1004601\n",
      "ignoring > bi-allelic SNPs: scaffold_131 1151041\n",
      "ignoring indels: scaffold_131 1241775\n",
      "something is wrong with the reference allele at scaffold_131 1447049: CG\n",
      "ignoring indels: scaffold_132 1343329\n",
      "ignoring indels: scaffold_132 1684204\n",
      "ignoring > bi-allelic SNPs: scaffold_132 1685036\n",
      "ignoring indels: scaffold_132 1685102\n",
      "ignoring indels: scaffold_132 1920647\n",
      "ignoring indels: scaffold_133 194126\n",
      "ignoring indels: scaffold_133 215250\n",
      "ignoring indels: scaffold_133 590757\n",
      "ignoring indels: scaffold_133 973140\n",
      "ignoring > bi-allelic SNPs: scaffold_133 1109605\n",
      "ignoring > bi-allelic SNPs: scaffold_134 31633\n",
      "ignoring indels: scaffold_134 1236934\n",
      "ignoring indels: scaffold_134 1236945\n",
      "ignoring indels: scaffold_135 62365\n",
      "ignoring > bi-allelic SNPs: scaffold_135 302199\n",
      "ignoring > bi-allelic SNPs: scaffold_135 1911657\n",
      "ignoring indels: scaffold_136 317208\n",
      "ignoring indels: scaffold_136 487559\n",
      "ignoring indels: scaffold_136 996387\n",
      "ignoring indels: scaffold_136 1803883\n",
      "ignoring indels: scaffold_137 1055200\n",
      "ignoring indels: scaffold_137 1800744\n",
      "ignoring > bi-allelic SNPs: scaffold_138 375862\n",
      "ignoring indels: scaffold_138 432738\n",
      "ignoring indels: scaffold_138 712166\n",
      "ignoring indels: scaffold_138 764148\n",
      "ignoring indels: scaffold_138 1106979\n",
      "ignoring indels: scaffold_138 1106986\n",
      "ignoring > bi-allelic SNPs: scaffold_138 1239377\n",
      "ignoring indels: scaffold_139 597429\n",
      "ignoring > bi-allelic SNPs: scaffold_139 1053713\n",
      "ignoring > bi-allelic SNPs: scaffold_139 1365478\n",
      "ignoring indels: scaffold_140 95228\n",
      "ignoring indels: scaffold_140 862506\n",
      "ignoring indels: scaffold_140 862514\n",
      "ignoring > bi-allelic SNPs: scaffold_140 902869\n",
      "ignoring indels: scaffold_141 171880\n",
      "ignoring indels: scaffold_141 306788\n",
      "ignoring > bi-allelic SNPs: scaffold_141 466786\n",
      "ignoring indels: scaffold_141 796920\n",
      "something is wrong with the reference allele at scaffold_141 800938: TG\n",
      "ignoring indels: scaffold_141 1309410\n",
      "ignoring indels: scaffold_142 161633\n",
      "ignoring indels: scaffold_142 566569\n",
      "ignoring indels: scaffold_142 1528055\n",
      "ignoring indels: scaffold_142 1528094\n",
      "ignoring > bi-allelic SNPs: scaffold_142 1554240\n",
      "ignoring > bi-allelic SNPs: scaffold_142 1554257\n",
      "ignoring indels: scaffold_143 182576\n",
      "ignoring indels: scaffold_143 203890\n",
      "ignoring > bi-allelic SNPs: scaffold_143 1622093\n",
      "ignoring > bi-allelic SNPs: scaffold_143 1622096\n",
      "ignoring > bi-allelic SNPs: scaffold_144 476614\n",
      "ignoring indels: scaffold_145 291262\n",
      "ignoring indels: scaffold_146 1200274\n",
      "ignoring > bi-allelic SNPs: scaffold_146 1591524\n",
      "ignoring indels: scaffold_146 1720140\n",
      "ignoring indels: scaffold_147 912122\n",
      "ignoring > bi-allelic SNPs: scaffold_148 478821\n",
      "ignoring > bi-allelic SNPs: scaffold_149 156657\n",
      "ignoring indels: scaffold_149 363786\n",
      "ignoring indels: scaffold_149 409170\n",
      "ignoring indels: scaffold_149 1443118\n",
      "ignoring indels: scaffold_150 1023717\n",
      "ignoring > bi-allelic SNPs: scaffold_150 1708020\n",
      "ignoring indels: scaffold_151 277893\n",
      "ignoring > bi-allelic SNPs: scaffold_151 661901\n",
      "ignoring > bi-allelic SNPs: scaffold_151 1084827\n",
      "ignoring indels: scaffold_152 687160\n",
      "ignoring > bi-allelic SNPs: scaffold_153 615131\n",
      "ignoring > bi-allelic SNPs: scaffold_153 849798\n",
      "ignoring > bi-allelic SNPs: scaffold_153 1167324\n",
      "ignoring indels: scaffold_153 1198618\n",
      "ignoring indels: scaffold_153 1198632\n",
      "ignoring indels: scaffold_154 262094\n",
      "ignoring > bi-allelic SNPs: scaffold_154 319612\n",
      "ignoring indels: scaffold_154 510929\n",
      "ignoring indels: scaffold_154 510941\n",
      "ignoring > bi-allelic SNPs: scaffold_154 510943\n",
      "ignoring > bi-allelic SNPs: scaffold_154 510947\n",
      "ignoring indels: scaffold_154 515684\n",
      "ignoring indels: scaffold_154 651413\n",
      "ignoring indels: scaffold_154 1171997\n",
      "ignoring indels: scaffold_155 190187\n",
      "ignoring > bi-allelic SNPs: scaffold_155 190318\n",
      "ignoring indels: scaffold_155 326366\n",
      "ignoring indels: scaffold_155 326375\n",
      "ignoring > bi-allelic SNPs: scaffold_155 1053909\n",
      "ignoring > bi-allelic SNPs: scaffold_155 1053911\n",
      "ignoring > bi-allelic SNPs: scaffold_155 1053913\n",
      "ignoring > bi-allelic SNPs: scaffold_155 1053914\n",
      "ignoring > bi-allelic SNPs: scaffold_155 1053915\n",
      "ignoring > bi-allelic SNPs: scaffold_156 14942\n",
      "ignoring indels: scaffold_156 129135\n",
      "ignoring > bi-allelic SNPs: scaffold_156 683981\n",
      "ignoring indels: scaffold_156 1153630\n",
      "ignoring > bi-allelic SNPs: scaffold_156 1362736\n",
      "ignoring > bi-allelic SNPs: scaffold_157 400476\n",
      "ignoring indels: scaffold_157 532154\n",
      "ignoring indels: scaffold_157 706269\n",
      "ignoring indels: scaffold_157 812260\n",
      "ignoring indels: scaffold_157 1039768\n",
      "ignoring indels: scaffold_157 1049850\n",
      "ignoring > bi-allelic SNPs: scaffold_157 1576385\n",
      "ignoring indels: scaffold_157 1579701\n",
      "ignoring indels: scaffold_158 537424\n",
      "ignoring > bi-allelic SNPs: scaffold_158 1294560\n",
      "ignoring > bi-allelic SNPs: scaffold_159 1052904\n",
      "something is wrong with the reference allele at scaffold_160 48632: CA\n",
      "ignoring indels: scaffold_160 256639\n",
      "ignoring indels: scaffold_160 321833\n",
      "ignoring indels: scaffold_160 1422869\n",
      "ignoring indels: scaffold_161 565294\n",
      "ignoring > bi-allelic SNPs: scaffold_161 565322\n",
      "ignoring indels: scaffold_161 685905\n",
      "ignoring indels: scaffold_162 253399\n",
      "ignoring indels: scaffold_162 253411\n",
      "ignoring > bi-allelic SNPs: scaffold_162 709442\n",
      "ignoring indels: scaffold_162 1284484\n",
      "ignoring indels: scaffold_163 993448\n",
      "ignoring > bi-allelic SNPs: scaffold_163 1347766\n",
      "ignoring indels: scaffold_164 1427040\n",
      "ignoring indels: scaffold_165 764200\n",
      "ignoring > bi-allelic SNPs: scaffold_165 1048444\n",
      "ignoring indels: scaffold_166 42831\n",
      "ignoring indels: scaffold_166 965790\n",
      "ignoring indels: scaffold_167 70580\n",
      "ignoring > bi-allelic SNPs: scaffold_168 111751\n",
      "ignoring > bi-allelic SNPs: scaffold_168 424255\n",
      "ignoring > bi-allelic SNPs: scaffold_168 452871\n",
      "ignoring indels: scaffold_168 667008\n",
      "ignoring indels: scaffold_168 706794\n",
      "ignoring indels: scaffold_169 158825\n",
      "ignoring > bi-allelic SNPs: scaffold_170 734260\n",
      "ignoring > bi-allelic SNPs: scaffold_170 761526\n",
      "ignoring > bi-allelic SNPs: scaffold_170 761527\n",
      "ignoring > bi-allelic SNPs: scaffold_170 761528\n",
      "ignoring > bi-allelic SNPs: scaffold_170 761536\n",
      "ignoring > bi-allelic SNPs: scaffold_170 761576\n",
      "ignoring > bi-allelic SNPs: scaffold_170 761586\n",
      "ignoring indels: scaffold_170 1304710\n",
      "ignoring indels: scaffold_171 148404\n",
      "ignoring indels: scaffold_171 383410\n",
      "ignoring > bi-allelic SNPs: scaffold_171 682169\n",
      "ignoring indels: scaffold_171 1086483\n",
      "ignoring > bi-allelic SNPs: scaffold_172 185536\n",
      "ignoring indels: scaffold_172 185588\n",
      "ignoring indels: scaffold_172 500264\n",
      "ignoring indels: scaffold_174 77842\n",
      "ignoring > bi-allelic SNPs: scaffold_174 594899\n",
      "ignoring > bi-allelic SNPs: scaffold_174 594902\n",
      "ignoring indels: scaffold_174 683958\n",
      "ignoring > bi-allelic SNPs: scaffold_174 838298\n",
      "ignoring > bi-allelic SNPs: scaffold_174 1088613\n",
      "ignoring > bi-allelic SNPs: scaffold_174 1293572\n",
      "ignoring indels: scaffold_174 1293591\n",
      "ignoring indels: scaffold_175 392174\n",
      "ignoring indels: scaffold_175 548385\n",
      "ignoring > bi-allelic SNPs: scaffold_175 548488\n",
      "ignoring indels: scaffold_175 823688\n",
      "ignoring > bi-allelic SNPs: scaffold_176 619181\n",
      "ignoring indels: scaffold_176 1193898\n",
      "ignoring > bi-allelic SNPs: scaffold_177 42616\n",
      "ignoring indels: scaffold_178 1995\n",
      "ignoring indels: scaffold_178 77983\n",
      "ignoring indels: scaffold_178 644614\n",
      "ignoring indels: scaffold_178 644691\n",
      "ignoring indels: scaffold_178 856412\n",
      "ignoring indels: scaffold_179 823747\n",
      "ignoring indels: scaffold_179 823813\n",
      "ignoring indels: scaffold_181 338128\n",
      "ignoring indels: scaffold_181 486799\n",
      "ignoring indels: scaffold_181 525089\n",
      "ignoring > bi-allelic SNPs: scaffold_181 1099324\n",
      "ignoring > bi-allelic SNPs: scaffold_182 1025355\n",
      "ignoring indels: scaffold_182 1188074\n",
      "ignoring > bi-allelic SNPs: scaffold_182 1188110\n",
      "ignoring > bi-allelic SNPs: scaffold_182 1188119\n",
      "ignoring > bi-allelic SNPs: scaffold_183 755749\n",
      "ignoring > bi-allelic SNPs: scaffold_184 888971\n",
      "ignoring > bi-allelic SNPs: scaffold_184 951830\n",
      "ignoring > bi-allelic SNPs: scaffold_184 1118226\n",
      "ignoring indels: scaffold_185 425785\n",
      "ignoring indels: scaffold_185 516860\n",
      "ignoring indels: scaffold_185 634825\n",
      "ignoring indels: scaffold_185 716339\n",
      "ignoring indels: scaffold_186 652529\n",
      "ignoring indels: scaffold_187 357424\n",
      "ignoring > bi-allelic SNPs: scaffold_187 678827\n",
      "ignoring > bi-allelic SNPs: scaffold_188 331581\n",
      "ignoring indels: scaffold_188 1147326\n",
      "ignoring indels: scaffold_189 131690\n",
      "ignoring indels: scaffold_189 622569\n",
      "ignoring > bi-allelic SNPs: scaffold_189 622575\n",
      "ignoring indels: scaffold_189 846356\n",
      "ignoring indels: scaffold_189 1023693\n",
      "ignoring > bi-allelic SNPs: scaffold_190 281803\n",
      "ignoring > bi-allelic SNPs: scaffold_190 427788\n",
      "ignoring indels: scaffold_191 22455\n",
      "ignoring > bi-allelic SNPs: scaffold_191 1066849\n",
      "ignoring indels: scaffold_192 354619\n",
      "ignoring indels: scaffold_193 20551\n",
      "ignoring indels: scaffold_193 144386\n",
      "ignoring indels: scaffold_193 206336\n",
      "ignoring indels: scaffold_193 402198\n",
      "ignoring > bi-allelic SNPs: scaffold_194 674425\n",
      "ignoring indels: scaffold_194 1083981\n",
      "ignoring indels: scaffold_195 751302\n",
      "ignoring > bi-allelic SNPs: scaffold_195 900607\n",
      "ignoring > bi-allelic SNPs: scaffold_197 205074\n",
      "ignoring indels: scaffold_197 212083\n",
      "ignoring > bi-allelic SNPs: scaffold_198 426614\n",
      "ignoring indels: scaffold_198 738754\n",
      "ignoring > bi-allelic SNPs: scaffold_199 274980\n",
      "ignoring indels: scaffold_199 830930\n",
      "ignoring > bi-allelic SNPs: scaffold_199 831063\n",
      "ignoring indels: scaffold_200 222815\n",
      "ignoring indels: scaffold_201 417634\n",
      "ignoring indels: scaffold_203 791913\n",
      "ignoring > bi-allelic SNPs: scaffold_204 334873\n",
      "ignoring indels: scaffold_204 561454\n",
      "ignoring > bi-allelic SNPs: scaffold_205 651111\n",
      "ignoring indels: scaffold_206 703681\n",
      "ignoring indels: scaffold_206 843527\n",
      "ignoring indels: scaffold_207 2011\n",
      "ignoring indels: scaffold_209 34116\n",
      "ignoring indels: scaffold_209 258916\n",
      "ignoring indels: scaffold_209 843551\n",
      "ignoring indels: scaffold_210 666647\n",
      "ignoring > bi-allelic SNPs: scaffold_211 443822\n",
      "ignoring indels: scaffold_211 682430\n",
      "ignoring > bi-allelic SNPs: scaffold_211 896289\n",
      "ignoring indels: scaffold_212 558098\n",
      "ignoring indels: scaffold_214 443850\n",
      "ignoring indels: scaffold_214 578536\n",
      "ignoring indels: scaffold_215 175999\n",
      "ignoring indels: scaffold_215 260091\n",
      "ignoring indels: scaffold_215 374301\n",
      "ignoring indels: scaffold_216 868410\n",
      "ignoring > bi-allelic SNPs: scaffold_216 868422\n",
      "ignoring > bi-allelic SNPs: scaffold_218 315864\n",
      "ignoring indels: scaffold_218 483721\n",
      "ignoring > bi-allelic SNPs: scaffold_218 635001\n",
      "ignoring indels: scaffold_220 469576\n",
      "ignoring indels: scaffold_221 87768\n",
      "ignoring indels: scaffold_221 796853\n",
      "ignoring indels: scaffold_221 796886\n",
      "ignoring indels: scaffold_222 176241\n",
      "ignoring indels: scaffold_222 176309\n",
      "ignoring indels: scaffold_222 176317\n",
      "ignoring > bi-allelic SNPs: scaffold_223 116202\n",
      "ignoring > bi-allelic SNPs: scaffold_223 116203\n",
      "ignoring > bi-allelic SNPs: scaffold_223 116204\n",
      "ignoring > bi-allelic SNPs: scaffold_223 116205\n",
      "ignoring indels: scaffold_223 138200\n",
      "ignoring > bi-allelic SNPs: scaffold_223 186239\n",
      "ignoring indels: scaffold_223 217299\n",
      "ignoring indels: scaffold_223 373851\n",
      "ignoring indels: scaffold_224 145311\n",
      "ignoring > bi-allelic SNPs: scaffold_224 180085\n",
      "ignoring indels: scaffold_224 591425\n",
      "ignoring indels: scaffold_226 113001\n",
      "ignoring > bi-allelic SNPs: scaffold_226 204262\n",
      "ignoring indels: scaffold_227 487702\n",
      "ignoring indels: scaffold_228 41308\n",
      "ignoring indels: scaffold_228 169501\n",
      "ignoring indels: scaffold_230 578324\n",
      "ignoring indels: scaffold_232 232251\n",
      "ignoring indels: scaffold_232 398623\n",
      "ignoring > bi-allelic SNPs: scaffold_233 683491\n",
      "ignoring indels: scaffold_236 217123\n",
      "ignoring indels: scaffold_237 707322\n",
      "ignoring indels: scaffold_238 108803\n",
      "ignoring > bi-allelic SNPs: scaffold_240 60007\n",
      "ignoring > bi-allelic SNPs: scaffold_240 60012\n",
      "ignoring indels: scaffold_240 557855\n",
      "ignoring > bi-allelic SNPs: scaffold_244 220905\n",
      "ignoring indels: scaffold_245 247541\n",
      "ignoring > bi-allelic SNPs: scaffold_245 604160\n",
      "ignoring > bi-allelic SNPs: scaffold_245 604164\n",
      "ignoring > bi-allelic SNPs: scaffold_245 604169\n",
      "ignoring > bi-allelic SNPs: scaffold_245 604173\n",
      "ignoring indels: scaffold_248 115758\n",
      "ignoring indels: scaffold_248 164987\n",
      "ignoring indels: scaffold_249 50399\n",
      "ignoring indels: scaffold_249 164291\n",
      "ignoring indels: scaffold_249 168427\n",
      "ignoring indels: scaffold_249 209319\n",
      "ignoring indels: scaffold_250 59960\n",
      "ignoring indels: scaffold_250 272174\n",
      "ignoring > bi-allelic SNPs: scaffold_251 557094\n",
      "ignoring indels: scaffold_251 576320\n",
      "ignoring indels: scaffold_252 589415\n",
      "ignoring indels: scaffold_254 35449\n",
      "ignoring > bi-allelic SNPs: scaffold_254 70527\n",
      "ignoring > bi-allelic SNPs: scaffold_254 199629\n",
      "ignoring > bi-allelic SNPs: scaffold_254 201343\n",
      "ignoring > bi-allelic SNPs: scaffold_254 201352\n",
      "ignoring indels: scaffold_254 223066\n",
      "ignoring indels: scaffold_254 373748\n",
      "ignoring indels: scaffold_254 377233\n",
      "ignoring > bi-allelic SNPs: scaffold_254 393885\n",
      "ignoring > bi-allelic SNPs: scaffold_255 66703\n",
      "ignoring > bi-allelic SNPs: scaffold_255 150480\n",
      "ignoring > bi-allelic SNPs: scaffold_255 200808\n",
      "ignoring > bi-allelic SNPs: scaffold_255 200809\n",
      "ignoring > bi-allelic SNPs: scaffold_255 200811\n",
      "ignoring > bi-allelic SNPs: scaffold_255 200812\n",
      "ignoring > bi-allelic SNPs: scaffold_255 200814\n",
      "ignoring indels: scaffold_255 434089\n",
      "ignoring indels: scaffold_255 434108\n",
      "ignoring > bi-allelic SNPs: scaffold_259 402351\n",
      "ignoring indels: scaffold_259 589641\n",
      "ignoring indels: scaffold_259 601933\n",
      "ignoring > bi-allelic SNPs: scaffold_261 176346\n",
      "ignoring > bi-allelic SNPs: scaffold_261 176364\n",
      "ignoring > bi-allelic SNPs: scaffold_261 205832\n",
      "ignoring > bi-allelic SNPs: scaffold_261 489444\n",
      "ignoring > bi-allelic SNPs: scaffold_263 159244\n",
      "ignoring indels: scaffold_268 16873\n",
      "ignoring indels: scaffold_268 16882\n",
      "ignoring indels: scaffold_268 66762\n",
      "ignoring > bi-allelic SNPs: scaffold_271 69686\n",
      "ignoring indels: scaffold_271 331092\n",
      "ignoring indels: scaffold_272 52001\n",
      "ignoring indels: scaffold_272 516596\n",
      "ignoring indels: scaffold_274 201516\n",
      "ignoring indels: scaffold_274 314108\n",
      "ignoring > bi-allelic SNPs: scaffold_274 314110\n",
      "ignoring indels: scaffold_274 321878\n",
      "ignoring indels: scaffold_274 433001\n",
      "ignoring > bi-allelic SNPs: scaffold_274 433005\n",
      "ignoring > bi-allelic SNPs: scaffold_275 1252\n",
      "ignoring indels: scaffold_275 173613\n",
      "ignoring > bi-allelic SNPs: scaffold_275 213506\n",
      "ignoring > bi-allelic SNPs: scaffold_275 213633\n",
      "ignoring > bi-allelic SNPs: scaffold_276 371270\n",
      "ignoring > bi-allelic SNPs: scaffold_276 374453\n",
      "ignoring > bi-allelic SNPs: scaffold_277 334124\n",
      "ignoring indels: scaffold_277 334146\n",
      "ignoring > bi-allelic SNPs: scaffold_277 334148\n",
      "ignoring > bi-allelic SNPs: scaffold_277 334158\n",
      "ignoring > bi-allelic SNPs: scaffold_277 334175\n",
      "ignoring > bi-allelic SNPs: scaffold_277 334181\n",
      "ignoring indels: scaffold_278 332490\n",
      "ignoring > bi-allelic SNPs: scaffold_279 54466\n",
      "ignoring indels: scaffold_280 348521\n",
      "ignoring indels: scaffold_280 437034\n",
      "ignoring indels: scaffold_285 115913\n",
      "ignoring indels: scaffold_286 161851\n",
      "ignoring > bi-allelic SNPs: scaffold_286 226364\n",
      "ignoring > bi-allelic SNPs: scaffold_287 63116\n",
      "ignoring indels: scaffold_288 16219\n",
      "ignoring indels: scaffold_288 163437\n",
      "ignoring > bi-allelic SNPs: scaffold_288 176774\n",
      "ignoring indels: scaffold_291 84213\n",
      "ignoring indels: scaffold_291 385904\n",
      "ignoring > bi-allelic SNPs: scaffold_291 385908\n",
      "ignoring > bi-allelic SNPs: scaffold_291 415765\n",
      "ignoring indels: scaffold_291 462596\n",
      "ignoring > bi-allelic SNPs: scaffold_291 473229\n",
      "ignoring > bi-allelic SNPs: scaffold_292 137524\n",
      "ignoring > bi-allelic SNPs: scaffold_292 137659\n",
      "ignoring > bi-allelic SNPs: scaffold_292 137677\n",
      "ignoring > bi-allelic SNPs: scaffold_292 137678\n",
      "ignoring > bi-allelic SNPs: scaffold_292 137679\n",
      "ignoring indels: scaffold_293 128649\n",
      "ignoring indels: scaffold_293 211447\n",
      "ignoring indels: scaffold_293 288984\n",
      "ignoring > bi-allelic SNPs: scaffold_293 289064\n",
      "ignoring indels: scaffold_294 409068\n",
      "ignoring indels: scaffold_296 70053\n",
      "ignoring > bi-allelic SNPs: scaffold_300 241909\n",
      "ignoring indels: scaffold_301 137258\n",
      "ignoring indels: scaffold_301 225156\n",
      "ignoring > bi-allelic SNPs: scaffold_302 318853\n",
      "ignoring indels: scaffold_302 318906\n",
      "ignoring > bi-allelic SNPs: scaffold_304 155927\n",
      "ignoring > bi-allelic SNPs: scaffold_304 269579\n",
      "ignoring indels: scaffold_304 269618\n",
      "ignoring > bi-allelic SNPs: scaffold_305 237131\n",
      "ignoring > bi-allelic SNPs: scaffold_305 329285\n",
      "ignoring > bi-allelic SNPs: scaffold_307 146569\n",
      "ignoring > bi-allelic SNPs: scaffold_308 34174\n",
      "ignoring > bi-allelic SNPs: scaffold_308 363766\n",
      "ignoring > bi-allelic SNPs: scaffold_309 122727\n",
      "ignoring indels: scaffold_309 392416\n",
      "ignoring indels: scaffold_314 19430\n",
      "ignoring > bi-allelic SNPs: scaffold_314 220143\n",
      "ignoring > bi-allelic SNPs: scaffold_317 255901\n",
      "ignoring indels: scaffold_318 267161\n",
      "ignoring > bi-allelic SNPs: scaffold_322 43151\n",
      "ignoring > bi-allelic SNPs: scaffold_323 364830\n",
      "ignoring > bi-allelic SNPs: scaffold_327 328222\n",
      "ignoring > bi-allelic SNPs: scaffold_327 328232\n",
      "ignoring indels: scaffold_330 26118\n",
      "ignoring indels: scaffold_330 129876\n",
      "ignoring indels: scaffold_331 208329\n",
      "ignoring > bi-allelic SNPs: scaffold_331 237465\n",
      "ignoring > bi-allelic SNPs: scaffold_332 144597\n",
      "ignoring indels: scaffold_332 319338\n",
      "ignoring indels: scaffold_334 197669\n",
      "ignoring indels: scaffold_335 20501\n",
      "ignoring > bi-allelic SNPs: scaffold_335 20591\n",
      "ignoring indels: scaffold_337 136650\n",
      "ignoring indels: scaffold_339 91507\n",
      "ignoring indels: scaffold_341 76934\n",
      "ignoring indels: scaffold_341 199347\n",
      "ignoring indels: scaffold_342 62123\n",
      "ignoring > bi-allelic SNPs: scaffold_343 224376\n",
      "ignoring indels: scaffold_343 277774\n",
      "ignoring > bi-allelic SNPs: scaffold_344 7389\n",
      "ignoring > bi-allelic SNPs: scaffold_344 180137\n",
      "ignoring indels: scaffold_344 260564\n",
      "ignoring indels: scaffold_345 13003\n",
      "ignoring > bi-allelic SNPs: scaffold_345 123604\n",
      "ignoring indels: scaffold_345 272929\n",
      "ignoring > bi-allelic SNPs: scaffold_345 272943\n",
      "something is wrong with the reference allele at scaffold_347 4529: CT\n",
      "ignoring > bi-allelic SNPs: scaffold_348 211385\n",
      "ignoring > bi-allelic SNPs: scaffold_348 211405\n",
      "ignoring indels: scaffold_348 256853\n",
      "ignoring > bi-allelic SNPs: scaffold_352 51052\n",
      "ignoring > bi-allelic SNPs: scaffold_353 234343\n",
      "ignoring > bi-allelic SNPs: scaffold_353 234382\n",
      "ignoring indels: scaffold_355 213909\n",
      "ignoring indels: scaffold_355 275262\n",
      "ignoring > bi-allelic SNPs: scaffold_357 128765\n",
      "ignoring > bi-allelic SNPs: scaffold_358 125831\n",
      "ignoring > bi-allelic SNPs: scaffold_362 142281\n",
      "ignoring indels: scaffold_365 177063\n",
      "ignoring indels: scaffold_366 44333\n",
      "ignoring indels: scaffold_368 104690\n",
      "ignoring > bi-allelic SNPs: scaffold_372 224761\n",
      "ignoring indels: scaffold_379 42852\n",
      "ignoring indels: scaffold_379 157234\n",
      "ignoring > bi-allelic SNPs: scaffold_380 143473\n",
      "ignoring > bi-allelic SNPs: scaffold_385 37393\n",
      "ignoring > bi-allelic SNPs: scaffold_386 131355\n",
      "ignoring indels: scaffold_389 85048\n",
      "ignoring > bi-allelic SNPs: scaffold_389 85113\n",
      "ignoring > bi-allelic SNPs: scaffold_393 117947\n",
      "ignoring indels: scaffold_393 146276\n",
      "ignoring indels: scaffold_400 96870\n",
      "ignoring > bi-allelic SNPs: scaffold_401 64863\n",
      "ignoring indels: scaffold_401 115958\n",
      "something is wrong with the reference allele at scaffold_405 11561: AC\n",
      "ignoring > bi-allelic SNPs: scaffold_416 129006\n",
      "ignoring > bi-allelic SNPs: scaffold_421 101118\n",
      "ignoring indels: scaffold_425 115445\n",
      "ignoring indels: scaffold_427 37615\n",
      "ignoring indels: scaffold_428 41159\n",
      "ignoring indels: scaffold_431 4783\n",
      "ignoring indels: scaffold_435 94434\n",
      "ignoring indels: scaffold_438 1537\n",
      "ignoring indels: scaffold_438 6920\n",
      "ignoring > bi-allelic SNPs: scaffold_440 94842\n",
      "ignoring indels: scaffold_469 16198\n",
      "ignoring > bi-allelic SNPs: scaffold_479 15707\n",
      "ignoring > bi-allelic SNPs: scaffold_495 54175\n",
      "ignoring > bi-allelic SNPs: scaffold_497 43291\n",
      "ignoring > bi-allelic SNPs: scaffold_518 40177\n",
      "ignoring indels: scaffold_518 43128\n",
      "ignoring indels: scaffold_518 43171\n",
      "ignoring indels: scaffold_520 59202\n",
      "ignoring > bi-allelic SNPs: scaffold_531 35863\n",
      "ignoring indels: scaffold_542 5556\n",
      "ignoring > bi-allelic SNPs: scaffold_596 24425\n",
      "ignoring indels: scaffold_611 9334\n",
      "ignoring indels: scaffold_629 9967\n",
      "ignoring > bi-allelic SNPs: scaffold_630 9661\n",
      "ignoring > bi-allelic SNPs: scaffold_630 9668\n",
      "ignoring indels: scaffold_634 9380\n",
      "ignoring > bi-allelic SNPs: scaffold_637 17839\n",
      "ignoring > bi-allelic SNPs: scaffold_637 17889\n",
      "ignoring indels: scaffold_645 10981\n",
      "ignoring indels: scaffold_690 13956\n",
      "ignoring indels: scaffold_719 12884\n",
      "ignoring > bi-allelic SNPs: scaffold_774 1836\n",
      "ignoring indels: scaffold_774 1848\n",
      "ignoring indels: scaffold_776 6472\n",
      "ignoring > bi-allelic SNPs: scaffold_776 6568\n",
      "ignoring > bi-allelic SNPs: scaffold_874 5937\n",
      "ignoring indels: scaffold_909 1553\n",
      "ignoring > bi-allelic SNPs: scaffold_1009 1279\n",
      "ignoring indels: scaffold_1020 1545\n",
      "ignoring > bi-allelic SNPs: scaffold_1022 6549\n",
      "ignoring > bi-allelic SNPs: scaffold_1063 6131\n",
      "ignoring > bi-allelic SNPs: scaffold_1063 6181\n",
      "ignoring > bi-allelic SNPs: scaffold_1098 5345\n",
      "ignoring > bi-allelic SNPs: scaffold_1124 1048\n",
      "ignoring indels: scaffold_1124 1065\n",
      "ignoring > bi-allelic SNPs: scaffold_1456 914\n",
      "ignoring indels: scaffold_1456 927\n",
      "ignoring > bi-allelic SNPs: scaffold_1456 932\n",
      "ignoring indels: scaffold_1504 2288\n",
      "ignoring indels: scaffold_1608 2512\n",
      "ignoring > bi-allelic SNPs: scaffold_1608 2535\n",
      "ignoring > bi-allelic SNPs: scaffold_1608 2557\n",
      "ignoring > bi-allelic SNPs: scaffold_1608 2563\n",
      "ignoring indels: scaffold_2417 1361\n",
      "ignoring indels: scaffold_2723 841\n",
      "ignoring indels: scaffold_2794 162\n",
      "ignoring indels: scaffold_2827 181\n",
      "ignoring > bi-allelic SNPs: scaffold_2857 1154\n",
      "ignoring > bi-allelic SNPs: scaffold_2989 810\n",
      "ignoring indels: scaffold_3380 792\n",
      "ignoring > bi-allelic SNPs: scaffold_3460 632\n",
      "ignoring > bi-allelic SNPs: scaffold_3460 651\n",
      "writing to fasta\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "/media/chrishah/STORAGE/Dropbox/Github/genomisc/popogeno/vcf_2_consensus.py SNP-calling/full.vcf -p ../popmap -o SNP-calling/full \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "run RAXML without outgroup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "RAxML can't, parse the alignment file as phylip file \n",
      "it will now try to parse it as FASTA file\n",
      "\n",
      "IMPORTANT WARNING: Alignment column 185231 contains only undetermined values which will be treated as missing data\n",
      "IMPORTANT WARNING: Alignment column 347399 contains only undetermined values which will be treated as missing data\n",
      "IMPORTANT WARNING: Alignment column 532781 contains only undetermined values which will be treated as missing data\n",
      "IMPORTANT WARNING: Alignment column 552907 contains only undetermined values which will be treated as missing data\n",
      "\n",
      "IMPORTANT WARNING\n",
      "Found 4 columns that contain only undetermined values which will be treated as missing data.\n",
      "Normally these columns should be excluded from the analysis.\n",
      "\n",
      "Just in case you might need it, an alignment file with \n",
      "undetermined columns removed is printed to file ../SNP-calling/full.fas.reduced\n",
      "\n",
      "\n",
      "Using BFGS method to optimize GTR rate parameters, to disable this specify \"--no-bfgs\" \n",
      "\n",
      "\n",
      "Alignment has 4 completely undetermined sites that will be automatically removed from the input data\n",
      "\n",
      "\n",
      "This is the RAxML Master Pthread\n",
      "\n",
      "This is RAxML Worker Pthread Number: 2\n",
      "\n",
      "This is RAxML Worker Pthread Number: 1\n",
      "\n",
      "\n",
      "This is RAxML version 8.2.4 released by Alexandros Stamatakis on October 02 2015.\n",
      "\n",
      "With greatly appreciated code contributions by:\n",
      "Andre Aberer      (HITS)\n",
      "Simon Berger      (HITS)\n",
      "Alexey Kozlov     (HITS)\n",
      "Kassian Kobert    (HITS)\n",
      "David Dao         (KIT and HITS)\n",
      "Nick Pattengale   (Sandia)\n",
      "Wayne Pfeiffer    (SDSC)\n",
      "Akifumi S. Tanabe (NRIFS)\n",
      "\n",
      "Alignment has 24403 distinct alignment patterns\n",
      "\n",
      "Proportion of gaps and completely undetermined characters in this alignment: 2.43%\n",
      "\n",
      "RAxML rapid bootstrapping and subsequent ML search\n",
      "\n",
      "Using 1 distinct models/data partitions with joint branch length optimization\n",
      "\n",
      "\n",
      "\n",
      "Executing 100 rapid bootstrap inferences and thereafter a thorough ML search \n",
      "\n",
      "All free model parameters will be estimated by RAxML\n",
      "GAMMA model of rate heteorgeneity, ML estimate of alpha-parameter\n",
      "\n",
      "GAMMA Model parameters will be estimated up to an accuracy of 0.1000000000 Log Likelihood units\n",
      "\n",
      "Partition: 0\n",
      "Alignment Patterns: 24403\n",
      "Name: No Name Provided\n",
      "DataType: DNA\n",
      "Substitution Matrix: GTR\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "RAxML was called as follows:\n",
      "\n",
      "/home/chrishah/src/RAxML/RAxML-8.2.4/raxmlHPC-PTHREADS-SSE3 -f a -T 3 -x 12345 -p 12345 -N 100 -m GTRGAMMA -s ../SNP-calling/full.fas -n full \n",
      "\n",
      "\n",
      "\n",
      "Time for BS model parameter optimization 3.684232\n",
      "Bootstrap[0]: Time 25.526402 seconds, bootstrap likelihood -1577683.423976, best rearrangement setting 13\n",
      "Bootstrap[1]: Time 22.429604 seconds, bootstrap likelihood -1577861.390213, best rearrangement setting 13\n",
      "Bootstrap[2]: Time 20.537349 seconds, bootstrap likelihood -1577995.397829, best rearrangement setting 12\n",
      "Bootstrap[3]: Time 23.557997 seconds, bootstrap likelihood -1577220.817098, best rearrangement setting 15\n",
      "Bootstrap[4]: Time 27.422666 seconds, bootstrap likelihood -1578873.821397, best rearrangement setting 10\n",
      "Bootstrap[5]: Time 28.335568 seconds, bootstrap likelihood -1577327.820706, best rearrangement setting 10\n",
      "Bootstrap[6]: Time 25.874182 seconds, bootstrap likelihood -1577028.983167, best rearrangement setting 6\n",
      "Bootstrap[7]: Time 25.607515 seconds, bootstrap likelihood -1579508.149847, best rearrangement setting 14\n",
      "Bootstrap[8]: Time 25.228460 seconds, bootstrap likelihood -1578594.181444, best rearrangement setting 5\n",
      "Bootstrap[9]: Time 28.017461 seconds, bootstrap likelihood -1580580.462028, best rearrangement setting 7\n",
      "Bootstrap[10]: Time 24.012660 seconds, bootstrap likelihood -1573700.245556, best rearrangement setting 7\n",
      "Bootstrap[11]: Time 20.885405 seconds, bootstrap likelihood -1579692.752919, best rearrangement setting 5\n",
      "Bootstrap[12]: Time 24.761974 seconds, bootstrap likelihood -1575234.084756, best rearrangement setting 8\n",
      "Bootstrap[13]: Time 23.471326 seconds, bootstrap likelihood -1578927.320601, best rearrangement setting 7\n",
      "Bootstrap[14]: Time 23.925642 seconds, bootstrap likelihood -1575337.482067, best rearrangement setting 12\n",
      "Bootstrap[15]: Time 24.087135 seconds, bootstrap likelihood -1577053.525815, best rearrangement setting 15\n",
      "Bootstrap[16]: Time 25.109293 seconds, bootstrap likelihood -1576550.481359, best rearrangement setting 12\n",
      "Bootstrap[17]: Time 23.705732 seconds, bootstrap likelihood -1575680.657930, best rearrangement setting 12\n",
      "Bootstrap[18]: Time 21.191630 seconds, bootstrap likelihood -1577578.550687, best rearrangement setting 5\n",
      "Bootstrap[19]: Time 26.204406 seconds, bootstrap likelihood -1580589.091180, best rearrangement setting 8\n",
      "Bootstrap[20]: Time 23.241338 seconds, bootstrap likelihood -1576500.732076, best rearrangement setting 10\n",
      "Bootstrap[21]: Time 19.683088 seconds, bootstrap likelihood -1578456.069539, best rearrangement setting 6\n",
      "Bootstrap[22]: Time 25.312045 seconds, bootstrap likelihood -1577807.460478, best rearrangement setting 5\n",
      "Bootstrap[23]: Time 27.060160 seconds, bootstrap likelihood -1576695.534149, best rearrangement setting 7\n",
      "Bootstrap[24]: Time 21.981240 seconds, bootstrap likelihood -1577389.310388, best rearrangement setting 8\n",
      "Bootstrap[25]: Time 22.318373 seconds, bootstrap likelihood -1575872.869066, best rearrangement setting 6\n",
      "Bootstrap[26]: Time 28.325222 seconds, bootstrap likelihood -1577414.450791, best rearrangement setting 12\n",
      "Bootstrap[27]: Time 22.842872 seconds, bootstrap likelihood -1575551.454920, best rearrangement setting 10\n",
      "Bootstrap[28]: Time 23.564780 seconds, bootstrap likelihood -1578808.925042, best rearrangement setting 11\n",
      "Bootstrap[29]: Time 24.010152 seconds, bootstrap likelihood -1575077.129293, best rearrangement setting 12\n",
      "Bootstrap[30]: Time 26.378783 seconds, bootstrap likelihood -1577679.168790, best rearrangement setting 15\n",
      "Bootstrap[31]: Time 27.020836 seconds, bootstrap likelihood -1578840.101172, best rearrangement setting 15\n",
      "Bootstrap[32]: Time 20.803844 seconds, bootstrap likelihood -1577400.625118, best rearrangement setting 6\n",
      "Bootstrap[33]: Time 27.567185 seconds, bootstrap likelihood -1579473.468075, best rearrangement setting 15\n",
      "Bootstrap[34]: Time 24.294893 seconds, bootstrap likelihood -1577486.751263, best rearrangement setting 11\n",
      "Bootstrap[35]: Time 19.635519 seconds, bootstrap likelihood -1577353.031636, best rearrangement setting 8\n",
      "Bootstrap[36]: Time 22.046072 seconds, bootstrap likelihood -1574200.847938, best rearrangement setting 9\n",
      "Bootstrap[37]: Time 24.752924 seconds, bootstrap likelihood -1576751.584033, best rearrangement setting 11\n",
      "Bootstrap[38]: Time 26.067565 seconds, bootstrap likelihood -1577256.850604, best rearrangement setting 8\n",
      "Bootstrap[39]: Time 23.341581 seconds, bootstrap likelihood -1580487.431731, best rearrangement setting 13\n",
      "Bootstrap[40]: Time 23.745583 seconds, bootstrap likelihood -1577073.305652, best rearrangement setting 13\n",
      "Bootstrap[41]: Time 23.581542 seconds, bootstrap likelihood -1577902.249540, best rearrangement setting 10\n",
      "Bootstrap[42]: Time 21.166637 seconds, bootstrap likelihood -1578189.745227, best rearrangement setting 5\n",
      "Bootstrap[43]: Time 26.731785 seconds, bootstrap likelihood -1578760.408396, best rearrangement setting 14\n",
      "Bootstrap[44]: Time 23.611732 seconds, bootstrap likelihood -1578493.626780, best rearrangement setting 10\n",
      "Bootstrap[45]: Time 23.503939 seconds, bootstrap likelihood -1578584.526689, best rearrangement setting 15\n",
      "Bootstrap[46]: Time 23.358741 seconds, bootstrap likelihood -1574944.527291, best rearrangement setting 14\n",
      "Bootstrap[47]: Time 26.477846 seconds, bootstrap likelihood -1575711.614559, best rearrangement setting 8\n",
      "Bootstrap[48]: Time 24.407996 seconds, bootstrap likelihood -1574891.411893, best rearrangement setting 6\n",
      "Bootstrap[49]: Time 28.367236 seconds, bootstrap likelihood -1579276.454205, best rearrangement setting 7\n",
      "Bootstrap[50]: Time 24.712262 seconds, bootstrap likelihood -1580591.654509, best rearrangement setting 7\n",
      "Bootstrap[51]: Time 20.698222 seconds, bootstrap likelihood -1579591.097419, best rearrangement setting 11\n",
      "Bootstrap[52]: Time 26.604018 seconds, bootstrap likelihood -1574786.279078, best rearrangement setting 12\n",
      "Bootstrap[53]: Time 22.647537 seconds, bootstrap likelihood -1575507.968067, best rearrangement setting 8\n",
      "Bootstrap[54]: Time 23.063588 seconds, bootstrap likelihood -1578116.183364, best rearrangement setting 9\n",
      "Bootstrap[55]: Time 24.049229 seconds, bootstrap likelihood -1577766.121317, best rearrangement setting 11\n",
      "Bootstrap[56]: Time 17.861769 seconds, bootstrap likelihood -1576579.593812, best rearrangement setting 5\n",
      "Bootstrap[57]: Time 18.768854 seconds, bootstrap likelihood -1577478.085057, best rearrangement setting 7\n",
      "Bootstrap[58]: Time 19.740183 seconds, bootstrap likelihood -1577528.789321, best rearrangement setting 10\n",
      "Bootstrap[59]: Time 21.157268 seconds, bootstrap likelihood -1577717.453829, best rearrangement setting 9\n",
      "Bootstrap[60]: Time 22.680381 seconds, bootstrap likelihood -1577079.216720, best rearrangement setting 7\n",
      "Bootstrap[61]: Time 21.134120 seconds, bootstrap likelihood -1576219.832760, best rearrangement setting 5\n",
      "Bootstrap[62]: Time 22.596063 seconds, bootstrap likelihood -1578747.440978, best rearrangement setting 11\n",
      "Bootstrap[63]: Time 25.248623 seconds, bootstrap likelihood -1578681.046779, best rearrangement setting 12\n",
      "Bootstrap[64]: Time 23.595979 seconds, bootstrap likelihood -1576061.581624, best rearrangement setting 7\n",
      "Bootstrap[65]: Time 21.816283 seconds, bootstrap likelihood -1576402.849246, best rearrangement setting 5\n",
      "Bootstrap[66]: Time 21.301541 seconds, bootstrap likelihood -1576653.729953, best rearrangement setting 5\n",
      "Bootstrap[67]: Time 22.511783 seconds, bootstrap likelihood -1577058.034087, best rearrangement setting 15\n",
      "Bootstrap[68]: Time 23.679826 seconds, bootstrap likelihood -1576022.515466, best rearrangement setting 11\n",
      "Bootstrap[69]: Time 19.521625 seconds, bootstrap likelihood -1576745.741940, best rearrangement setting 5\n",
      "Bootstrap[70]: Time 21.490271 seconds, bootstrap likelihood -1577536.282986, best rearrangement setting 8\n",
      "Bootstrap[71]: Time 25.546326 seconds, bootstrap likelihood -1579973.306722, best rearrangement setting 10\n",
      "Bootstrap[72]: Time 18.047937 seconds, bootstrap likelihood -1580221.173526, best rearrangement setting 11\n",
      "Bootstrap[73]: Time 27.267027 seconds, bootstrap likelihood -1575217.115975, best rearrangement setting 9\n",
      "Bootstrap[74]: Time 18.119592 seconds, bootstrap likelihood -1578738.025784, best rearrangement setting 5\n",
      "Bootstrap[75]: Time 24.230115 seconds, bootstrap likelihood -1574867.263389, best rearrangement setting 8\n",
      "Bootstrap[76]: Time 28.437915 seconds, bootstrap likelihood -1578146.893859, best rearrangement setting 9\n",
      "Bootstrap[77]: Time 25.756987 seconds, bootstrap likelihood -1579145.649874, best rearrangement setting 11\n",
      "Bootstrap[78]: Time 24.225362 seconds, bootstrap likelihood -1578408.430781, best rearrangement setting 15\n",
      "Bootstrap[79]: Time 27.818555 seconds, bootstrap likelihood -1575771.142030, best rearrangement setting 9\n",
      "Bootstrap[80]: Time 26.032879 seconds, bootstrap likelihood -1576460.783881, best rearrangement setting 15\n",
      "Bootstrap[81]: Time 22.847995 seconds, bootstrap likelihood -1576086.999374, best rearrangement setting 7\n",
      "Bootstrap[82]: Time 21.685363 seconds, bootstrap likelihood -1574032.458411, best rearrangement setting 6\n",
      "Bootstrap[83]: Time 22.881390 seconds, bootstrap likelihood -1576098.663172, best rearrangement setting 10\n",
      "Bootstrap[84]: Time 22.771134 seconds, bootstrap likelihood -1575850.675086, best rearrangement setting 11\n",
      "Bootstrap[85]: Time 19.693489 seconds, bootstrap likelihood -1575930.617189, best rearrangement setting 8\n",
      "Bootstrap[86]: Time 26.672232 seconds, bootstrap likelihood -1577201.661895, best rearrangement setting 7\n",
      "Bootstrap[87]: Time 27.146581 seconds, bootstrap likelihood -1576086.794157, best rearrangement setting 7\n",
      "Bootstrap[88]: Time 26.250088 seconds, bootstrap likelihood -1577011.181140, best rearrangement setting 12\n",
      "Bootstrap[89]: Time 25.138998 seconds, bootstrap likelihood -1579144.421833, best rearrangement setting 10\n",
      "Bootstrap[90]: Time 24.097543 seconds, bootstrap likelihood -1577253.911812, best rearrangement setting 8\n",
      "Bootstrap[91]: Time 22.482893 seconds, bootstrap likelihood -1577636.789053, best rearrangement setting 12\n",
      "Bootstrap[92]: Time 20.074224 seconds, bootstrap likelihood -1578578.612860, best rearrangement setting 13\n",
      "Bootstrap[93]: Time 25.653958 seconds, bootstrap likelihood -1576110.145478, best rearrangement setting 15\n",
      "Bootstrap[94]: Time 25.201391 seconds, bootstrap likelihood -1576599.666913, best rearrangement setting 5\n",
      "Bootstrap[95]: Time 21.838741 seconds, bootstrap likelihood -1579366.552549, best rearrangement setting 5\n",
      "Bootstrap[96]: Time 20.659994 seconds, bootstrap likelihood -1578302.546713, best rearrangement setting 13\n",
      "Bootstrap[97]: Time 22.521713 seconds, bootstrap likelihood -1578058.071763, best rearrangement setting 11\n",
      "Bootstrap[98]: Time 22.485993 seconds, bootstrap likelihood -1579537.272774, best rearrangement setting 6\n",
      "Bootstrap[99]: Time 20.804516 seconds, bootstrap likelihood -1575217.550169, best rearrangement setting 14\n",
      "\n",
      "\n",
      "Overall Time for 100 Rapid Bootstraps 2381.528085 seconds\n",
      "Average Time per Rapid Bootstrap 23.815281 seconds\n",
      "\n",
      "Starting ML Search ...\n",
      "\n",
      "Fast ML optimization finished\n",
      "\n",
      "Fast ML search Time: 1855.085647 seconds\n",
      "\n",
      "Slow ML Search 0 Likelihood: -1577904.967106\n",
      "Slow ML Search 1 Likelihood: -1577904.966858\n",
      "Slow ML Search 2 Likelihood: -1577904.967176\n",
      "Slow ML Search 3 Likelihood: -1577904.967807\n",
      "Slow ML Search 4 Likelihood: -1577904.966921\n",
      "Slow ML Search 5 Likelihood: -1577904.967708\n",
      "Slow ML Search 6 Likelihood: -1577904.967640\n",
      "Slow ML Search 7 Likelihood: -1577906.146881\n",
      "Slow ML Search 8 Likelihood: -1577906.148016\n",
      "Slow ML Search 9 Likelihood: -1577906.146620\n",
      "Slow ML optimization finished\n",
      "\n",
      "Slow ML search Time: 502.168259 seconds\n",
      "Thorough ML search Time: 74.692727 seconds\n",
      "\n",
      "Final ML Optimization Likelihood: -1577904.965552\n",
      "\n",
      "Model Information:\n",
      "\n",
      "Model Parameters of Partition 0, Name: No Name Provided, Type of Data: DNA\n",
      "alpha: 0.020000\n",
      "Tree-Length: 0.012891\n",
      "rate A <-> C: 1.098404\n",
      "rate A <-> G: 3.430319\n",
      "rate A <-> T: 0.838597\n",
      "rate C <-> G: 0.915865\n",
      "rate C <-> T: 3.396612\n",
      "rate G <-> T: 1.000000\n",
      "\n",
      "freq pi(A): 0.267869\n",
      "freq pi(C): 0.231344\n",
      "freq pi(G): 0.232576\n",
      "freq pi(T): 0.268211\n",
      "\n",
      "\n",
      "ML search took 2431.947318 secs or 0.675541 hours\n",
      "\n",
      "Combined Bootstrap and ML search took 4813.475428 secs or 1.337077 hours\n",
      "\n",
      "Drawing Bootstrap Support Values on best-scoring ML tree ...\n",
      "\n",
      "\n",
      "\n",
      "Found 1 tree in File /media/chrishah/STORAGE/RAD/stacks/ALL/mapping/excl_PCR_dupl/BWA-8MM/M_zebra/3-populations/Diplotaxodon_4pop/m5_mpop5_kernel_iterate_ONE_SNP_PER_TAG_EXCL_SINGLETONS_BOOTSTRAP/r_0.8-p_4/Phylogeny/incl_monomorphic/RAXML-unrooted/RAxML_bestTree.full\n",
      "\n",
      "\n",
      "\n",
      "Found 1 tree in File /media/chrishah/STORAGE/RAD/stacks/ALL/mapping/excl_PCR_dupl/BWA-8MM/M_zebra/3-populations/Diplotaxodon_4pop/m5_mpop5_kernel_iterate_ONE_SNP_PER_TAG_EXCL_SINGLETONS_BOOTSTRAP/r_0.8-p_4/Phylogeny/incl_monomorphic/RAXML-unrooted/RAxML_bestTree.full\n",
      "\n",
      "Program execution info written to /media/chrishah/STORAGE/RAD/stacks/ALL/mapping/excl_PCR_dupl/BWA-8MM/M_zebra/3-populations/Diplotaxodon_4pop/m5_mpop5_kernel_iterate_ONE_SNP_PER_TAG_EXCL_SINGLETONS_BOOTSTRAP/r_0.8-p_4/Phylogeny/incl_monomorphic/RAXML-unrooted/RAxML_info.full\n",
      "All 100 bootstrapped trees written to: /media/chrishah/STORAGE/RAD/stacks/ALL/mapping/excl_PCR_dupl/BWA-8MM/M_zebra/3-populations/Diplotaxodon_4pop/m5_mpop5_kernel_iterate_ONE_SNP_PER_TAG_EXCL_SINGLETONS_BOOTSTRAP/r_0.8-p_4/Phylogeny/incl_monomorphic/RAXML-unrooted/RAxML_bootstrap.full\n",
      "\n",
      "Best-scoring ML tree written to: /media/chrishah/STORAGE/RAD/stacks/ALL/mapping/excl_PCR_dupl/BWA-8MM/M_zebra/3-populations/Diplotaxodon_4pop/m5_mpop5_kernel_iterate_ONE_SNP_PER_TAG_EXCL_SINGLETONS_BOOTSTRAP/r_0.8-p_4/Phylogeny/incl_monomorphic/RAXML-unrooted/RAxML_bestTree.full\n",
      "\n",
      "Best-scoring ML tree with support values written to: /media/chrishah/STORAGE/RAD/stacks/ALL/mapping/excl_PCR_dupl/BWA-8MM/M_zebra/3-populations/Diplotaxodon_4pop/m5_mpop5_kernel_iterate_ONE_SNP_PER_TAG_EXCL_SINGLETONS_BOOTSTRAP/r_0.8-p_4/Phylogeny/incl_monomorphic/RAXML-unrooted/RAxML_bipartitions.full\n",
      "\n",
      "Best-scoring ML tree with support values as branch labels written to: /media/chrishah/STORAGE/RAD/stacks/ALL/mapping/excl_PCR_dupl/BWA-8MM/M_zebra/3-populations/Diplotaxodon_4pop/m5_mpop5_kernel_iterate_ONE_SNP_PER_TAG_EXCL_SINGLETONS_BOOTSTRAP/r_0.8-p_4/Phylogeny/incl_monomorphic/RAXML-unrooted/RAxML_bipartitionsBranchLabels.full\n",
      "\n",
      "Overall execution time for full ML analysis: 4813.480640 secs or 1.337078 hours or 0.055712 days\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir RAXML-unrooted\n",
    "cd RAXML-unrooted\n",
    "/home/chrishah/src/RAxML/RAxML-8.2.4/raxmlHPC-PTHREADS-SSE3 -f a -T 3 -x 12345 -p 12345 -N 100 -m GTRGAMMA -s ../SNP-calling/full.fas -n full\n",
    "cd .."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
